<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/all-faq.html" />
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/" />

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material" />
<meta name="DC.type" content="text" />
<meta name="DC.title" content="" />
<meta name="DC.publisher" content="Galaxy Training Network" />
<meta name="DC.date" content="1970-01-01 00:33:41 +0000" />

        
        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:site_name" content="Galaxy Training Network" />
        <meta property="og:title" content="Galaxy Training" />
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />

    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/all-faq.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/all-faq.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-05-01/all-faq.html" title="Version 2022-05-01">2022-05-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-04-01/all-faq.html" title="Version 2022-04-01">2022-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/all-faq.html" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        



<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item "><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
     
     

     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
<!--<div class="contributors-line">Authors: 
    </div>-->
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="tutorial-faqs"> <a href="#tutorial-faqs"> Tutorial Questions </a></h3>

<hr/>



<h4 class="faq-area"> It says I already have an account when registering for ecology.use<span class="notranslate">galaxy</span>.eu <a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-it-says-i-already-have-an-account-when-registering-for-ecologyusegalaxyeu"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: It says I already have an account when registering for ecology.use<span class="notranslate">galaxy</span>.eu</h3>  <p>The <a href="https://usegalaxy.eu">ecology.use<span class="notranslate">galaxy</span>.eu</a> (and any other <span class="notranslate">Galaxy</span> server ending in <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.eu</code>) is the SAME server as the regular <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a> server, just modified for Ecology analyses.</p>  <p>You can use the <strong>SAME credentials</strong> you used to register on use<span class="notranslate">galaxy</span>.eu to log into the ecology server.</p>  <p>If you do not have an account on <span class="notranslate">Galaxy</span> EU yet, will need to create one.</p></blockquote>




<h4 class="faq-area"> Automatically trim adapters (without providing custom sequences) <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-automatically-trim-adapters-without-providing-custom-sequences"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Automatically trim adapters (without providing custom sequences)</h3>  <p>There are many tools for this: Trimmomatic, Trim Galore, and a few others (search: ‚ÄúTrim‚Äù).In some of these there are options to automatically trim adaptors, but they are not so specific to the sequence you are working on necessarily.</p></blockquote>




<h4 class="faq-area"> What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library? <a href="/training-material/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-advantages-does-a-chromatogram-library-have-over-a-dda-generated-library-or-predicted-spectral-library"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library?</h3>  <p>While generating a Chromatogram Library is the most time consuming step of the EncyclopeDIA <span class="notranslate">workflow</span>, it is beneficial to DIA data analysis. DIA is a novel technique and methods for DIA data analysis are still being developed. One method commonly used includes searching DIA data against DDA-generated libraries. However, there are limitations in this method. Firstly, DDA-generated libraries are not always an accurate representation of DIA data: differences in the methods of data collection play an important role in the efficacy of the library. Secondly, DDA-generated libraries often require labs to run completely separate DDA experiments to simply generate a library with which to analyze their DIA data.Chromatogram Libraries mitigate some of the previous shortcomings mentioned. DIA data is incorporated into the generation of the Chromatogram Library and therefore provides context to the DIA data being analyzed. Secondly, the ELIB format of the Chromatogram Library allows for extra data to be included in the analysis of the DIA data, including intensity, m/z ratio, and retention time compared to the use of a DDA-generated DLIB library. Lastly, a Chromatogram Library can be generated without the use of a spectral library (as mentioned in the last question). Therefore, it is possible to forgo DDA data collection as the DLIB DDA-generated library is not strictly needed for Chromatogram Library generation and to run the EncyclopeDIA <span class="notranslate">workflow</span> (saving time and resources).</p></blockquote>




<h4 class="faq-area"> Why is Alevin is not working? <a href="/training-material/topics/transcriptomics/tutorials/droplet-quantification-preprocessing/faqs/alevin_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><p>Check your tool version, you need to use <code class="language-plaintext highlighter-rouge">1.3.0+<span class="notranslate">galaxy</span>2</code></p><p>Follow <a href="/training-material/faqs/galaxy/tools_change_version.html">these instructions</a> to switch between tool versions.</p>




<h4 class="faq-area"> Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-is-it-possible-to-replace-the-existing-alignment-tools-such-as-hisat-and-freebayes-with-other-tools"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools?</h3>  <p>The tools in this <span class="notranslate">workflow</span> are customizable, however, the user has to ensure that the inputs are in the correct format, while using the same reference genome database.</p></blockquote>




<h4 class="faq-area"> Is it possible to use alte<span class="notranslate">rna</span>tive tools to those proposed in the tutorial? <a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-is-it-possible-to-use-alternative-tools-to-those-proposed-in-the-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Is it possible to use alte<span class="notranslate">rna</span>tive tools to those proposed in the tutorial?</h3>  <p>Yes! There are many tools whose functionality are similar (e.g. Illumina <span class="notranslate">reads</span> can be mapped by using HISAT2 instead of Bowtie2).</p></blockquote>




<h4 class="faq-area"> AnnData Import/ AnnData Manipulate not working? <a href="/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-anndata-import-anndata-manipulate-not-working"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: AnnData Import/ AnnData Manipulate not working?</h3>  <p>This is a known issue, please do not use version <code class="language-plaintext highlighter-rouge">0.7.4</code> of the tool, and use version <code class="language-plaintext highlighter-rouge">0.6.2</code> instead. The  Inspect AnnData tool should work fine however.</p>  <p><a href="/training-material/faqs/galaxy/tools_change_version.html">Switching tool versions</a></p></blockquote>




<h4 class="faq-area"> Why does my assembly graph in Bandage look different to the one pictured in the tutorial? <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-does-my-assembly-graph-in-bandage-look-different-to-the-one-pictured-in-the-tutorial"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why does my assembly graph in Bandage look different to the one pictured in the tutorial?</h3>  <p>The assembly process in Flye is heuristic, and the resulting assembly will not necessarily be exactly the same each time. This may happen even if running the same data with the same version of Flye. It can also happen with a different version of Flye.</p>  <p>To make things more complicated (stop reading now if you would like!)‚Ä¶ the chloroplast genome has a structure that includes repeats (the inverted repeats), and, the small-single-copy region of the chloroplast exists in two orientations between these repeats. So, sometimes the assembly will be a perfect circle, sometimes the inverted repeats will be collapsed into one piece, and sometimes the small-single-copy region will be attached ambiguously.  To make things even more complicated‚Ä¶the chloroplast genome may even be a dynamic structure, due to flip flop recombination.</p>  <p>For more see <a href="https://academic.oup.com/gbe/article/11/12/3372/5637229">this article</a></p></blockquote>




<h4 class="faq-area"> The folder `recipes/belerophon/` and the file `meta.yaml` already exist in bioconda? <a href="/training-material/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-the-folder-recipesbelerophon-and-the-file-metayaml-already-exist-in-bioconda"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> The folder <code class="language-plaintext highlighter-rouge">recipes/belerophon/</code> and the file <code class="language-plaintext highlighter-rouge">meta.yaml</code> already exist in bioconda?</h3>  <p>The recipe has already been added previously. If you want to create the recipe from scratch you may just do this in another directory below <code class="language-plaintext highlighter-rouge">recipes/</code>.</p></blockquote>




<h4 class="faq-area"> In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-in-bowtie-2-parameters-in-place-of-1000-for-other-experiments-should-we-mention-the-median-fragment-length-observed-in-our-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library?</h3>  <p>Not the median fragment length but the maximum fragment length you expect. However, you will see that in illumina sequencers, the longer the fragments are the less efficiently they are sequenced so long fragment length pairs are not very numerous.</p></blockquote>




<h4 class="faq-area"> The Build tissue-specific <span class="notranslate">expression</span> dataset tool (step one) exits with an error code. <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-the-build-tissue-specific-expression-dataset-tool-step-one-exits-with-an-error-code"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: The Build tissue-specific <span class="notranslate">expression</span> dataset tool (step one) exits with an error code.</h3>  <p>For the HPS source files version select HPA normal tissue 23/10/2018 rather than the version from 01/04/2020.</p></blockquote>




<h4 class="faq-area"> Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-do-we-change-the-chromosome-names-in-the-ensembl-gtf-to-match-the-ucsc-genome-reference"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference?</h3>  <p>UCSC chromosome names begin with the prefix <code class="language-plaintext highlighter-rouge">chr</code>, but Ensembl chromosome names do not. For example, chromosome 19 would be denoted as <code class="language-plaintext highlighter-rouge">chr19</code> in UCSC, and as <code class="language-plaintext highlighter-rouge">19</code> in Ensemble. Most tools would view those as different when looking for matches/overlaps. Therefore it is always a good idea to make sure these match before you perform any <span class="notranslate">downstream</span> analysis.</p></blockquote>




<h4 class="faq-area"> Do you have resources to help me get started working in the cloud? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-do-you-have-resources-to-help-me-get-started-working-in-the-cloud"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Do you have resources to help me get started working in the cloud?</h3>  <p>Yes, we have a number of documents and videos to help you start working with SRA data in the cloud:</p>  <ul>  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud/">Cloud landing page</a></li>  <li><a href="https://www.youtube.com/watch?v=DkNz-RCCm-M">Exploring SRA metadata in BigQuery</a></li>  <li><a href="https://www.youtube.com/watch?v=_F4FhcDWSJg">Exploring SARS-CoV-2 metadata in Athena</a></li></ul></blockquote>




<h4 class="faq-area"> I want to use a collection for outputs but it always passes the test even when the script fails. Why? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/collections_test.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-i-want-to-use-a-collection-for-outputs-but-it-always-passes-the-test-even-when-the-script-fails-why"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> I want to use a collection for outputs but it always passes the test even when the script fails. Why?</h3>  <ul>  <li>Collections are tricky for generating tests.    <ul>      <li>The contents appear only after the tool has been run and even then may vary with settings.</li>    </ul>  </li>  <li>A manual test override is currently the only way to test collections properly.</li>  <li>Automation is hard. If you can help, pull requests are welcomed.</li>  <li>Until it‚Äôs automated, please take a look at the plotter sample.</li>  <li>It is recommended that you modify the test over-ride that appears in that sample form. Substitute one or more of the file names you expect to see after the collection is filled by your new tool for the <code class="language-plaintext highlighter-rouge">&lt;element.../&gt;</code> used in the plotter sample‚Äôs tool test.</li></ul></blockquote>




<h4 class="faq-area"> How does one compare metaproteomics measurements from two experimental conditions? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-one-compare-metaproteomics-measurements-from-two-experimental-conditions"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does one compare metaproteomics measurements from two experimental conditions?</h3>  <p>For comparing taxonomy composition or functional content of two conditions in metaproteomics or metatranscriptomics studies, users are recommended to use metaQuantome. GTN tutorials for metaQuantome are available <a href="/training-material/topics/proteomics/">in the proteomics topic.</a></p></blockquote>




<h4 class="faq-area"> How does one convert RAW files to MGF peak lists within <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-one-convert-raw-files-to-mgf-peak-lists-within-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does one convert RAW files to MGF peak lists within <span class="notranslate">Galaxy</span>?</h3>  <p><span class="notranslate">Galaxy</span> has implemented msconvert tool so that RAW files from Thermo instruments can be converted into MGF or mzML formats.</p></blockquote>




<h4 class="faq-area"> I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-have-fastq-files-from-metagenomics-or-metatranscriptomics-datasets-how-can-i-convert-them-into-a-protein-fasta-file-for-metaproteomics-searches"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches?</h3>  <p><span class="notranslate">Galaxy</span> has a tool named Sixgill that can be used to convert the nucleic acid sequences to ‚Äòmetapeptide‚Äô sequences. There are other options available within <span class="notranslate">Galaxy</span> such as the <span class="notranslate">Galaxy</span>Graph approach and Metagenome Binning, Assembly and Annotation <span class="notranslate">Workflow</span>. Please <a href="https://help.galaxyproject.org/">contact us</a>, if you need assistance.</p></blockquote>




<h4 class="faq-area"> What software tools are available to determine taxonomic composition from mass spectrometry data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-software-tools-are-available-to-determine-taxonomic-composition-from-mass-spectrometry-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What software tools are available to determine taxonomic composition from mass spectrometry data?</h3>  <p>Within the <span class="notranslate">Galaxy</span> framework we recommend the use of Unipept software that uses NCBI taxonomy and UniProt databases to detect unique peptides for taxonomy. Other software tools such as MetaTryp 2.0 (PMID: 32897080) can also be used to determine the taxonomic composition of the metaproteomics datasets.</p></blockquote>




<h4 class="faq-area"> Could I use a different p-adj value for filtering differentially expressed genes? <a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-could-i-use-a-different-p-adj-value-for-filtering-differentially-expressed-genes"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Could I use a different p-adj value for filtering differentially expressed genes?</h3>  <p>Yes, you can modify this value, to perform a more rigorous analysis, or extend the range of genes selected. A higher p-value will significantly increase the number of genes selected, at the expense of including possible false positives.</p></blockquote>




<h4 class="faq-area"> What is the principle of an enrichment analysis? <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-is-the-principle-of-an-enrichment-analysis"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What is the principle of an enrichment analysis?</h3>  <p>Enrichment analysis approach (also called over-representation analysis (ORA)) was introduced to test whether pre-specified sets of proteins (e.g. those acting together in a given biological process), change in abundance more systematically than as expected by chance. This type of analysis investigates hypotheses that are more directly relevant to the biological function, and can also help highlight a process over-represented within a subset of proteins.</p>  <p>Further reading: <a href="https://pubmed.ncbi.nlm.nih.gov/19033363/">‚ÄúHuang DW, Sherman BT, and Lempicki RA (2009) Bioinformatics enrichment tools: Paths toward the comprehensive functional analysis of large gene lists. Nucleic Acids Res 37:1‚Äì13‚Äù</a></p></blockquote>




<h4 class="faq-area"> My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called 'args' ? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-my-rscript-tool-generates-a-strange-r-error-on-stdout-about-an-invalid-operation-on-a-closure-called-args-"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called ‚Äòargs‚Äô ?</h3>  <p>Did your code declare the <code class="language-plaintext highlighter-rouge">args vector</code> with something like <code class="language-plaintext highlighter-rouge">args = commandArgs(trailingOnly=TRUE)</code> before it tried to access args[1] ?See the plotter tool for a sample</p></blockquote>




<h4 class="faq-area"> `docker-compose up` fails with error `/usr/bin/start.sh: line 133: /<span class="notranslate">galaxy</span>/.venv/bin/uwsgi: No such file or directory` <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/error_compose.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-docker-compose-up-fails-with-error-usrbinstartsh-line-133-galaxyvenvbinuwsgi-no-such-file-or-directory"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> <code class="language-plaintext highlighter-rouge">docker-compose up</code> fails with error <code class="language-plaintext highlighter-rouge">/usr/bin/start.sh: line 133: /<span class="notranslate">galaxy</span>/.venv/bin/uwsgi: No such file or directory</code></h3>  <ul>  <li>This is why it‚Äôs useful to watch the boot process without detaching</li>  <li>This can happen if a container has become corrupt on disk after being interrupted    <ul>      <li>cured by a complete cleanup.</li>    </ul>  </li>  <li>Make sure no docker <span class="notranslate">galaxy</span>-server related processes are running - use docker ps to check and stop them manually</li>  <li>delete the <code class="language-plaintext highlighter-rouge">..compose/export directory</code> with <code class="language-plaintext highlighter-rouge">sudo rm -rf export/*</code> to clean out any corrupted files</li>  <li>run <code class="language-plaintext highlighter-rouge">docker system prune</code> to clear out any old corrupted containers, images or networks. Then run <code class="language-plaintext highlighter-rouge">docker volume prune</code> in the same way to remove the shared volumes.</li>  <li>run <code class="language-plaintext highlighter-rouge">docker-compose pull</code> again to ensure the images are correct</li>  <li>run <code class="language-plaintext highlighter-rouge">docker-compose up</code> to completely rebuild the appliance from scratch. <strong>Please be patient</strong>.</li></ul></blockquote>




<h4 class="faq-area"> Where can I find example queries for use in the cloud and elsewhere? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-find-example-queries-for-use-in-the-cloud-and-elsewhere"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I find example queries for use in the cloud and elsewhere?</h3>  <p>We have examples on our website for Athena (<a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-athena-examples/">link</a>) and BigQuery (<a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-bigquery-examples/">link</a>) which can be easily adapted to other environments.</p></blockquote>




<h4 class="faq-area"> Why do we use FASTQ interlacer and not the FASTQ joiner? <a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-do-we-use-fastq-interlacer-and-not-the-fastq-joiner"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why do we use FASTQ interlacer and not the FASTQ joiner?</h3>  <p>The reason ASaiM-MT uses FASTQ-interlacer than FASTQ-joiner for combining forward and reverse <span class="notranslate">reads</span> is because the joiner tool combines the forward and reverse read sequence together while the interlacer puts the forward and reverse read sequences in the same file while retaining the entity of each read along with an additional file with unpaired sequences and it maintains the integrity of the <span class="notranslate">reads</span> while helping us distinguish between the forward and reverse <span class="notranslate">reads</span>.</p></blockquote>




<h4 class="faq-area"> My Scanpy FindMarkers step is giving me an empty table <a href="/training-material/topics/transcriptomics/tutorials/scrna-seq-basic-pipeline/faqs/findmarkers_empty.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-my-scanpy-findmarkers-step-is-giving-me-an-empty-table"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> My Scanpy FindMarkers step is giving me an empty table</h3>  <p>Try selecting: ‚ÄúUse programme defaults: Yes‚Äù and see if that fixes it.</p></blockquote>




<h4 class="faq-area"> What file/data formats are defined for I/O in <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/formats.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-filedata-formats-are-defined-for-io-in-galaxy"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What file/data formats are defined for I/O in <span class="notranslate">Galaxy</span>?</h3>  <ul>  <li><a href="https://galaxyproject.org/learn/datatypes"><span class="notranslate">Galaxy</span> Datatypes</a></li>  <li><code class="language-plaintext highlighter-rouge">[<span class="notranslate">galaxy</span>-root]/config/datatypes_conf.xml</code> is read at startup so new datatypes can be defined.</li></ul></blockquote>




<h4 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</h3>  <p>Within the <span class="notranslate">Galaxy</span> framework we recommend the use of Unipept software that uses UniProt databases and annotation to detect proteins (EC terms) and functional groups such as GO Ontology and InterPro terms. Other software tools such as <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">EggNOG Mapper</a> are also available within the <span class="notranslate">Galaxy</span> platform. Other software such as MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA), ProPHAnE also generate functional outputs.</p></blockquote>




<h4 class="faq-area"> Where can I read more about Quality Control of data? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-read-more-about-quality-control-of-data"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I read more about Quality Control of data?</h3>  <p>I really like <a href="https://sequencing.qcfail.com/">QCFAIL</a>, It has some nice user stories of quality control issues encountered in real data and experiments</p></blockquote>




<h4 class="faq-area"> What is Gene Ontology (GO)? <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-is-gene-ontology-go"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What is Gene Ontology (GO)?</h3>  <p>A very commonly used way of specifying these sets is to gather genes/proteins that sharethe same Gene Ontology (GO) term, as specified by the Gene Ontology Consortium.</p>  <p>The GO project provides an ontology that describes gene products and their relationsin three non-overlapping domains of molecular biology, namely ‚ÄúMolecular Function‚Äù,‚ÄúBiological Process‚Äù, and ‚ÄúCellular Component‚Äù. Genes/proteins are annotated by oneor several GO terms, each composed of a label, a definition and a unique identifier.GO terms are organized within a classification scheme that supports relationships, andformalized by a hierarchical structure that forms a directed acyclic graph (DAG). Insuch a graph is used the notions of child and parent, where a child inherits from oneor multiple parents, child class having a more specific annotation than parent class(e.g. ‚Äúglucose metabolic process‚Äù inherits from ‚Äúhexose metabolic‚Äù parent term whichitself inherits from ‚Äúmonosaccharide metabolic process‚Äù etc.). In this graph, each node corresponds to a GO term composed of genes/proteins sharing the same annotation, whiledirected edges between nodes represents their relation (e.g. ‚Äòis a‚Äô, ‚Äòpart of‚Äô) andtheir roles in the hierarchy (i.e. parent and child).</p>  <p><a href="http://www.geneontology.org/page/introduction-go-resource">Further reading</a></p></blockquote>




<h4 class="faq-area"> Where can I get planemo? <a href="/training-material/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-get-planemo"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I get planemo?</h3>  <p>Plese see <a href="https://training.galaxyproject.org/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html#installing-planemo">the installation section</a>. Essentially you can <code class="language-plaintext highlighter-rouge">pip install planemo</code>. If you don‚Äôt have pip, you need to install this first.</p>  <p>On windows you‚Äôll need <code class="language-plaintext highlighter-rouge">WSL2</code> and then you can <code class="language-plaintext highlighter-rouge">apt-get install python3-pip</code>, same for ubuntu. For OSX users it is probably present.</p></blockquote>




<h4 class="faq-area"> Changing the heatmap colours <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-heatmap-colours"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the heatmap colours</h3>  <p>You can change the heatmap color, by expanding the <strong>Show advanced options</strong> section. There are many options here, including setting the colors.</p></blockquote>




<h4 class="faq-area"> Most tools seem to have options for assembly using long and short <span class="notranslate">reads</span>, what are the pros and cons of the different tools? <a href="/training-material/topics/assembly/faqs/hybrid_assembly.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-most-tools-seem-to-have-options-for-assembly-using-long-and-short-reads-what-are-the-pros-and-cons-of-the-different-tools"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Most tools seem to have options for assembly using long and short <span class="notranslate">reads</span>, what are the pros and cons of the different tools?</h3>  <p>In our experience, when both long and short <span class="notranslate">reads</span> are allowed as input, the difference comes down to the order in which set is assembled first. For example, Unicycler assembles the short <span class="notranslate">reads</span> first (which can be good, because they are more accurate), and then scaffolds these into larger contigs using long <span class="notranslate">reads</span>. Other tools (or workflows) often assemble long <span class="notranslate">reads</span> first (which can also be good because these can span repeat regions), then correct this assembly with information from the more accurate short <span class="notranslate">reads</span>. There may also be other variations on long/short read assembly, and/or iterations of these types of steps (assemble, correct). My preference is to assemble long <span class="notranslate">reads</span> first, but that‚Äôs because I‚Äôm really interested in covering repeat regions. If accuracy was the aim, rather than contig length, the short-<span class="notranslate">reads</span>-first approach may be better. For even more complexity ‚Ä¶ I think some tools now allow input of ‚Äútrusted contigs‚Äù - i.e. contigs assembled from other tools. Ryan Wick has a new tool called <a href="https://github.com/rrwick/Trycycler/wiki">Trycyler</a> that can take in multiple assemblies to make a consensus (bacterial genomes).</p></blockquote>




<h4 class="faq-area"> Which icons are available to use in my tutorial? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-icons-are-available-to-use-in-my-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which icons are available to use in my tutorial?</h3>  <p>To use icons in your tutorial, take the name of the icon, ‚Äòdetails‚Äô in this example, and write something like this in your tutorial:</p>  <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{% icon details %}<br></code></pre></div></div>  <p>The following icons are currently available:</p>  <div class="row">	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bullhorn" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>announcement</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>code-in</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-laptop-code" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>code-out</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>cofest</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>comment</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>congratulations</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>curriculum</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>details</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>docker_image</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-envelope" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>email</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>exchange</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>event</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>feedback</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-barchart</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-bug</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-chart-select-data</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-clear</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-columns</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-cross</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-dropdown</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-eye</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-gear</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-home" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-home</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-info</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-library</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-link</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-panelview</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-pencil</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-refresh</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-rulebuilder-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-save</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-scratchbook</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-selector</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-star</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-tags</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-toggle</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-upload</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-arrows-alt-h" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-connection</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-edit</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-new</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-report-download</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-external-link-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>_instance</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>github</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>gitter</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>gtn-theme</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>hall-of-fame</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>hands_on</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>help</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-comment" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>history-annotate</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>history-share</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>instances</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>interactive_tour</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>keypoints</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>language</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>last_modification</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>level</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>license</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-linkedin" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>linkedin</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>new-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-file-code" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>notebook</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>objectives</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-orcid" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>orcid</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-check</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-collection</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-file</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-files</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-repeat</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-select</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-text</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-toggle</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>question</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>references</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>requirements</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-rss" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>rss-feed</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>search</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>slides</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>solution</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-sticky-note" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>sticky-note</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>time</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-file-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>text-document</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tip</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tool</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tool-versions</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>topic</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>trophy</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tutorial</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-twitter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>twitter</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>warning</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-chevron-circle-right" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>wf-input</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span>-runtime-toggle</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span></div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span>-run</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>video</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-play-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>video-slides</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>zenodo_link</div>	</div></div></blockquote>




<h4 class="faq-area"> The input for a tool is not listed in the dropdown <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/input_not_found.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-the-input-for-a-tool-is-not-listed-in-the-dropdown"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: The input for a tool is not listed in the dropdown</h3>  <p>This tutorial uses collections, some tools will require collections as input (e.g. Taxonomy-to-Krona). To select a collection as in put to a file, click on the <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> button in front of the input parameter you want to supply the collection to.</p></blockquote>




<h4 class="faq-area"> The input for a tool is not listed in the dropdown <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-the-input-for-a-tool-is-not-listed-in-the-dropdown"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: The input for a tool is not listed in the dropdown</h3>  <p>This tutorial uses collections, some tools will require collections as input (e.g. Taxonomy-to-Krona). To select a collection as in put to a file, click on the <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> button in front of the input parameter you want to supply the collection to.</p></blockquote>




<h4 class="faq-area"> JBrowse is taking a long time to complete? <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-jbrowse-is-taking-a-long-time-to-complete"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> JBrowse is taking a long time to complete?</h3>  <p>Normally this should be done in around 3 minutes. However, it might be busy on the servers, so please be patient and come back to it later.</p></blockquote>




<h4 class="faq-area"> First job I submitted remains grey or running for a long time - is it broken? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/job_grey.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-first-job-i-submitted-remains-grey-or-running-for-a-long-time---is-it-broken"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> First job I submitted remains grey or running for a long time - is it broken?</h3>  <ul>  <li>Check with <code class="language-plaintext highlighter-rouge">top</code> or your system monitor - if Conda is running, things are working but it‚Äôs slow the first time a dependency is installed.</li>  <li>The first run generally takes a while to install all the needed dependencies.</li>  <li>Subsequent runs should start immediately with all dependencies already in place.</li>  <li>Installing new Conda dependencies just takes time so tools that have new Conda packages will take longer to run the first time if they must be installed.</li>  <li>In general, a <code class="language-plaintext highlighter-rouge">planemo_test</code> job usually takes around a minute - planemo has to build and tear down a new <span class="notranslate">Galaxy</span> for generating test results and then again for testing properly. Longer if the tool has Conda dependencies.</li>  <li>The very first test in a fresh appliance may take 6 minutes so be patient.</li></ul></blockquote>




<h4 class="faq-area"> TB Variant Report crashes (with an error about KeyError: 'protein') <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-tb-variant-report-crashes-with-an-error-about-keyerror-protein"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> TB Variant Report crashes (with an error about KeyError: ‚Äòprotein‚Äô)</h3>  <p>This is a bug present in TB Variant Report (aka tbvcfreport) version 0.1.8 and earlier. In this case it is triggered by the presence of variants in Rv3798. You only see this bug, however, if you forget to run <code class="language-plaintext highlighter-rouge">tb_variant_filter</code> (TB Variant Filter). Rv3798 is a suspected transposase and any variants in this gene region would be filtered out by <code class="language-plaintext highlighter-rouge">tb_variant_filter</code>, so if you see this crash, make sure you have run the filter step before the TB Variant Report step.</p></blockquote>




<h4 class="faq-area"> Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local <span class="notranslate">galaxy</span> server from the <span class="notranslate">toolshed</span>? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/local_install.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-do-i-need-that-big-5gb-complicated-docker-thing---can-i-just-install-the-toolfactory-into-our-local-galaxy-server-from-the-toolshed"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local <span class="notranslate">galaxy</span> server from the <span class="notranslate">toolshed</span>?</h3>  <p>You can but it can‚Äôt really be very useful. The ToolFactory is a <span class="notranslate">Galaxy</span> tool, but it installs newly generated tools automatically into the local <span class="notranslate">Galaxy</span> server. This is not normally possible because a tool cannot escape <span class="notranslate">Galaxy</span>‚Äôs job execution environment isolation. The ToolFactory needs to write to the normally forbidden server‚Äôs configuration so the new tool appears in the tool menu and is installed in the TFtools directory which is a subdirectory of the <span class="notranslate">Galaxy</span> tools directory.The Appliance is configured so the ToolFactory and the Planemo test tool use remote procedure calls (RPC using rpyc) to do what tools cannot normally do. The rpyc server runs in a separate container. Without it, tool installation and testing are difficult to do inside <span class="notranslate">Galaxy</span> tools.Known good tools can be uploaded to a local <span class="notranslate">toolshed</span> from your private appliance for installation to that server of yours. Debugging tools on a production server is not secure SOP. You just never know what might break. <em>That‚Äôs why a desktop disposable appliance is a better choice.</em></p></blockquote>




<h4 class="faq-area"> Where can I find the full listing and description of the columns in each metadata table? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-find-the-full-listing-and-description-of-the-columns-in-each-metadata-table"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I find the full listing and description of the columns in each metadata table?</h3>  <p>Table definitions are available here:</p>  <ul>  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud-based-examples/">SRA Cloud-based Examples</a></li>  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/aligned-metadata-tables/">Aligned metadata Tables</a></li></ul></blockquote>




<h4 class="faq-area"> Can we use the ASaiM-MT <span class="notranslate">workflow</span> on multiple input files at the same time? <a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/multiple_inputs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-we-use-the-asaim-mt-workflow-on-multiple-input-files-at-the-same-time"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can we use the ASaiM-MT <span class="notranslate">workflow</span> on multiple input files at the same time?</h3>  <p>Currently, that is one of its limitations. However, <span class="notranslate">Galaxy</span> offers a <span class="notranslate">workflow</span> within <span class="notranslate">workflow</span> feature which can help process multiple files at the same time and this output can be combined into one using the <code class="language-plaintext highlighter-rouge">MT2MQ</code> tool.</p></blockquote>




<h4 class="faq-area"> MultiQC error for your FastQC reports? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-multiqc-error-for-your-fastqc-reports"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: MultiQC error for your FastQC reports?</h3>  <p>Please double-check that:</p>  <ol>  <li>You selected <code class="language-plaintext highlighter-rouge">FastQC</code> tool as the source of the log files in MultiQC.</li>  <li>And you provided the <strong>Raw Data</strong> of FastQC and not the HTML reports.</li></ol></blockquote>




<h4 class="faq-area"> In the MVP platform, is it possible to view the genomic location of all the peptides? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-in-the-mvp-platform-is-it-possible-to-view-the-genomic-location-of-all-the-peptides"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> In the MVP platform, is it possible to view the genomic location of all the peptides?</h3>  <p>Not really, you can only view the genomic localization of the peptides that were present in the genomic <span class="notranslate">mapping</span> file (output from the first <span class="notranslate">workflow</span>).</p></blockquote>




<h4 class="faq-area"> I cannot run client tests because yarn is not installed. <a href="/training-material/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-i-cannot-run-client-tests-because-yarn-is-not-installed"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> I cannot run client tests because yarn is not installed.</h3>  <p>Make sure you have executed <code class="language-plaintext highlighter-rouge">scripts/common_startup.sh</code> and have activated the virtual environment (<code class="language-plaintext highlighter-rouge">. .venv/vin/activate</code>) in your <em>current terminal session</em>.</p></blockquote>




<h4 class="faq-area"> When will aligned read objects be available for other data types? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-when-will-aligned-read-objects-be-available-for-other-data-types"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> When will aligned read objects be available for other data types?</h3>  <p>We hope to have these constructed for long read SARS-CoV-2 data in the near future. If there is strong community interest we may expand this offering to other organisms or data types such as metagenome submissions. If you would like this format for other datasets, write to the SRA helpdesk (<a href="mailto:sra@ncbi.nlm.nih">sra@ncbi.nlm.nih</a>) and let us know!</p></blockquote>




<h4 class="faq-area"> When I get a warning for base per sequence content, what should I do? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-when-i-get-a-warning-for-base-per-sequence-content-what-should-i-do"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: When I get a warning for base per sequence content, what should I do?</h3>  <p>So far it does not mean that your data is bad. Your protocol or your data might have a bias that you normally expect. Check first the following things:</p>  <ul>  <li>Adapter content (maybe some adapters are still in your data)</li>  <li>Kmer content/Over represented sequences (this would indicate a contamination or a protocol/sequence bias)</li>  <li>Per base quality plot. If the overall quality is not good, then probably the sequencing was poorly performed.</li>  <li>Read about your protocol, e.g., ChIP-Seq and ATAC-Seq typically have a nucleotide bias. For example this article about <a href="https://mobilednajournal.biomedcentral.com/articles/10.1186/1759-8753-3-3">ATAC-Seq</a>.</li></ul></blockquote>




<h4 class="faq-area"> Running more than one round of Pilon polishing <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-more-than-one-round-of-pilon-polishing"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running more than one round of Pilon polishing</h3>  <p>Include the most recent polished assembly as input to the next round. You will also need to make a new bam file (here, we have <code class="language-plaintext highlighter-rouge">round1.bam</code> and <code class="language-plaintext highlighter-rouge">round2.bam</code>).</p>  <p>Round 1</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>assembly.fasta + illumina <span class="notranslate">reads</span> =&gt; BWA MEM =&gt; round1.bam<br></code></pre></div></div>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>round1.bam + assembly.fasta =&gt; pilon =&gt; polished.fasta<br></code></pre></div></div>  <p>Round 2</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>polished.fasta + illumina <span class="notranslate">reads</span> =&gt; BWA MEM =&gt; round2.bam<br></code></pre></div></div>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>round2.bam + polished.fasta =&gt; pilon =&gt; polished2.fasta<br></code></pre></div></div>  <p><strong>How to know when enough polishing iterations have run?</strong></p>  <p>There is no single answer, but a common way is to see when pilon stops making many polishing changes between rounds. So if round1 made 100 changes, and round2 made only 3, this seems like there would not be much more polishing to do.</p>  <p><strong>How can I see how many changes Pilon has made?</strong></p>  <p>There are two ways that I know of to see how many changes that Pilon made:</p>  <p>The first is to look at the tool <em>standard output</em> (<code class="language-plaintext highlighter-rouge">stdout</code>) from Pilon (<a href="/training-material/faqs/galaxy/tools_logs.html">instructions</a>).</p>  <p>Somewhere near the top of this log file will be a line that says how many corrections (changes) were made.</p>  <p>The second way is to count the number of lines in the changes file. To do this, use the tool called <strong>Line/Word/Character count</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, and select the line count option.</p></blockquote>




<h4 class="faq-area"> Only one Planemo test runs at a time. Why doesn't the server allow more than one at once? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-only-one-planemo-test-runs-at-a-time-why-doesnt-the-server-allow-more-than-one-at-once"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Only one Planemo test runs at a time. Why doesn‚Äôt the server allow more than one at once?</h3>  <ul>  <li>When a new dependency is being installed in the Planemo Conda repository, there is no locking to prevent a second process from overwriting or otherwise interfering with it‚Äôs own independent repository update.</li>  <li>The result is not pretty.</li>  <li>Allowing two tests to run at once has proven to be unstable so the Appliance is currently limited to one.</li></ul></blockquote>




<h4 class="faq-area"> On Scanpy PlotEmbed, the tool is failing <a href="/training-material/topics/transcriptomics/tutorials/scrna-seq-basic-pipeline/faqs/plotembed_fails.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-on-scanpy-plotembed-the-tool-is-failing"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> On Scanpy PlotEmbed, the tool is failing</h3>  <p>Try selecting ‚ÄúUse raw attributes if present: NO‚Äù</p></blockquote>




<h4 class="faq-area"> On the Scanpy PlotEmbed step, my object doesn‚Äôt have Il2ra or Cd8b1 or Cd8a etc. <a href="/training-material/topics/transcriptomics/tutorials/scrna-seq-basic-pipeline/faqs/plotembed_results_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-on-the-scanpy-plotembed-step-my-object-doesnt-have-il2ra-or-cd8b1-or-cd8a-etc"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> On the Scanpy PlotEmbed step, my object doesn‚Äôt have Il2ra or Cd8b1 or Cd8a etc.</h3>  <p>Check your <code class="language-plaintext highlighter-rouge">Anndata</code> object - it should be <code class="language-plaintext highlighter-rouge">7874 x 14832</code>, i.e. 7874 cells x 14832 genes. Is it actually 2000 genes only (i.e. and therefore missing the above markers)? You may have selected to remove genes at the Scanpy FindVariableGenes step (last toggle, ‚ÄòRemove genes not marked as highly variable‚Äô &lt; Select NO.) (Most likely you did this correctly the first time, but later in investigating how many got marked as highly variable, may have run this tool again and removed the nonvariable ones. We‚Äôve updated the text to more clearly prevent this, but you may have gotten caught out!)</p></blockquote>




<h4 class="faq-area"> Can we polish the assembly with long <span class="notranslate">reads</span> too? <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-can-we-polish-the-assembly-with-long-reads-too"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Can we polish the assembly with long <span class="notranslate">reads</span> too?</h3>  <p>Yes. In this tutorial, we only polish the assembly with the short <span class="notranslate">reads</span>. This may be enough for bacterial genomes. However, for an even better polish (usually), a common approach is to also polish the assembly with the long <span class="notranslate">reads</span>. A typical <span class="notranslate">workflow</span> for this would assemble with long <span class="notranslate">reads</span>, then polish with long <span class="notranslate">reads</span> (x 4 rounds, with Racon), polish with long <span class="notranslate">reads</span> again (x 1 round, with Medaka), then polish with short <span class="notranslate">reads</span> (x2 rounds with Pilon).</p></blockquote>




<h4 class="faq-area"> How do I know what protocol my data was sequenced with? <a href="/training-material/topics/transcriptomics/tutorials/scrna-umis/faqs/protocol.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-how-do-i-know-what-protocol-my-data-was-sequenced-with"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> How do I know what protocol my data was sequenced with?</h3>  <p>If you have 10x data, then you just need to count the length of the R1 <span class="notranslate">reads</span> to guess the Chromium version (see <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html">this tutorial</a>). For other types of data, you must know the protocol in advance, and even then you must also know the multiplexing strategy and the list of expected (whitelisted) barcodes. The whitelist may vary from sequencing lab to sequencing lab, so always ask the wetlab people how the FASTQ data was generated.</p></blockquote>




<h4 class="faq-area"> Why don't the aligned read files have quality scores? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-dont-the-aligned-read-files-have-quality-scores"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why don‚Äôt the aligned read files have quality scores?</h3>  <p>Quality scores take up the majority of space in our compressed sequence files, so removing them makes the files much smaller (~80% or more). In addition, many uses don‚Äôt require per-base quality scores to successfully complete their work (some pipelines even require fastq format but don‚Äôt actually use the quality scores), so these files represent a faster route to completing many analyses. The full quality scores are still available in the original SRA Runs for anyone that requires them, using the SRA Tools available in <span class="notranslate">Galaxy</span>.</p></blockquote>




<h4 class="faq-area"> Can I use alte<span class="notranslate">rna</span>tive tools for the Quantification step? <a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-i-use-alternative-tools-for-the-quantification-step"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can I use alte<span class="notranslate">rna</span>tive tools for the Quantification step?</h3>  <p>There are some alte<span class="notranslate">rna</span>tives to Salmon for reference transcriptome-based <span class="notranslate">RNA</span> quantification. <strong>Kallisto</strong> and <strong>Sailfish</strong> use a similar approach, known as pseudoalignment.</p></blockquote>




<h4 class="faq-area"> Why does the query `SRR11772204 OR SRR11597145 OR SRR11667145` in the Run Selector not return any results? <a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-does-the-query-srr11772204-or-srr11597145-or-srr11667145-in-the-run-selector-not-return-any-results"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why does the query <code class="language-plaintext highlighter-rouge">SRR11772204 OR SRR11597145 OR SRR11667145</code> in the Run Selector not return any results?</h3>  <p>The query for sars-cov-2 in SRA Entrez returns over 250K results, but only the first 20k are sent to the Run Selector.  Enter the above query in Entrez directly to find the three runs used for the tutorial and send them to the Run Selector to send to <span class="notranslate">Galaxy</span>.</p></blockquote>




<h4 class="faq-area"> Are Barcodes always on R1 and Sequence data on R2? <a href="/training-material/topics/transcriptomics/tutorials/scrna-umis/faqs/r1r2.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-are-barcodes-always-on-r1-and-sequence-data-on-r2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Are Barcodes always on R1 and Sequence data on R2?</h3>  <p>No, it really depends on the protocol. In some protocols this convention is swapped, in others the barcodes can be distributed across both <span class="notranslate">reads</span>.</p></blockquote>




<h4 class="faq-area"> Is the ToolFactory a complete replacement for manual tool building? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-is-the-toolfactory-a-complete-replacement-for-manual-tool-building"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Is the ToolFactory a complete replacement for manual tool building?</h3>  <ul>  <li>No, except where all the requirements for the package or script can be satisfied by the limited automated functions of the code generator, or where there is a script with all the complex logic that might otherwise go into XML</li>  <li>Many advanced XML features are not available such as output filters.</li>  <li>Adding DIY output filters, XML macros and some other advanced features is possible if anyone is sufficiently enthusiastic - some features in the <span class="notranslate">galaxy</span>xml package would be relatively straightforward to add.</li></ul></blockquote>




<h4 class="faq-area"> Additional resources to learn more about proteomic data analysis <a href="/training-material/topics/proteomics/faqs/resources.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-additional-resources-to-learn-more-about-proteomic-data-analysis"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Additional resources to learn more about proteomic data analysis</h3>  <p>To learn more about proteomic data analysis, we suggest you look at:</p>  <ul>  <li><a href="https://video.ucdavis.edu/playlist/details/0_4jkc4swu">UC Davis Proteomics Online Short Course </a></li>  <li><a href="https://www.youtube.com/c/MayInstituteNEU/videos">May Institute YouTube channel</a></li>  <li><a href="http://www.kilgourlab.com/teaching-resources/">Kilgour Lab Training Resources</a></li>  <li><a href="https://timms.uni-tuebingen.de/List/List?id=UT_201%5B45%5D_____00%5B12%5D_cpm_000_&amp;Headline=Computational%20Proteomics%20and%20Metabolomics">T√ºbingen University Computational Proteomics and Metabolomics Lessons</a></li></ul></blockquote>




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Can I use these workflows on datasets generated from our laboratory?</h3>  <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p></blockquote>




<h4 class="faq-area"> My snippy is running for a very long time. Is this normal? <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-my-snippy-is-running-for-a-very-long-time-is-this-normal"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> My snippy is running for a very long time. Is this normal?</h3>  <p>As this tutorial uses real world data some of the tools can run for quite a while. During a course we can expected longer run times as the <span class="notranslate">Galaxy</span> servers are heavily used. Typically expected runtimes are approximately:</p>  <table>  <thead>    <tr>      <th>Tool name</th>      <th>Runtime</th>    </tr>  </thead>  <tbody>    <tr>      <td>FastQC</td>      <td>2 minutes</td>    </tr>    <tr>      <td>MultiQC</td>      <td>5 minutes</td>    </tr>    <tr>      <td>Trimmomatic</td>      <td>5 minutes</td>    </tr>    <tr>      <td>kraken2</td>      <td>5 - 12 minutes</td>    </tr>    <tr>      <td>snippy</td>      <td>15 - 25 minutes</td>    </tr>    <tr>      <td>TB Variant Filter</td>      <td>2 minutes</td>    </tr>    <tr>      <td>TB-Profiler</td>      <td>5 minutes</td>    </tr>    <tr>      <td>Text transformation</td>      <td>Less than 1 minute</td>    </tr>    <tr>      <td>TB Variant Report</td>      <td>1 minute</td>    </tr>    <tr>      <td>JBrowse</td>      <td>5 minutes</td>    </tr>    <tr>      <td>Samtools stats (optional)</td>      <td>1 minute</td>    </tr>    <tr>      <td>BAM Coverage plotter (optional)</td>      <td>1 minute</td>    </tr>  </tbody></table></blockquote>




<h4 class="faq-area"> What should I do special if on use<span class="notranslate">galaxy</span>.be? <a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-should-i-do-special-if-on-usegalaxybe"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What should I do special if on use<span class="notranslate">galaxy</span>.be?</h3>  <p><strong>Note for anyone trying to follow the tutorial on use<span class="notranslate">galaxy</span>.be:</strong></p>  <p>In step 3 of the hands-on section of setting up the sars-cov-2 analysis bot, when suggested to run</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>planemo run vcf2lineage.ga vcf2lineage-job.yml --profile planemo-tutorial --history_name "vcf2lineage test"<br></code></pre></div></div>  <p>please use directly the <span class="notranslate">workflow</span> ID <code class="language-plaintext highlighter-rouge">814dd8d1c056bc54</code> instead of <code class="language-plaintext highlighter-rouge">vcf2lineage.ga</code>. This ID points to a public <span class="notranslate">workflow</span> that‚Äôs using the version of the pangolin tool installed on use<span class="notranslate">galaxy</span>.be`.</p></blockquote>




<h4 class="faq-area"> Which search algorithms are recommended for searching the metaproteomics data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-search-algorithms-are-recommended-for-searching-the-metaproteomics-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which search algorithms are recommended for searching the metaproteomics data?</h3>  <p>SearchGUI supports search using nine search algorithms (X! Tandem. MS-GF+. OMSSA, Comet, Tide, MyriMatch, MS_Amanda, DirecTag and Novor). For this tutorial, we have used the first two search algorithms in the list. In our hands, the first four search algorithms have given us the most optimal results.</p></blockquote>




<h4 class="faq-area"> How many search engines can you use in SearchGUI? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-how-many-search-engines-can-you-use-in-searchgui"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> How many search engines can you use in SearchGUI?</h3>  <p>SearchGUI has options to use upto 9 search algorithms. However, running all at the same time can be time consuming. According to our initial test, upto 4 search engines can give you good results.</p></blockquote>




<h4 class="faq-area"> I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-have-a-really-large-search-database-what-search-strategies-do-you-recommend-for-searching-my-mass-spectrometry-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset?</h3>  <p>Readers are encouraged to use the database sectioning approach described by <a href="https://pubmed.ncbi.nlm.nih.gov/32396365/">Praveen Kumar et al</a> and available within <span class="notranslate">Galaxy</span>. Readers are also encouraged to consider other approaches such as MetaNovo (not yet available in <span class="notranslate">Galaxy</span>). In absence of any database or taxonomic information about the microbiome dataset, other methods such as COMPIL 2.0  and De novo search methods can also be considered.</p></blockquote>




<h4 class="faq-area"> When I try to run a Selenium test, I get an error <a href="/training-material/topics/dev/tutorials/debugging/faqs/selenium_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-when-i-try-to-run-a-selenium-test-i-get-an-error"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> When I try to run a Selenium test, I get an error</h3>  <p>If you get the following error:</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>selenium.common.exceptions.SessionNotCreatedException <span class="o">(</span>...This version of ChromeDriver only supports Chrome version...<span class="o">)</span><br></code></pre></div></div>  <p>Make sure that (a) the version of your ChromeDriver is the same as the version of Chrome:</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>chromedriver <span class="nt">--version</span><br><span class="nv">$ </span>chrome <span class="nt">--version</span><br></code></pre></div></div>  <p>If they are not the same:</p>  <ul>  <li>download the appropriate version of <a href="https://chromedriver.chromium.org/downloads">ChromeDriver</a>.</li>  <li>unzip the file</li>  <li>move the chromedriver file into the appropriate location.    <ul>      <li>On Linux, that could be <code class="language-plaintext highlighter-rouge">/usr/bin</code>, <code class="language-plaintext highlighter-rouge">$HOME/.local/bin</code>, etc.</li>      <li>Use the <code class="language-plaintext highlighter-rouge">which</code> command to check the location: <code class="language-plaintext highlighter-rouge">$ which chromedriver</code></li>    </ul>  </li>  <li>Make sure the permissions are correct (755).</li></ul></blockquote>




<h4 class="faq-area"> Can this ASaiM <span class="notranslate">workflow</span> be used for single-end data? <a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/single_end.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-this-asaim-workflow-be-used-for-single-end-data"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can this ASaiM <span class="notranslate">workflow</span> be used for single-end data?</h3>  <p>Yes, the inputs have to be changed to a single-end file rather than a paired-end.</p></blockquote>




<h4 class="faq-area"> What is a SNP? <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/snp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-is-a-snp"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What is a SNP?</h3>  <p>SNP (pronounced ‚Äúsnip‚Äù) stands for Single Nucleotide Polymorphism. This means a single nucleotide change as compared to the reference genome.</p></blockquote>




<h4 class="faq-area"> What other methods are available to study the functional state of the microbiome within <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-other-methods-are-available-to-study-the-functional-state-of-the-microbiome-within-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What other methods are available to study the functional state of the microbiome within <span class="notranslate">Galaxy</span>?</h3>  <p>Other software such as EggNOG Mapper, MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA) and  ProPHAnE also generate <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">functional outputs</a>.</p></blockquote>




<h4 class="faq-area"> Is it possible to subsample some samples if you have more <span class="notranslate">reads</span>? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-is-it-possible-to-subsample-some-samples-if-you-have-more-reads"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Is it possible to subsample some samples if you have more <span class="notranslate">reads</span>?</h3>  <p>Yes, we would recommend to process all <span class="notranslate">reads</span> and just before the peak calling. You can use <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Samtools view</strong> to sample the BAM file.</p></blockquote>




<h4 class="faq-area"> Are these data free to use and download? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-are-these-data-free-to-use-and-download"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Are these data free to use and download?</h3>  <p>Yes, the metadata, aligned <span class="notranslate">reads</span>, and other SARS-CoV-2 data that is mentioned in this training are free to download and have no associated egress charges.</p></blockquote>




<h4 class="faq-area"> Do I have to run the tools in the order of the tutorial? <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-do-i-have-to-run-the-tools-in-the-order-of-the-tutorial"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Do I have to run the tools in the order of the tutorial?</h3>  <p>The tools are presented in the order that a typical analysis would use. If you want to run some tools in parallel (to save time) you can do so. <a href="https://usegalaxy.eu/u/pvanheus/w/m-tuberculosis-variant-analysis-tutorial-imported-from-uploaded-file">This <span class="notranslate">workflow</span></a> illustrates the analysis done in the tutorial and shows that there are multiple ‚Äúpaths‚Äù leading to outputs that have some steps that could be run at the same time: MultiQC, Kraken2, JBrowse and TB Variant Report.</p></blockquote>




<h4 class="faq-area"> Which version of SearchGUI and PeptideShaker shall I use for this tutorial? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-version-of-searchgui-and-peptideshaker-shall-i-use-for-this-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which version of SearchGUI and PeptideShaker shall I use for this tutorial?</h3>  <p>We highly recommend the usage of SearchGUI <span class="notranslate">Galaxy</span> version 3.3.10.1 and PeptideShaker version <span class="notranslate">Galaxy</span> Version 1.16.36.3. The newer versions of SearchGUI and PeptideShaker have not yet been tested for this <span class="notranslate">workflow</span>.</p></blockquote>




<h4 class="faq-area"> UCSC import: what should my file look like? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-ucsc-import-what-should-my-file-look-like"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> UCSC import: what should my file look like?</h3>  <p>~2020 lines, with the following header line:</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bin    name    chrom   strand  txStart txEnd   cdsStart        cdsEnd  exonCount       exonStarts      exonEnds        score   name2   cdsStartStat    cdsEndStat      exonFrames<br></code></pre></div></div>  <p>Where:</p>  <ul>  <li><code class="language-plaintext highlighter-rouge">txStart</code>: Transcript start site</li>  <li><code class="language-plaintext highlighter-rouge">cdsStart</code>: CodingSequence start site</li></ul>  <p><strong>Note:</strong> UCSC is updated frequently, you might get a slightly different number of lines. If you only get one row in this file, make sure you requested the entire chr22, not just one position.</p></blockquote>




<h4 class="faq-area"> The UMAP Plots errors out sometimes? <a href="/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-the-umap-plots-errors-out-sometimes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: The UMAP Plots errors out sometimes?</h3>  <p>Try a different colour palette. For <span class="notranslate">upstream</span> code reasons, the default color palette sometimes causes the tool to error out.</p>  <ul>  <li>Under <strong>Plot attributes</strong>, do    <ul>      <li><em>‚ÄúColour map to use for continuous variables‚Äù</em>: <code class="language-plaintext highlighter-rouge">viridis</code></li>      <li><em>‚ÄúColors to use for plotting categorical annotation groups‚Äù</em>: <code class="language-plaintext highlighter-rouge">plasma</code></li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Downloading the files from the NCBI server fails or takes too long. <a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-the-files-from-the-ncbi-server-fails-or-takes-too-long"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading the files from the NCBI server fails or takes too long.</h3>  <p>Download the data from Zenodo instead (see overview box at top of tutorial). This method uses <span class="notranslate">Galaxy</span>‚Äôs generic data import functionality, and is more reliable and faster than the download from NCBI.</p></blockquote>




<h4 class="faq-area"> Why do we have a variant <span class="notranslate">mapping</span> file when it is not being used in the <span class="notranslate">workflow</span>? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-do-we-have-a-variant-mapping-file-when-it-is-not-being-used-in-the-workflow"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why do we have a variant <span class="notranslate">mapping</span> file when it is not being used in the <span class="notranslate">workflow</span>?</h3>  <p>We are working on updating the existing annotation tool to include the variant <span class="notranslate">mapping</span> file. Once that is done, the variant <span class="notranslate">mapping</span> file will also be an input for those tools.</p></blockquote>




<h4 class="faq-area"> Why don't we perform the V-Search dereplication step of ASaiM for metatrascriptomic data? <a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/vsearch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-why-dont-we-perform-the-v-search-dereplication-step-of-asaim-for-metatrascriptomic-data"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Why don‚Äôt we perform the V-Search dereplication step of ASaiM for metatrascriptomic data?</h3>  <p>In the metatranscriptomics data, duplicated <span class="notranslate">reads</span> are expected. And to keep the integrity of the sample, we would like to retain the reverse <span class="notranslate">reads</span>.</p></blockquote>




<h4 class="faq-area"> Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library? <a href="/training-material/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-encyclopedia-be-run-on-a-dia-ms-dataset-without-a-spectral-library"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library?</h3>  <p>Yes. In this GTN, the <span class="notranslate">workflow</span> presented is the Standard EncyclopeDIA <span class="notranslate">workflow</span>; however, there is a variation upon the Standard EncyclopeDIA <span class="notranslate">workflow</span>, named the WALNUT EncyclopeDIA <span class="notranslate">workflow</span> in which a spectral library is not required. Simply, the WALNUT variation of the <span class="notranslate">workflow</span> omits the DLIB spectral/PROSIT library input, hence requiring just the GPF DIA dataset collection, Experimental DIA dataset collection, and the FASTA Protein Database file. Therefore, the Chromatogram Library is generated using the GPF DIA dataset collection and the FASTA Protein Database alone. This method does generate fewer searches than if a spectral library is used. The <span class="notranslate">Galaxy</span>-P team tested the efficacy of the WALNUT <span class="notranslate">workflow</span> compared to the Standard EncyclopeDIA <span class="notranslate">workflow</span>, and more information on that comparison and those results can be found at this <a href="http://galaxyp.org/wp-content/uploads/2020/10/BCC_2020-Pratik.pdf">link</a>.</p></blockquote>



 <br> 

<h3 id="Account"> <a href="#Account"> Account </a></h3>

<hr/>



<h4 class="faq-area"> How to create an account on a public <span class="notranslate">Galaxy</span> instance? <a href="/training-material/faqs/galaxy/galaxy_creating_an_account.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-create-an-account-on-a-public-galaxy-instance"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to create an account on a public <span class="notranslate">Galaxy</span> instance?</h3>  <ol>  <li>To create an account at any public <span class="notranslate">Galaxy</span> instance, choose your server from the available <a href="https://galaxyproject.org/use/">list of <span class="notranslate">Galaxy</span> Platform</a>.  There are 3 main public <span class="notranslate">Galaxy</span> servers: <a href="https://usegalaxy.org/">Use<span class="notranslate">Galaxy</span>.org</a>, <a href="https://usegalaxy.eu/">Use<span class="notranslate">Galaxy</span>.eu</a>, and <a href="https://usegalaxy.org.au/">Use<span class="notranslate">Galaxy</span>.org.au</a>.</li>  <li>Click on <strong>‚ÄúLogin or Register‚Äù</strong> in the masthead on the server.</li>  <li>Click on <strong>Register here</strong> and fill in the required information.</li>  <li>Click on the <strong>Create</strong> button, your account is successfully created.</li>  <li>Check for a <strong>Confirmation Email</strong> in the email you used for account creation.</li>  <li>Click on the <strong>Email confirmation link</strong> to fully activate your account.</li></ol></blockquote>




<h4 class="faq-area"> How can I reduce quota usage while still retaining prior work (data, tools, methods)? <a href="/training-material/faqs/galaxy/reduce_quota_usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-reduce-quota-usage-while-still-retaining-prior-work-data-tools-methods"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I reduce quota usage while still retaining prior work (data, tools, methods)?</h3>  <ul>  <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_download_datasets.html"><strong>Download</strong></a> Datasets as individual files or entire Histories as an archive. Then <em>purge</em> them from the public server.</li>  <li><strong>Transfer/Move</strong> <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html">Datasets</a> or <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html">Histories</a> to another <span class="notranslate">Galaxy</span> server, including your own <span class="notranslate">Galaxy</span>. Then purge.</li>  <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/histories_copy_dataset.html"><strong>Copy</strong></a> your most important Datasets into a new/other History (inputs, results), then purge the original full History.</li>  <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_extract_from_history.html"><strong>Extract</strong></a> a <strong><span class="notranslate">Workflow</span></strong> from the History, then purge it.</li>  <li><strong>Back-up your work</strong>. It is a best practice to download an archive of your FULL original Histories periodically, even those still in use, as a backup.</li></ul>  <p><strong>Resources</strong> Much discussion about all of the above options can be found at the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help forum</a>.</p></blockquote>




<h4 class="faq-area"> How to update account preferences? <a href="/training-material/faqs/galaxy/updating_account_preferences.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-update-account-preferences"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to update account preferences?</h3>  <ol>  <li>    <p>Log into your existing account and update the email address under <strong>User &gt; Preferences</strong>. DO NOT create another new account or it will be detected as a duplicate account. Any use of multiple accounts can result in data loss.</p>  </li>  <li>    <p>If you instead want to start over with a new account, delete your old account(s) first under <strong>User &gt; Preferences</strong>, then create a new account.</p>  </li></ol>  <p>Note: Updating a registered account email address will not prevent multiple/duplicate account detection. If you have more than one, existing or new or older, review below about how to clear that up. Use just one account to avoid problems. Share data, not accounts.</p>  <ul>  <li>    <p>Manage your own data and delete old/extra accounts under <strong>User &gt; Preferences</strong>.</p>  </li>  <li>    <p>Multiple accounts that are detected, suspended, and administratively managed, without followup on your part, result in data loss. Creating another account is not the solution ‚Äì the new account will also be detected up as a duplicate and removed.</p>  </li>  <li>    <p>You will not receive a special notification when accounts are administratively managed due to a breach of the terms of service. The original terms of service included directly on the account registration form and in the account activation emails are your notice.</p>  </li></ul></blockquote>



 <br> 

<h3 id="DESeq2"> <a href="#DESeq2"> Deseq2 </a></h3>

<hr/>



<h4 class="faq-area"> The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-the-tutorial-uses-the-normalised-count-table-for-visualisation-what-about-using-vst-normalised-counts-or-rlog-normalised-counts"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts?</h3>  <p>this depends on what you would like to do with the table. The DESeq2 wrapper in <span class="notranslate">Galaxy</span> can output all of these, and there is a nice discussion in the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#count-data-transformations">DESeq2 vignette</a> about this topic.</p></blockquote>



 <br> 

<h3 id="De novo transcriptome reconstruction with RNA-Seq"> <a href="#De novo transcriptome reconstruction with RNA-Seq"> De novo transcriptome reconstruction with <span class="notranslate">rna</span>-seq </a></h3>

<hr/>



<h4 class="faq-area"> I‚Äôm using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why? <a href="/training-material/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-im-using-the-same-training-data-tools-and-parameters-as-the-tutorial-but-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I‚Äôm using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why?</h3>  <p>This is okay! Many aspects of the tutorial can potentially affect the exact results you obtain. For example, the reference genome version used and versions of tools. It‚Äôs less important to get the exact results shown in the tutorial, and more important to understand the concepts so you can apply them to your own data.</p></blockquote>



 <br> 

<h3 id="Single-Cell RNA"> <a href="#Single-Cell RNA"> <span class="notranslate">Single-cell</span> <span class="notranslate">rna</span> </a></h3>

<hr/>



<h4 class="faq-area"> Why is amplification more of an issue in sc<span class="notranslate">RNA</span>-seq than <span class="notranslate">RNA</span>-seq? <a href="/training-material/topics/transcriptomics/faqs/amplification.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why is amplification more of an issue in sc<span class="notranslate">RNA</span>-seq than <span class="notranslate">RNA</span>-seq?</h3>  <p>Due to the extremely small amount of starting material, the initial amplification is likely to be uneven due to the first cycle of amplified products being overrepresented in the second cycle of amplification leading to further bias. In Bulk <span class="notranslate">RNA</span>-seq, the larger selection of <span class="notranslate">RNA</span> molecules to amplify, evens out the odds that any one transcript will be amplified more than others.</p></blockquote>




<h4 class="faq-area"> Can <span class="notranslate">RNA</span>-seq techniques be applied to sc<span class="notranslate">RNA</span>-seq? <a href="/training-material/topics/transcriptomics/faqs/techniques.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-can-rna-seq-techniques-be-applied-to-scrna-seq"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Can <span class="notranslate">RNA</span>-seq techniques be applied to sc<span class="notranslate">RNA</span>-seq?</h3>  <p>The short answer is ‚Äòno, but yes‚Äô. At the beginning this was impossible due to the over-prevalence of dropout events (‚Äúzeroes‚Äù) in the data complicating the normalisation techniques, but this is not so much of a problem any more with newer methods.</p></blockquote>




<h4 class="faq-area"> Are UMIs not actually unique? <a href="/training-material/topics/transcriptomics/faqs/umi.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-are-umis-not-actually-unique"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Are UMIs not actually unique?</h3>  <p>Not strictly, but unique enough. The distribution of UMIs should ideally be uniform so that the chance of any two same UMIs capturing the same transcript (via different amplicons) is small. As barcodes have increased in size, the number of UMIs has also increased allowing for UMIs to reach more or less the same numbers of transcripts.</p></blockquote>



 <br> 

<h3 id="analysis"> <a href="#analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Including custom modifications into MaxQuant in <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-including-custom-modifications-into-maxquant-in-galaxy"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Including custom modifications into MaxQuant in <span class="notranslate">Galaxy</span>?</h3>  <p>Unfortunately the inclusion of custom modifications is not possible by the user because it requires profound changes in the underlying code. Please let us know the modification you need by creating a new issue: https://github.com/<span class="notranslate">galaxy</span>proteomics/tools-<span class="notranslate">galaxy</span>p/issues entitled MaxQuant new modification request.</p></blockquote>




<h4 class="faq-area"> Which isobaric labeled quantification methods does MaxQuant in <span class="notranslate">Galaxy</span> support? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-which-isobaric-labeled-quantification-methods-does-maxquant-in-galaxy-support"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Which isobaric labeled quantification methods does MaxQuant in <span class="notranslate">Galaxy</span> support?</h3>  <p>The current MaxQuant version supports: iTRAQ 4 and 8 plex; TMT 2,6,8,10,11 plex; iodoTMT6plex. Includion of TMT16 plex is in preparation.</p></blockquote>




<h4 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-how-can-i-adapt-this-tutorial-to-my-own-data"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> How can I adapt this tutorial to my own data?</h3>  <p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p></blockquote>




<h4 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-how-can-i-adapt-this-tutorial-to-my-own-data"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> How can I adapt this tutorial to my own data?</h3>  <p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p></blockquote>




<h4 class="faq-area"> Adding a custom database/build (dbkey) <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> may have several reference genomes built-in, but you can also create your own.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-custom-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a custom database/build (dbkey)</h3>  <ul>  <li>In the top menu bar, go to the <em>User</em>, and select <em>Custom Builds</em></li>  <li>Choose a name for your reference build</li>  <li>Choose a dbkey for your reference build</li>  <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>  <li>Under <strong>FASTA-file</strong>, select your fasta file</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Extended Help for Differential <span class="notranslate">Expression</span> Analysis Tools <a href="/training-material/faqs/galaxy/analysis_extended_Extended_help_differential_expression_analysis_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extended-help-for-differential-expression-analysis-tools"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extended Help for Differential <span class="notranslate">Expression</span> Analysis Tools</h3>  <p>The error and usage help in this FAQ applies to:</p>  <ul>  <li>Deseq2</li>  <li>Limma</li>  <li>edgeR</li>  <li>goseq</li>  <li>DEXSeq</li>  <li>Diffbind</li>  <li>StringTie</li>  <li>Featurecounts</li>  <li>HTSeq</li>  <li>Kalisto</li>  <li>Salmon</li>  <li>Sailfish</li>  <li>DexSeq-count</li></ul>  <p>Expect odd errors or content problems if any of the usage requirements below are not met:</p>  <ul>  <li>Differential <span class="notranslate">expression</span> tools all require count dataset replicates when used in <span class="notranslate">Galaxy</span>. At least two per factor level and the same number per factor level. These must all contain unique content.</li>  <li>Factor/Factor level names should only contain alphanumeric characters and optionally underscores. Avoid starting these with a number and do not include spaces.</li>  <li>If the tool uses <code class="language-plaintext highlighter-rouge">Conditions</code>, the same naming requirements apply. <code class="language-plaintext highlighter-rouge">DEXSeq</code> additionally requires that the first Condition is labeled as <code class="language-plaintext highlighter-rouge">Condition</code>.</li>  <li>Reference annotation should be in GTF format for most of these tools, with no header/comment lines. Remove all GTF header lines with the tool <strong>Remove beginning of a file</strong>. If any are comment lines are inte<span class="notranslate">rna</span>l to the file, those should be removed. The tool <strong>Select</strong> can be used.</li>  <li>Make sure that if a GTF dataset is used, and tool form settings are expecting particular attributes, those are actually in your annotation file (example: gene_id).</li>  <li>GFF3 data (when accepted by a tool) should have single <code class="language-plaintext highlighter-rouge">#</code> comment line and any others (at the start or inte<span class="notranslate">rna</span>l) that usually start with a <code class="language-plaintext highlighter-rouge">##</code> should be removed. The tool <strong>Select</strong> can be used.</li>  <li>If a GTF dataset is not available for your genome, a two-column tabular dataset containing <code class="language-plaintext highlighter-rouge">transcript &lt;tab&gt; gene</code> can be used instead with most of these tools. Some reformatting of a different annotation file type might be needed. Tools in the groups under <strong>GENERAL TEXT TOOLS</strong> can be used.</li>  <li>Make sure that if your count inputs have a header, the option <strong>Files have header?</strong> is set to <strong>Yes</strong>. If no header, set to <strong>No</strong>.</li>  <li>Custom genomes/transcriptomes/exomes must be formatted correctly before <span class="notranslate">mapping</span>.</li>  <li>Any reference annotation should be an exact match for any genome/transcriptome/exome used for <span class="notranslate">mapping</span>. Build and version matter.</li>  <li>Avoid using <a href="https://genome.ucsc.edu/">UCSC‚Äôs</a> annotation extracted from their Table Browser. All GTF datasets from the UCSC Table Browser have the same content populated for the transcript_id and gene_id values. Both are the ‚Äútranscript_id‚Äù, which creates scientific content problems, effectively meaning that the counts will be summarized ‚Äúby transcript‚Äù and not ‚Äúby gene‚Äù, even if labeled in a tool‚Äôs output as being ‚Äúby gene‚Äù. It is usually possible to extract gene/transcript in tabular format from other related tables. Review the Table Browser usage at <a href="https://genome.ucsc.edu/">UCSC</a> for how to link/extract data or ask them for guidance if you need extra help to get this information for a specific data track.</li></ul>  <p>Note: Selected genomes at UCSC do have a reference anotatation GTF pre-computed and available with a Gene Symbol populated into the ‚Äúgene_id‚Äù value. Find these in the UCSC ‚ÄúDownloads‚Äù area. When available, the link can be directly copy/pasted into the Upload tool in <span class="notranslate">Galaxy</span>. Allow <span class="notranslate">Galaxy</span> to <em>autodetect the datatype</em> to produce an uncompressed GTF dataset in your history ready to use with tools.</p></blockquote>




<h4 class="faq-area"> Understanding 'canceled by admin' or cluster failure error messages <a href="/training-material/faqs/galaxy/analysis_job_failure_cluster_failure.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-canceled-by-admin-or-cluster-failure-error-messages"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding ‚Äòcanceled by admin‚Äô or cluster failure error messages</h3>  <p>The initial error message could be:</p>  <pre><br>This job failed because it was cancelled by an administrator.<br>Please click the bug icon to report this problem if you need help.<br></pre>  <p>Or</p>  <pre><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>  <ul>  <li>Causes:    <ul>      <li>Server or cluster error.</li>      <li>Less frequently, input problems are a factor.</li>    </ul>  </li>  <li>Solutions:    <ul>      <li>Try at least one rerun. Server/cluster errors like this are usually transient.</li>      <li>Review the Solutions section of the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_job_failure_input_problem.html">Understanding input error messages</a> FAQ.</li>      <li>If after any corrections, the job still fails, please report the technical issue <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_reporting_issues.html">following the extended issue guidelines</a>.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Understanding 'exceeds memory allocation' error messages <a href="/training-material/faqs/galaxy/analysis_job_failure_excess_memory.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-exceeds-memory-allocation-error-messages"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding ‚Äòexceeds memory allocation‚Äô error messages</h3>  <p>The error message to be displayed are as follows:</p>  <pre><br>job info:<br>This job was terminated because it used more memory than it was allocated.<br>Please click the bug icon to report this problem if you need help.<br></pre>  <p>Or</p>  <pre><br>stderr:<br>Fatal error: Exit code 1 ()<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>  <p>Sometimes this message may appear at the bottom</p>  <pre><br>job stderr:<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>  <p>In rare cases when the memory quota is exceeded very quickly, an error message such as the following can appear</p>  <pre><br>job stderr:<br>Fatal error: Exit code 1 ()<br>Traceback (most recent call last):<br>(other lines)<br>Memory Error<br></pre>  <p><strong>Note:</strong> Job runtime memory is different from the amount of free storage space (quota) in an account.</p>  <ul>  <li>Causes:    <ul>      <li>The job ran out of memory while executing on the cluster node that ran the job.</li>      <li>The most common reasons for this error are input and tool parameters problems that must be adjusted/corrected.</li>    </ul>  </li>  <li>Solutions:    <ul>      <li>Try at least one rerun to execute the job on a different cluster node.</li>      <li>Review the Solutions section of the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_job_failure_input_problem.html">Understanding input error messages</a> FAQ.</li>      <li>Your data may actually be too large to process at a public <span class="notranslate">Galaxy</span> server. Alte<span class="notranslate">rna</span>tives include setting up a <a href="https://training.galaxyproject.org/training-material/faqs/gtn/galaxy_usage.html">private <span class="notranslate">Galaxy</span> server</a>.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Understanding input error messages <a href="/training-material/faqs/galaxy/analysis_job_failure_input_problem.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-input-error-messages"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding input error messages</h3>  <p>Input problems are very common across any analysis that makes use of programmed tools.</p>  <ul>  <li>Causes:    <ul>      <li>No quality assurance or content/formatting checks were run on the first datasets of an analysis <span class="notranslate">workflow</span>.</li>      <li>Incomplete dataset Upload.</li>      <li>Incorrect or unassigned <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#datasets">datatype</a> or <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_change_dbkey.html">database</a>.</li>      <li>Tool-specific formatting requirements for inputs were not met.</li>      <li>Parameters set on a tool form are a mismatch for the input data content or format.</li>      <li>Inputs were in an error state (red) or were putatively successful (green) but are empty.</li>      <li>Inputs do not meet the datatype specification.</li>      <li>Inputs do not contain the exact content that a tool is expecting or that was input in the form.</li>      <li>Annotation files are a mismatch for the selected or assigned reference genome build.</li>      <li><strong>Special case:</strong> Some of the data were generated outside of <span class="notranslate">Galaxy</span>, but later a built-in indexed genome build was assigned in <span class="notranslate">Galaxy</span> for use with <span class="notranslate">downstream</span> tools. This scenario can work, but only if those two reference genomes are an exact match.</li>    </ul>  </li>  <li>Solutions:    <ul>      <li>Review our <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting Tips</a> for what and where to check.</li>      <li>Review the <a href="https://training.galaxyproject.org/">GTN</a> for related tutorials on tools/analysis plus FAQs.</li>      <li>Review <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> for prior discussion with extended solutions.</li>      <li>Review <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">datatype FAQs</a>.</li>      <li>Review the tool form.        <ul>          <li>Input selection areas include usage help.</li>          <li>The help section at the bottom of a tool form often has examples. Does your own data match the format/content?</li>          <li>See the links to publications and related resources.</li>        </ul>      </li>      <li>Review the inputs.        <ul>          <li>All inputs must be in a success state (green) and actually contain content.</li>          <li>Did you <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_change_datatype.html">directly assign the datatype</a> or <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_convert_datatype.html">convert the datatype</a>? What results when the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_detect_datatype.html">datatype is detected</a> by <span class="notranslate">Galaxy</span>? If these differ, there is likely a content problem.</li>          <li>For most analysis, allowing <span class="notranslate">Galaxy</span> to detect the datatype during Upload is best and adjusting a datatype later should rarely be needed. If a datatype is modified, the change has a specific purpose/reason.</li>          <li>Does your data have headers? Is that in specification for the datatype? Does the tool form have an option to specify if the input has headers or not? Do you need to remove headers first for the correct datatype to be detected? Example <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF</a>.</li>          <li>Large inputs? Consider modifying your inputs to be smaller. Examples: <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_working_with_fastq.html">FASTQ</a> and <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>        </ul>      </li>      <li>Run quality checks on your data.        <ul>          <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword ‚Äúqa-qc‚Äù for examples.</li>          <li>Search <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> with the keywords ‚Äúqa-qc‚Äù and your datatype(s) for more help.</li>        </ul>      </li>      <li>Reference annotation tips.        <ul>          <li>In most cases, <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF is preferred over GFF3</a>.</li>          <li>Search <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> with the keywords ‚Äúgtf‚Äù and ‚Äúgff3‚Äù for more help.</li>        </ul>      </li>      <li>Input mismatch tips.        <ul>          <li>Do the chromosome/sequence identifiers exactly match between all inputs? Search <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> for more help about how to correct build/version identifier mismatches between inputs.</li>          <li>‚ÄúChr1‚Äù and ‚Äúchr1‚Äù and ‚Äú1‚Äù do not mean the same thing to a tool.</li>        </ul>      </li>      <li>Custom genome transcriptome exome tips. See <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Understanding ValueError error messages <a href="/training-material/faqs/galaxy/analysis_job_failure_value_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-valueerror-error-messages"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding ValueError error messages</h3>  <p>The full error is usually a longer message seen only after clicking on the bug icon or by reviewing the job details <code class="language-plaintext highlighter-rouge">stderr</code>.</p>  <p>How to do both is covered in the Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">FAQ</a>.</p>  <pre><br>stderr<br>...<br>Many lines of text, may include parameters<br>...<br>...<br>ValueError: invalid literal for int() with base 10: some-sequence-read-name<br></pre>  <ul>  <li>Causes:    <ul>      <li>MACS2 produces this error the first time it is run. MACS is not the only tool that can produce this issue, but it is the most common.</li>    </ul>  </li>  <li>Solutions:    <ul>      <li>Try at least one rerun.</li>      <li>MACS/2 is not capable of interpreting sequence read names with spaces included. Try following these two:        <ul>          <li>Remove unmapped <span class="notranslate">reads</span> from the SAM dataset. There are several filtering tools in the groups <strong>SAMTools</strong> and <strong>Picard</strong> that can do this.</li>          <li>Convert the SAM input to BAM format with the tool <strong>SAMtools: SAM-to-BAM</strong>. When compressed input is given to MACS, the spaces are no longer an issue.</li>        </ul>      </li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Understanding walltime error messages <a href="/training-material/faqs/galaxy/analysis_job_failure_walltime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-walltime-error-messages"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding walltime error messages</h3>  <p>The full error message will be reported as below, and can be found by clicking on the bug icon for a failed job run (red dataset):</p>  <pre><br>job info:<br>This job was terminated because it ran longer than the maximum allowed job run time.<br>Please click the bug icon to report this problem if you need help.<br></pre>  <p>Or sometimes,</p>  <pre><br>job stderr:<br>slurmstepd: error: *** JOB XXXX ON XXXX CANCELLED AT 2019-XX-XXTXX:XX:XX DUE TO TIME LIMIT ***<br><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>  <ul>  <li>Causes:    <ul>      <li>The job execution time exceeded the ‚Äúwall-time‚Äù on the cluster node that ran the job.</li>      <li>The server may be undergoing maintenance.</li>      <li>Very often input problems also cause this same error.</li>    </ul>  </li>  <li>Solutions:    <ul>      <li>Try at least one rerun.</li>      <li>Check the server homepage for banners or notices. Selected servers also post status <a href="https://status.galaxyproject.org/">here</a>.</li>      <li>Review the Solutions section of the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_job_failure_input_problem.html">Understanding input error messages</a> FAQ.</li>      <li>Your data may actually be too large to process at a public <span class="notranslate">Galaxy</span> server. Alte<span class="notranslate">rna</span>tives include setting up a <a href="https://training.galaxyproject.org/training-material/faqs/gtn/galaxy_usage.html">private <span class="notranslate">Galaxy</span> server</a>.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Will my jobs keep running? <a href="/training-material/faqs/galaxy/analysis_jobs_keep_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-will-my-jobs-keep-running"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Will my jobs keep running?</h3>  <p><span class="notranslate">Galaxy</span> is a fantastic system, but some users find themselves wondering:</p>  <blockquote>  <p>Will my jobs keep running once I‚Äôve closed the tab? Do I need to keep my browser open?</p></blockquote>  <p>No, you don‚Äôt! You can safely:</p>  <ol>  <li>Start jobs</li>  <li>Shut down your computer</li></ol>  <p>and your jobs will keep running in the background! Whenever you next visit <span class="notranslate">Galaxy</span>, you can check if your jobs are still running or completed.</p>  <p><strong>However</strong>, this is not true for uploading data from your computer. You must wait for uploading a dataset <em>from your computer</em> to finish. (Uploading via URL is not affected by this, if you‚Äôre uploading from URL you can close your computer.)</p></blockquote>




<h4 class="faq-area"> My jobs aren't running! <a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-jobs-arent-running"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My jobs aren‚Äôt running!</h3>  <ol>  <li>    <p><strong>Please make sure you are logged in</strong>. At the top menu bar, you should see a section labeled ‚ÄúUser‚Äù. If you see ‚ÄúLogin/Register‚Äù here you are not logged in.</p>  </li>  <li><strong>Activate your account</strong>. If you have recently registered your account, you may first have to activate it. You will receive an e-mail with an activation link.    <ul>      <li>Make sure to check your spam folder!</li>    </ul>  </li>  <li>    <p><strong>Be patient</strong>. <span class="notranslate">Galaxy</span> is a free service, when a lot of people are using it, you may have to wait longer than usual (especially for ‚Äòbig‚Äô jobs, e.g. alignments).</p>  </li>  <li><strong>Contact Support</strong>. If you really think something is wrong with the server, you can <a href="/training-material/faqs/galaxy/support.html">ask for support</a></li></ol></blockquote>




<h4 class="faq-area"> Reporting usage problems, security issues, and bugs <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-reporting-usage-problems-security-issues-and-bugs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Reporting usage problems, security issues, and bugs</h3>  <ul>  <li>For reporting <strong>Usage Problems</strong>, related to tools and functions, head to the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> site.    <ul>      <li><strong>Red Error Datasets</strong>:        <ul>          <li>Refer to the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a> FAQ for red error in datasets.</li>        </ul>      </li>      <li><strong>Unexpected results in Green Success Dataset</strong>:        <ul>          <li>To resolve it you may be asked to send in a shared history link and possibly a shared <span class="notranslate">workflow</span> link. For sharing your history, refer to this <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/histories_sharing.html">link</a>.</li>          <li>To reach our support team, visit <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#support">Support FAQs</a>.</li>        </ul>      </li>      <li><strong>Functionality problems</strong>:        <ul>          <li>Using <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> is the best way to get help in most cases.</li>          <li>If the problem is more complex, email a description of the problem and how to reproduce it.</li>        </ul>      </li>      <li><strong>Administrative problems</strong>:        <ul>          <li>If the problem is present in your own <span class="notranslate">Galaxy</span>, the administrative configuration may be a factor.</li>          <li>For the fastest help directly from the development community, admin issues can be alte<span class="notranslate">rna</span>tively reported to the mailing list or the <span class="notranslate">Galaxy</span>Project <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a>.</li>        </ul>      </li>    </ul>  </li>  <li>For <strong>Security Issues</strong>, do not report them via GitHub. Kindly disclose these as explained in this <a href="https://github.com/galaxyproject/galaxy/blob/dev/SECURITY_POLICY.md">document</a>.</li>  <li>For <strong>Bug Reporting</strong>, create a Github issue. Include the steps mentioned <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">here</a>.</li>  <li>Search the <a href="https://training.galaxyproject.org/search">GTN Search</a> to find prior Q &amp; A, FAQs, tutorials, and other documentation across all <span class="notranslate">Galaxy</span> resources, to verify in case your issue was already faced by someone.</li></ul></blockquote>




<h4 class="faq-area"> Results may vary <a href="/training-material/faqs/galaxy/analysis_results_may_vary.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>




<h4 class="faq-area"> Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When you get a red dataset in your history, it means something went wrong. But how can you find out what it was? And how can you report errors?
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-troubleshooting-errors"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Troubleshooting errors</h3>  <p>When something goes wrong in <span class="notranslate">Galaxy</span>, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your <span class="notranslate">Galaxy</span> errors.</p>  <ol>  <li><strong>Expand</strong> the red history dataset by clicking on it.    <ul>      <li>Sometimes you can already see an error message here</li>    </ul>  </li>  <li>    <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></p>  </li>  <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:    <ul>      <li>Expand the history item</li>      <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>      <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:        <ul>          <li>Tool Standard Output</li>          <li>Tool Standard Error</li>        </ul>      </li>    </ul>  </li>  <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.    <ul>      <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></li>      <li>Write down any information you think might help solve the problem        <ul>          <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>        </ul>      </li>      <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> Report</strong> button</li>    </ul>  </li>  <li><strong>Ask for help!</strong>    <ul>      <li>Where?        <ul>          <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a></li>          <li>In the <a href="https://gitter.im/galaxyproject/Lobby"><span class="notranslate">Galaxy</span> Gitter Channel</a></li>          <li>Browse the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>        </ul>      </li>      <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> What is the advantage of breaking down protein to peptides before mass spec? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-is-the-advantage-of-breaking-down-protein-to-peptides-before-mass-spec"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What is the advantage of breaking down protein to peptides before mass spec?</h3>  <p>Mass spectrometry works better for peptides: LC separation and ionization is working better on peptides than on proteins and proteins generate too complex and overlaying mass spectra due to their isotopes and their mass might be shifted due to posttranslational modifications or point mutations.</p></blockquote>




<h4 class="faq-area"> Creating an account <a href="/training-material/faqs/galaxy/creating_account.html"><i class="fas fa-external-link-alt"></i></a> </h4>
To get access to all features of a <span class="notranslate">Galaxy</span> instance, you need to create an account.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-an-account"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating an account</h3>  <p>To create an account:</p>  <ol>  <li>Click <strong>Login or Register</strong></li>  <li>At the bottom of the form click <strong>Register here</strong></li>  <li>Fill the form and click <strong>Create</strong></li></ol></blockquote>




<h4 class="faq-area"> When can you use (or cannot use) Match between runs in MaxQuant? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-when-can-you-use-or-cannot-use-match-between-runs-in-maxquant"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> When can you use (or cannot use) Match between runs in MaxQuant?</h3>  <p>No golden rule here. For quantitative comparison of different sample groups it can be valuable to use MBR to increase the number of identified + quantified proteins in all samples and then have more proteins that occur in most of the samples to compare them.</p></blockquote>




<h4 class="faq-area"> If you use a mqpar file, can you include modifications that are not in the <span class="notranslate">Galaxy</span> version? For instance, propionamide (Cys alkylation by acrylamide). <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-if-you-use-a-mqpar-file-can-you-include-modifications-that-are-not-in-the-galaxy-version-for-instance-propionamide-cys-alkylation-by-acrylamide"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> If you use a mqpar file, can you include modifications that are not in the <span class="notranslate">Galaxy</span> version? For instance, propionamide (Cys alkylation by acrylamide).</h3>  <p>No, one is limited to the modifications which are installed in MaxQuant. The mqpar only contains more parameters / options than the GUI in <span class="notranslate">galaxy</span>.Note: one must use an mqpar from the same version like MaxQuant!</p></blockquote>




<h4 class="faq-area"> What does it mean to normalize the LFQ intensities? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-does-it-mean-to-normalize-the-lfq-intensities"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What does it mean to normalize the LFQ intensities?</h3>  <p>Median normalization typically refers to subtracting the median of all intensities within one sample from all of the intensities (e.g. Intensity of Protein A - Median of all intensities from Sample 1) , to account for measurement variations. Before normalization log2 transformation is required since many statistical tests demand that the data is actually normal distributed. (Non log intensities show very high values but have a minimum (limit of quantification) leading to a somehow right skewed distribution, after log-transformation the intensity distribution is more like a gaussian distribution. Beside the median (or median-polish) normalization there is also other e.g. the quantile  normalization.</p></blockquote>




<h4 class="faq-area"> For the ‚Äúquantitation method‚Äù what is the default if I just leave it as ‚ÄúNone‚Äù? Label free? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-for-the-quantitation-method-what-is-the-default-if-i-just-leave-it-as-none-label-free"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> For the ‚Äúquantitation method‚Äù what is the default if I just leave it as ‚ÄúNone‚Äù? Label free?</h3>  <p>It will report raw intensity (NON-normalized) values which were not normalized like e.g. the LFQ intensities.</p></blockquote>




<h4 class="faq-area"> How many proteins can be identified and quantified in shotgun proteomics? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-how-many-proteins-can-be-identified-and-quantified-in-shotgun-proteomics"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> How many proteins can be identified and quantified in shotgun proteomics?</h3>  <p>This is depending on the sample, the used technique(s) and the mass spectrometer. Routinely most labs obtain 4000 proteins, but with more effort  10.000 proteins could be analyzed in a single run.</p></blockquote>




<h4 class="faq-area"> MSStats: what does ‚Äòcompare groups = yes‚Äô mean? And the comparison matrix to define the contrast between the 2 groups? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-msstats-what-does-compare-groups--yes-mean-and-the-comparison-matrix-to-define-the-contrast-between-the-2-groups"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> MSStats: what does ‚Äòcompare groups = yes‚Äô mean? And the comparison matrix to define the contrast between the 2 groups?</h3>  <p>MSstats consists of three parts:</p>  <ul>  <li>Reading the input files and converting them into an MSstats compatible format, doing some processing of the data at the same time</li>  <li>Data processing: such as protein inference (summary), log2 transformation, normalization and missing value imputation</li>  <li><code class="language-plaintext highlighter-rouge">compare groups = yes</code>, means that the third step is performed, which is statistical analysis: Statistical modelling to find differentially abundant protein between different groups. The groups should be specified as ‚Äúcondition‚Äù in the annotation file and the group comparison matrix file specifies which groups to compare against each other. In the example this is quite simple because there are only 2 groups, with 3 or more groups the comparison matrix could become more complex.</li></ul></blockquote>




<h4 class="faq-area"> Does MaxQuant in <span class="notranslate">Galaxy</span> support TMT, iTRAQ, etc.? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-does-maxquant-in-galaxy-support-tmt-itraq-etc"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Does MaxQuant in <span class="notranslate">Galaxy</span> support TMT, iTRAQ, etc.?</h3>  <p>Yes, iTRAQ 4 and 8 plex;  TMT 2,6,8,10,11 plex; iodoTMT6plex</p></blockquote>




<h4 class="faq-area"> I got slightly different numbers than were in the tutorial <a href="/training-material/topics/introduction/faqs/ucsc.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-got-slightly-different-numbers-than-were-in-the-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I got slightly different numbers than were in the tutorial</h3>  <p>This tutorial uses UCSC which is constantly updating it‚Äôs data! As a result it gets outdated very quickly before we can update it :( But it‚Äôs ok! It‚Äôs expected here to get different numbers.</p></blockquote>



 <br> 

<h3 id="ansible"> <a href="#ansible"> Ansible </a></h3>

<hr/>



<h4 class="faq-area"> Operating system compatibility <a href="/training-material/topics/admin/faqs/admin-testing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-operating-system-compatibility"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Operating system compatibility</h3>  <p>These Ansible roles and training materials were last tested on Centos 7 and Ubuntu 18.04, but will probably work on other RHEL and Debian variants.</p>  <p>The roles that are used in these training are currently used by <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.*</code>, and other, servers in maintaining their infrastructure. (<a href="https://github.com/galaxyproject/infrastructure-playbook/">US</a>, <a href="https://github.com/usegalaxy-eu/infrastructure-playbook">EU</a>, both are running CentOS 7)</p>  <p>If you have an issue running these trainings on your OS flavour, please report <a href="https://github.com/galaxyproject/training-material/issues/new">the issue</a> in the training material and we can see if it is possible to solve.</p></blockquote>




<h4 class="faq-area"> Variable connection <a href="/training-material/topics/admin/faqs/ansible-connection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-variable-connection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Variable connection</h3>  <p>When the playbook runs, as part of the setup, it collects any variables that are set. For a playbook affecting a group of hosts named <code class="language-plaintext highlighter-rouge">my_hosts</code>, it checks many different places for variables, including ‚Äúgroup_vars/my_hosts.yml‚Äù. If there are variables there, they‚Äôre added to the collection of current variables. It also checks ‚Äúgroup_vars/all.yml‚Äù (for the built-in host group <code class="language-plaintext highlighter-rouge">all</code>). There is a precedence order, but then these variables are available for roles and tasks to consume.</p></blockquote>




<h4 class="faq-area"> Define once, reference many times <a href="/training-material/topics/admin/faqs/ansible_define-once.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-define-once-reference-many-times"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Define once, reference many times</h3>  <p>Using variables, either by defining them ahead of time, or simply accessing them via existing data structures that have been defined, e.g.:</p>  <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># defining a variable that gets reused is great!</span><br><span class="na"><span class="notranslate">galaxy</span>_user</span><span class="pi">:</span> <span class="s"><span class="notranslate">galaxy</span></span><br><br><span class="na"><span class="notranslate">galaxy</span>_config</span><span class="pi">:</span><br>  <span class="na"><span class="notranslate">galaxy</span></span><span class="pi">:</span><br>    <span class="c1"># Re-using the <span class="notranslate">galaxy</span>_config_dir variable saves time and ensures everything</span><br>    <span class="c1"># is in sync!</span><br>    <span class="na">datatypes_config_file</span><span class="pi">:</span> <span class="s2">"</span><span class="s">/datatypes_conf.xml"</span><br><br><span class="c1"># and now we can re-use ""</span><br><span class="c1"># in other places!</span><br><br><br><span class="na"><span class="notranslate">galaxy</span>_config_templates</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="na">src</span><span class="pi">:</span> <span class="s">templates/<span class="notranslate">galaxy</span>/config/datatypes_conf.xml</span><br>    <span class="na">dest</span><span class="pi">:</span> <span class="s2">"</span><span class="s">"</span><br></code></pre></div></div>  <p>Practices like those shown above help to avoid problems caused when paths are defined differently in multiple places. The datatypes config file will be copied to the same path as <span class="notranslate">Galaxy</span> is configured to find it in, because that path is only defined in one place. Everything else is a reference to the original definition! If you ever need to update that definition, everything else will be updated accordingly.</p></blockquote>




<h4 class="faq-area"> What if you forget `--diff`? <a href="/training-material/topics/admin/faqs/ansible_diff.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-if-you-forget---diff"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What if you forget <code class="language-plaintext highlighter-rouge">--diff</code>?</h3>  <p>If you forget to use <code class="language-plaintext highlighter-rouge">--diff</code>, it is not easy to see what has changed. Some modules like the <code class="language-plaintext highlighter-rouge">copy</code> and <code class="language-plaintext highlighter-rouge">template</code> modules have a <code class="language-plaintext highlighter-rouge">backup</code> option. If you set this option, then it will keep a backup copy next to the destination file.</p>  <p>However, most modules do not have such an option, so if you want to know what changes, always use <code class="language-plaintext highlighter-rouge">--diff</code>.</p></blockquote>




<h4 class="faq-area"> Error: "skipping: no hosts matched" <a href="/training-material/topics/admin/faqs/ansible_error-no-hosts-matched.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-error-skipping-no-hosts-matched"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Error: ‚Äúskipping: no hosts matched‚Äù</h3>  <p>There can be multiple reasons this happens, so we‚Äôll step through all of them.We‚Äôll start by assuming you‚Äôre running the command</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ansible-playbook <span class="notranslate">galaxy</span>.yml<br></code></pre></div></div>  <p>The following things can cause issues:</p>  <ol>  <li>Within your <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span>.yml</code>, you‚Äôve referred to a host group that doesn‚Äôt exist or is misspelled. Check the <code class="language-plaintext highlighter-rouge">hosts: <span class="notranslate">galaxy</span>servers</code> to ensure it matches the host group defined in the <code class="language-plaintext highlighter-rouge">hosts</code> file.</li>  <li>Vice-versa, the group in your <code class="language-plaintext highlighter-rouge">hosts</code> file should match the hosts selected in the playbook, <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span>.yml</code>.</li>  <li>If neither of these are the issue, it‚Äôs possible Ansible doesn‚Äôt know to check the <code class="language-plaintext highlighter-rouge">hosts</code> file for the inventory. Make sure you‚Äôve specified <code class="language-plaintext highlighter-rouge">inventory = hosts</code> in your <code class="language-plaintext highlighter-rouge">ansible.cfg</code>.</li></ol></blockquote>




<h4 class="faq-area"> Running Ansible on your remote machine <a href="/training-material/topics/admin/faqs/ansible_local.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-ansible-on-your-remote-machine"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running Ansible on your remote machine</h3>  <p>It is possible to have ansible installed on the remote machine and run it there, not just from your local machine connecting to the remote machine.</p>  <p>Your hosts file will need to use <code class="language-plaintext highlighter-rouge">localhost</code>,  and whenever you run playbooks with <code class="language-plaintext highlighter-rouge">ansible-playbook -i hosts playbook.yml</code>, you will need to add <code class="language-plaintext highlighter-rouge">-c local</code> to your command.</p>  <p>Be <strong>certain</strong> that the playbook that you‚Äôre writing on the remote machine is stored somewhere safe, like your user home directory, or backed up on your local machine. The cloud can be unreliable and things can disappear at any time.</p></blockquote>




<h4 class="faq-area"> What is the difference between the roles with `role:` prefix and without? <a href="/training-material/topics/admin/faqs/ansible_role-prefix.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-is-the-difference-between-the-roles-with-role-prefix-and-without"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What is the difference between the roles with <code class="language-plaintext highlighter-rouge">role:</code> prefix and without?</h3>  <p>The bare role name is just simplified syntax for the roles, you could equally specifiy <code class="language-plaintext highlighter-rouge">role: &lt;name&gt;</code> every time but it‚Äôs only necessary if you want to set additional variables like <code class="language-plaintext highlighter-rouge">become_user</code></p></blockquote>




<h4 class="faq-area"> How do I know what I can do with a role? What variables are available? <a href="/training-material/topics/admin/faqs/ansible_role-properties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-do-i-know-what-i-can-do-with-a-role-what-variables-are-available"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How do I know what I can do with a role? What variables are available?</h3>  <p>You don‚Äôt. There is no standard way for reporting this, but well written roles by trusted authors (e.g. geerlingguy, <span class="notranslate">galaxy</span>project) do it properly and write all of the variables in the README file of the repository. We try to pick sensible roles for you in this course, but, in real life it may not be that simple.</p>  <p>So, definitely check there first, but if they aren‚Äôt there, then you‚Äôll need to read through <code class="language-plaintext highlighter-rouge">defaults/</code> and <code class="language-plaintext highlighter-rouge">tasks/</code> and <code class="language-plaintext highlighter-rouge">templates/</code> to figure out what the role does and how you can control and modify it to accomplish your goals.</p></blockquote>




<h4 class="faq-area"> Is YAML sensitive to True/true/False/false <a href="/training-material/topics/admin/faqs/ansible_yaml.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-is-yaml-sensitive-to-truetruefalsefalse"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Is YAML sensitive to True/true/False/false</h3>  <p>By <a href="https://yaml.org/refcard.html">this reference</a>, YAML doesn‚Äôt really care:</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{ Y, true, Yes, ON   }    : Boolean true<br>{ n, FALSE, No, off  }    : Boolean false<br></code></pre></div></div></blockquote>




<h4 class="faq-area"> <span class="notranslate">Galaxy</span> Admin Training Path <a href="/training-material/topics/admin/faqs/git-gat-path.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-galaxy-admin-training-path"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> <span class="notranslate">Galaxy</span> Admin Training Path</h3>  <p>The yearly <span class="notranslate">Galaxy</span> Admin Training follows a specific ordering of tutorials. Use this timeline to help keep track of where you are in <span class="notranslate">Galaxy</span> Admin Training.</p>  <ol id="git-gat-timeline">    <a href="/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html">    <li class="disabled">        <span>Step 1</span>        <span>ansible-<span class="notranslate">galaxy</span></span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/tus/tutorial.html">    <li class="disabled">        <span>Step 2</span>        <span>tus</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/cvmfs/tutorial.html">    <li class="disabled">        <span>Step 3</span>        <span>cvmfs</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/singularity/tutorial.html">    <li class="disabled">        <span>Step 4</span>        <span>singularity</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/tool-management/tutorial.html">    <li class="disabled">        <span>Step 5</span>        <span>tool-management</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/data-library/tutorial.html">    <li class="disabled">        <span>Step 6</span>        <span>data-library</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html">    <li class="disabled">        <span>Step 7</span>        <span>connect-to-compute-cluster</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/job-destinations/tutorial.html">    <li class="disabled">        <span>Step 8</span>        <span>job-destinations</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/pulsar/tutorial.html">    <li class="disabled">        <span>Step 9</span>        <span>pulsar</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/gxadmin/tutorial.html">    <li class="disabled">        <span>Step 10</span>        <span>gxadmin</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/monitoring/tutorial.html">    <li class="disabled">        <span>Step 11</span>        <span>monitoring</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/tiaas/tutorial.html">    <li class="disabled">        <span>Step 12</span>        <span>tiaas</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/reports/tutorial.html">    <li class="disabled">        <span>Step 13</span>        <span>reports</span>    </li>        </a>        <span aria-hidden="true">        <i class="fas fa-arrow-right" aria-hidden="true"></i>    </span>        <a href="/training-material/topics/admin/tutorials/ftp/tutorial.html">    <li class="disabled">        <span>Step 14</span>        <span>ftp</span>    </li>        </a>    </ol>  <style type="text/css">#git-gat-timeline {    display: flex;    flex-direction: row;    flex-wrap: wrap;}#git-gat-timeline li  {    display: flex;    flex-direction: column;    border: 1px solid black;    border-radius: 5px;    padding: 0.5em;    margin: 0.5em;}#git-gat-timeline li.active {    background: #86d486;    color: black;}#git-gat-timeline li.disabled {    background: #eee;}#git-gat-timeline span {    align-self: center;}</style></blockquote>



 <br> 

<h3 id="ansible-galaxy"> <a href="#ansible-galaxy"> Ansible-<span class="notranslate">galaxy</span> </a></h3>

<hr/>



<h4 class="faq-area"> Customising the welcome page <a href="/training-material/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-customising-the-welcome-page"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Customising the welcome page</h3>  <p>Customising the <code class="language-plaintext highlighter-rouge">welcome.html</code> page is very easy!</p>  <ol>  <li>    <p>Create a custom <code class="language-plaintext highlighter-rouge">welcome.html</code> in <code class="language-plaintext highlighter-rouge">templates/<span class="notranslate">galaxy</span>/config/</code> with whatever content you want, e.g.</p>    <div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;html&gt;</span><br>  <span class="nt">&lt;body&gt;</span><br>    <span class="nt">&lt;h1&gt;</span>Welcome to <span class="notranslate">Galaxy</span><span class="nt">&lt;/h1&gt;</span><br>    <span class="nt">&lt;iframe</span> <span class="na">src=</span><span class="s">"https://<span class="notranslate">galaxy</span>project.org/#"</span> <span class="na">width=</span><span class="s">"100%"</span> <span class="na">height=</span><span class="s">"500px"</span> <span class="nt">/&gt;</span><br>  <span class="nt">&lt;/body&gt;</span><br><span class="nt">&lt;/html&gt;</span><br></code></pre></div>    </div>  </li>  <li>    <p>Add it to your deployed configuration</p>    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> a/group_vars/<span class="notranslate">galaxy</span>servers.yml<br>+++ b/group_vars/<span class="notranslate">galaxy</span>servers.yml<br>@@ <span class="nt">-83</span>,12 +83,7 @@ certbot_agree_tos: <span class="nt">--agree-tos</span><br> <span class="notranslate">galaxy</span>_config_templates:<br> - src: templates/<span class="notranslate">galaxy</span>/config/job_conf.yml.j2<br>   dest: <span class="s2">"{{ <span class="notranslate">galaxy</span>_config.<span class="notranslate">galaxy</span>.job_config_file }}"</span><br> - src: templates/<span class="notranslate">galaxy</span>/config/reports.yml<br>   dest: <span class="s2">"{{ <span class="notranslate">galaxy</span>_reports_path }}"</span><br>+- src: templates/<span class="notranslate">galaxy</span>/config/welcome.html<br>+  dest: <span class="s2">"{{ <span class="notranslate">galaxy</span>_config_dir }}"</span><br></code></pre></div>    </div>  </li>  <li>    <p>Update the <code class="language-plaintext highlighter-rouge">templates/nginx/<span class="notranslate">galaxy</span>.j2</code> to point to the new location</p>    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> a/templates/nginx/<span class="notranslate">galaxy</span>.j2<br>+++ b/templates/nginx/<span class="notranslate">galaxy</span>.j2<br>@@ <span class="nt">-83</span>,12 +83,7 @@<br> location config/welcome.html <span class="o">{</span><br>-    <span class="nb">alias</span> <span class="o">{{</span> <span class="notranslate">galaxy</span>_server_dir <span class="o">}}</span>/static/welcome.html.sample<span class="p">;</span><br>+    <span class="nb">alias</span> <span class="o">{{</span> <span class="notranslate">galaxy</span>_config_dir <span class="o">}}</span>/welcome.html<span class="p">;</span><br>     expires 24h<span class="p">;</span><br> <span class="o">}</span><br></code></pre></div>    </div>  </li>  <li>Run the playbook</li>  <li>Done! You might need to refresh / clear your cache before the changes are visible.</li></ol></blockquote>



 <br> 

<h3 id="collections"> <a href="#collections"> Collections </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag to a collection <a href="/training-material/faqs/galaxy/collections_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag-to-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag to a collection</h3>  <ul>  <li>Click on the collection</li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code> in the <code class="language-plaintext highlighter-rouge">Add tags</code> field</p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Press <kbd>Enter</kbd></li>  <li>Check that the tag is appearing below the collection name</li></ul></blockquote>




<h4 class="faq-area"> Creating a dataset collection <a href="/training-material/faqs/galaxy/collections_build_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a dataset collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>    <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>  </li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>




<h4 class="faq-area"> Creating a paired collection <a href="/training-material/faqs/galaxy/collections_build_list_paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-paired-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a paired collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build List of Dataset Pairs</strong></p>  </li>  <li>Change the text of <em>unpaired forward</em> to a common selector for the forward <span class="notranslate">reads</span></li>  <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse <span class="notranslate">reads</span></li>  <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>




<h4 class="faq-area"> Renaming a collection <a href="/training-material/faqs/galaxy/collections_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a collection</h3>  <ol>  <li>Click on the collection</li>  <li>Click on the name of the collection at the top</li>  <li>Change the name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>



 <br> 

<h3 id="contributing"> <a href="#contributing"> Contributing </a></h3>

<hr/>



<h4 class="faq-area"> How to Contribute to <span class="notranslate">Galaxy</span> <a href="/training-material/topics/dev/faqs/contributing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Contributing to <span class="notranslate">Galaxy</span> is a multi-step process, this will guide you through it.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-contribute-to-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to Contribute to <span class="notranslate">Galaxy</span></h3>  <p>To contribute to <span class="notranslate">galaxy</span>, a GitHub account is required. Changes are proposed via a <a href="https://docs.github.com/en/github/collaborating-with-pull-requests">pull request</a>. This allows the project maintainers to review the changes and suggest improvements.</p>  <p>The general steps are as follows:</p>  <ol>  <li>Fork the <span class="notranslate">Galaxy</span> repository</li>  <li>Clone your fork</li>  <li>Make changes in a new branch</li>  <li>Commit your changes, push branch to your fork</li>  <li>Open a pull request for this branch in the <span class="notranslate">upstream</span> <span class="notranslate">Galaxy</span> repository</li></ol>  <blockquote class="details">  <h3 data-toc-skip id="details-git-github-and-galaxy-core"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Git, Github, and <span class="notranslate">Galaxy</span> Core</h3>  <p>For a lot more information about Git branching and managing a repository on Github, see the <a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a> tutorial.</p>  <p>The <a href="/training-material/topics/dev/tutorials/architecture/slides.html"><span class="notranslate">Galaxy</span> Core Architecture slides</a> have a lot of important <span class="notranslate">Galaxy</span> core-related information related to branches, project management, and contributing to <span class="notranslate">Galaxy</span> - under the Project Management section of the slides.</p></blockquote></blockquote>



 <br> 

<h3 id="contributors"> <a href="#contributors"> Contributors </a></h3>

<hr/>



<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-information-should-i-include-when-reporting-a-problem"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What information should I include when reporting a problem?</h3>  <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>  <p><strong>What to include</strong></p>  <ol>  <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>  <li>Which server(s) did you run those commands on?</li>  <li>What account/use<span class="notranslate">rna</span>me did you use?</li>  <li>Where did it go wrong?</li>  <li>What were the stdout/stderr of the tool that failed? Include the text.</li>  <li>Did you try any workarounds? What results did those produce?</li>  <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>  <li>If there are job IDs, please include them as text so administrators don‚Äôt have to manually transcribe the job ID in your picture.</li></ol>  <p>It makes the process of answering ‚Äòbug reports‚Äô much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>  <p><strong>What does a GOOD bug report look like?</strong></p>  <p>The people who provide support for <span class="notranslate">Galaxy</span> are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>  <blockquote class="quote">  <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My use<span class="notranslate">rna</span>me is jane-doe.</p>  <p>Here is everything that I know:</p>  <ul>    <li>The dataset is green, the job did not fail</li>    <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>    <li>I have read it but I do not understand what X/Y means.</li>    <li>The job ID from the output information page is 123123abdef.</li>    <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>  </ul>  <p>Could you help me?</p></blockquote></blockquote>




<h4 class="faq-area"> How does the GTN implement the "Ten simple rules for collaborative lesson development" <a href="/training-material/faqs/gtn/collaborative_development.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-the-gtn-implement-the-ten-simple-rules-for-collaborative-lesson-development"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does the GTN implement the ‚ÄúTen simple rules for collaborative lesson development‚Äù</h3>  <p>The GTN framework is inherently collaborative and community-driven, and comprises a growing number of contributors with expertise in a wide range of scientific and technical domains. Given this highly collaborative nature of a community with very different skill sets, the GTN framework has evolved over the years to facilitate the contribution and maintenance of the tutorials. We aim to adhere to best-practice guidelines for collaborative lesson development described in <span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>. The structure of the tutorials and repository has been made modular with unified syntax and use of snippets enabling easy access for authors to add common tips and tricks new users might need to know. This system allows for easy updating of all tutorials, if there is a change in tools or interface. More generally, we continually strive to lower contribution barriers for content creators by providing a framework that is easy to use for training developers regardless of their level of knowledge of the underlying technical framework.</p>  <p>Implementation of the ‚ÄúTen simple rules for collaborative lesson development‚Äù (<span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>) in the training material:</p>  <table>  <thead>    <tr>      <th>Rules</th>      <th>Implementation in the GTN framework</th>    </tr>  </thead>  <tbody>    <tr>      <td>Clarify audience</td>      <td>Tutorial metadata includes level indicators (introductory, intermediate, advanced) and a list of prerequisite tutorials as recommended prior knowledge. This information is rendered at the top of each tutorial.</td>    </tr>    <tr>      <td>Make lessons modular</td>      <td>Development of small tutorials linked together via learning paths</td>    </tr>    <tr>      <td>Teach best practice lesson development</td>      <td>We maintain the topic <a href="/training-material/topics/contributing/"><em>Contributing to the <span class="notranslate">Galaxy</span> Training Material</em></a> including numerous tutorials describing how to create new content. Furthermore, quarterly online collaboration fest (CoFests) are organized, where contributors can get direct support. Development of a Train the Trainer program and a mentoring program for instructors, in which lesson development is taught</td>    </tr>    <tr>      <td>Encourage and empower contributors</td>      <td>Involve them in reviews. Mentor them. Encourage them to become maintainers.</td>    </tr>    <tr>      <td>Build community around lessons</td>      <td>Quarterly online collaboration fest (CoFests) and Community calls. Chat on our Gitter/Matrix channel.</td>    </tr>    <tr>      <td>Publish periodically and recognize contributions</td>      <td>Author listed on tutorials. Hall of fame listing all contributors. Full tutorial citation at the end of the tutorial. Tweet about new or updated tutorials. List of new or updated tutorials in <span class="notranslate">Galaxy</span> Community newsletter. Soon: publication of tutorials via article</td>    </tr>    <tr>      <td>Evaluate lessons at several scales</td>      <td>Tutorial change (Pull Request) review. Embedded feedback form in tutorials for trainee feedback. Instructor feedback. Automatic <span class="notranslate">workflow</span> testing</td>    </tr>    <tr>      <td>Reduce, re-use, recycle</td>      <td>Sharing content between tutorials, specially using snippets. Development of small modular tutorials linked by learning paths</td>    </tr>    <tr>      <td>Link to other resources</td>      <td>Links to original paper, documentation, exte<span class="notranslate">rna</span>l tutorials and other material</td>    </tr>    <tr>      <td>You can‚Äôt please everyone</td>      <td>but we can try (several different <span class="notranslate">Galaxy</span> introduction tutorials for different audience). Aim to clearly state what the tutorial does and does not cover, at the start.</td>    </tr>  </tbody></table></blockquote>




<h4 class="faq-area"> Thanks! <a href="/training-material/faqs/gtn/contributors_appreciated.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-thanks"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Thanks!</h3>  <p>First off, thanks for your interest in contributing to the <span class="notranslate">Galaxy</span> training materials!</p>  <p>Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.</p>  <p>We will address your issues and/or assess your change proposal as promptly as we can, and help you become a member of our community. You can also check our <a href="/training-material/topics/contributing/">tutorials</a> for more details.</p></blockquote>




<h4 class="faq-area"> How can I get started with contributing? <a href="/training-material/faqs/gtn/contributors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-get-started-with-contributing"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I get started with contributing?</h3>  <p>If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.</p>  <h4 id="1-checking-existing-tutorials">1. Checking existing tutorials</h4>  <p>A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?</p>  <p>Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.</p>  <p><em>Basic</em></p>  <ul>  <li><strong>Test</strong> the tutorial on a running <span class="notranslate">Galaxy</span> instance    <ul>      <li>For example <a href="https://usegalaxy.org"><span class="notranslate">Galaxy</span> Main</a>, <a href="https://usegalaxy.eu"><span class="notranslate">Galaxy</span> Europe</a>, or <a href="https://usegalaxy.org.au"><span class="notranslate">Galaxy</span> Australia</a></li>      <li>Report any issues you run into</li>    </ul>  </li>  <li><strong>Language</strong> editing    <ul>      <li>Fix spelling and grammar mistakes</li>      <li>Simplify the English (to make it more accessible)</li>    </ul>  </li></ul>  <p><em>Intermediate</em></p>  <ul>  <li><strong>Metadata</strong>    <ul>      <li>Are the objectives, keypoints and time estimate filled in?</li>      <li>Do they fit with the contents of the tutorial?</li>    </ul>  </li>  <li><strong>Content</strong>    <ul>      <li>Is there enough background information provided in the introduction section and throughout the tutorial?</li>      <li><strong>Question boxes</strong>        <ul>          <li>Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)</li>          <li>Check that answers are still up-to-date</li>        </ul>      </li>      <li><strong>Screenshots and Videos</strong>        <ul>          <li>Make sure there is also a textual description of the image/video contents</li>          <li>Does the screenshot add value to the tutorial or can it be removed?</li>        </ul>      </li>    </ul>  </li></ul>  <p><em>Advanced</em></p>  <ul>  <li><strong>Workflows</strong>    <ul>      <li>Add a <span class="notranslate">workflow</span> definition file <code class="language-plaintext highlighter-rouge">.ga</code> if none is present</li>      <li>Check that the existing <span class="notranslate">workflow</span> is up-to-date with the tutorial contents</li>      <li>Enable <a href="https://github.com/usegalaxy-eu/workflow-testing"><span class="notranslate">workflow</span> testing</a></li>    </ul>  </li>  <li><strong>Tours</strong>    <ul>      <li>Add a tour if none exists</li>      <li>Run the existing tour and check that it is up-to-date with the tutorial contents</li>    </ul>  </li>  <li><strong>Datasets</strong>    <ul>      <li>Check that all datasets used in the tutorial are present in Zenodo</li>      <li>Add a data-library.yaml file if none exists</li>    </ul>  </li></ul>  <h4 id="2-reviewing-pull-requests">2. Reviewing pull requests</h4>  <p>Another great way to help out the project is by reviewing <a href="https://github.com/galaxyproject/training-material/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc">open pull requests</a>. You can use the above checklist as a guide for your review. Some documentation about how to add your review in the GitHub interface can be found <a href="https://help.github.com/articles/about-pull-request-reviews/">here</a></p></blockquote>




<h4 class="faq-area"> How can I contribute in "advanced" mode? <a href="/training-material/faqs/gtn/contributors_git_advanced.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-contribute-in-advanced-mode"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I contribute in ‚Äúadvanced‚Äù mode?</h3>  <p>Most of the content is written in <a href="https://guides.github.com/features/mastering-markdown/">GitHub Flavored Markdown</a> with some metadata (or variables) found in <a href="http://yaml.org/">YAML</a> files. Everything is stored on our <a href="https://github.com/galaxyproject/training-material">GitHub repository</a>. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (<em>e.g.</em> <code class="language-plaintext highlighter-rouge">transcriptomics</code> directory contains the material related to transcriptomic analysis). Each topic has the following <a name="structure">structure</a>:</p>  <p><img src="/training-material/shared/images/repo_organization.png" alt="Structure of the repository" width="400px" /></p>  <ul>  <li>a metadata file in YAML format</li>  <li>a directory with the topic introduction slide deck in Markdown with introductions to the topic</li>  <li>    <p>a directory with the tutorials:</p>    <p>Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:</p>    <ul>      <li>a tutorial file written in Markdown with hands-on</li>      <li>an optional slides file in Markdown with slides to support the tutorial</li>      <li>a directory with <span class="notranslate">Galaxy</span> Interactive Tours to reproduce the tutorial</li>      <li>a directory with workflows extracted from the tutorial</li>      <li>a YAML file with the links to the input data needed for the tutorial</li>      <li>a YAML file with the description of needed tools to run the tutorial</li>    </ul>  </li>  <li>a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).</li></ul>  <p>To manage changes, we use <a href="https://guides.github.com/introduction/flow/">GitHub flow</a> based on Pull Requests (check our <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">tutorial</a>):</p>  <ol>  <li><a href="https://help.github.com/articles/fork-a-repo/">Create a fork</a> of this repository on GitHub</li>  <li>Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (<code class="language-plaintext highlighter-rouge">git submodule init</code> and <code class="language-plaintext highlighter-rouge">git submodule update</code>)</li>  <li>Create a new branch in your local copy for each significant change</li>  <li>Commit the changes in that branch</li>  <li>Push that branch to your fork on GitHub</li>  <li>Submit a pull request from that branch to the <a href="https://github.com/galaxyproject/training-material">original repository</a></li>  <li>If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically</li>  <li>Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.</li></ol>  <p>Globally, the process of development of new content is open and transparent:</p>  <ol>  <li>Creation of a branch derived from the main branch of the GitHub repository</li>  <li>Initialization of a new directory for the tutorial</li>  <li>Filling of the metadata with title, questions, learning objectives, etc</li>  <li>Generation of the input dataset for the tutorial</li>  <li>Filling of the tutorial content</li>  <li>Extraction of the workflows of the tutorial</li>  <li>Automatic extraction of the required tools to populate the tool file</li>  <li>Automatic annotation of the public <span class="notranslate">Galaxy</span> servers</li>  <li>Generation of an interactive tour for the tutorial with the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tourbuilder</a> web-browser extension</li>  <li>Upload of the datasets to Zenodo and addition of the links in the data library file.</li>  <li>Once ready, opening a Pull Request</li>  <li>Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI</li>  <li>Review of the content by several other instructors via discussions</li>  <li>After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI</li>  <li>Regeneration of the website and publication on <a href="/training-material/">https://training.<span class="notranslate">galaxy</span>project.org/training-material/</a></li>  <li>Generation of PDF artifacts of the tutorials and slides and upload on the FTP server</li>  <li>Population of <a href="https://tess.elixir-europe.org/">TeSS</a>, the ELIXIR‚Äôs Training Portal, via the metadata</li></ol>  <p><img src="/training-material/shared/images/development_process.png" alt="Development process" /></p>  <p>To learn how to add new content, check out our <a href="/training-material/topics/contributing/">series of tutorials on creating new content</a>:</p>  <ol>   <li><a href="/training-material/topics/contributing/slides/introduction.html">Overview of the <span class="notranslate">Galaxy</span> Training Material</a></li>       <li><a href="/training-material/topics/contributing/tutorials/slides-with-video/tutorial.html">Adding auto-generated video to your slides</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-quiz/tutorial.html">Adding Quizzes to your Tutorial</a></li>          <li><a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a></li>          <li><a href="/training-material/topics/contributing/tutorials/github-interface-contribution/tutorial.html">Contributing with GitHub via its interface</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">Creating a new tutorial</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">Creating content in Markdown</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-tours/tutorial.html">Creating Interactive <span class="notranslate">Galaxy</span> Tours</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html">Creating Slides</a></li>           <li><a href="/training-material/topics/contributing/tutorials/generating-pdf/tutorial.html">Generating PDF artefacts of the website</a></li>          <li><a href="/training-material/topics/contributing/tutorials/schemas/tutorial.html">GTN Metadata</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-topic/tutorial.html">Including a new topic</a></li>          <li><a href="/training-material/topics/contributing/tutorials/running-jekyll/tutorial.html">Running the GTN website locally</a></li>          <li><a href="/training-material/topics/contributing/tutorials/gitpod/tutorial.html">Running the GTN website online using GitPod</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html">Tools, Data, and Workflows for tutorials</a></li>          <li><a href="/training-material/topics/contributing/tutorials/admin-knitting/tutorial.html">Updating diffs in admin training</a></li>      </ol>  <p>We also strongly recommend you read and follow <a href="https://carpentries.org/">The Carpentries</a> recommendations on <a href="https://carpentries.github.io/lesson-example/01-design/">lesson design</a> and <a href="https://carpentries.github.io/instructor-training/15-lesson-study/">lesson writing</a> if you plan to add or change some training materials, and also to check the <a href="#structure">structure of the training material</a>.</p></blockquote>




<h4 class="faq-area"> How can I fix mistakes or expand an existing tutorial using the GitHub interface? <a href="/training-material/faqs/gtn/contributors_github_interface.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I fix mistakes or expand an existing tutorial using the GitHub interface?</h3>  <p>Check our tutorial to learn how to use the GitHub interface (soon‚Ä¶)</p></blockquote>




<h4 class="faq-area"> How can I give feedback? <a href="/training-material/faqs/gtn/contributors_giving_feedback.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-give-feedback"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I give feedback?</h3>  <p>At the end of each tutorial, there is a link to <a href="https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit">a feedback form</a>. We use this information to improve our tutorials.</p>  <p>For global feedbacks, you can open an <a href="https://github.com/galaxyproject/training-material/issues/new">issue on GitHub</a>, write us on <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or send us an <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>.</p></blockquote>




<h4 class="faq-area"> What can I do to help the project? <a href="/training-material/faqs/gtn/contributors_helping_project.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-can-i-do-to-help-the-project"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What can I do to help the project?</h3>  <p>In <a href="https://github.com/galaxyproject/training-material/issues">issues</a>, you will find lists of issues to fix and features to implement (with the <a href="https://github.com/galaxyproject/training-material/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Anewcomer-friendly">‚Äúnewcomer-friendly‚Äù</a> label for example). Feel free to work on them!</p></blockquote>




<h4 class="faq-area"> How can I report mistakes or errors? <a href="/training-material/faqs/gtn/contributors_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-report-mistakes-or-errors"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I report mistakes or errors?</h3>  <p>The easiest way to start contributing is to <a href="https://github.com/galaxyproject/training-material/issues/new">file an issue</a> to tell us about a problem such as a typo, spelling mistake, or a factual error. You can then introduce yourself and meet some of our community members.</p></blockquote>




<h4 class="faq-area"> How can I test an Interactive Tour? <a href="/training-material/faqs/gtn/contributors_tours.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-test-an-interactive-tour"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I test an Interactive Tour?</h3>  <p>Perhaps you‚Äôve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tour builder</a> browser extension.</p>  <ol>  <li>Install the Tour Builder extension to your browser (<a href="https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg">Chrome Web Store</a>, <a href="https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/">Firefox add-on</a>).</li>  <li>Navigate to a <span class="notranslate">Galaxy</span> instance supporting the tutorial. To find which <span class="notranslate">Galaxy</span> instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the use<span class="notranslate">galaxy</span>.* instances (<a href="https://usegalaxy.org"><span class="notranslate">Galaxy</span> Main</a>, <a href="https://usegalaxy.eu"><span class="notranslate">Galaxy</span> Europe</a>, or <a href="https://usegalaxy.org.au"><span class="notranslate">Galaxy</span> Australia</a>) is usually a good bet.</li>  <li>Start the Tour Builder plugin by clicking on the icon in your browser menu bar</li>  <li>Copy the contents of the <code class="language-plaintext highlighter-rouge">tour.yaml</code> file into the Tour builder editor window</li>  <li>Click <code class="language-plaintext highlighter-rouge">Save</code> and then <code class="language-plaintext highlighter-rouge">Run</code></li></ol></blockquote>




<h4 class="faq-area"> How can I create new content without dealing with git? <a href="/training-material/faqs/gtn/contributors_without_git.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-create-new-content-without-dealing-with-git"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I create new content without dealing with git?</h3>  <p>If you feel uncomfortable with using the git and the <a href="https://guides.github.com/introduction/flow/">GitHub flow</a>, you can write a new tutorial with any text editor and then contact us (via <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>). We will work together to integrate the new content.</p></blockquote>




<h4 class="faq-area"> How does the GTN ensure our training materials are FAIR? <a href="/training-material/faqs/gtn/fair_training.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-the-gtn-ensure-our-training-materials-are-fair"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does the GTN ensure our training materials are FAIR?</h3>  <p>This infrastructure has been developed in accordance with the FAIR (Findable, Accessible, Interoperable, Reusable) principles for training materials <span class="citation"><a href="#Garcia2020">Garcia <i>et al.</i> 2020</a></span>. Following these principles enables trainers and trainees to find, reuse, adapt, and improve the available tutorials.</p>  <table>  <thead>    <tr>      <th>10 Simple Rules</th>      <th>Implementation in GTN framework</th>    </tr>  </thead>  <tbody>    <tr>      <td>Plan to share your training materials online</td>      <td>Online <a href="https://training.galaxyproject.org/">training material portfolio</a>, managed via <a href="https://github.com/galaxyproject/training-material">a public GitHub repository</a></td>    </tr>    <tr>      <td>Improve findability of your training materials by properly describing them</td>      <td>Rich metadata associated with each tutorial that are visible and accessible via schema.org on each tutorial webpage.</td>    </tr>    <tr>      <td>Give your training materials a unique identity</td>      <td>URL persistency with redirection in case of renaming of tutorials. Data used for tutorials stored on Zenodo and associated with a Digital Object Identifiers (DOI)</td>    </tr>    <tr>      <td>Register your training materials online</td>      <td>Tutorials automatically registered on TeSS, the <a href="https://tess.elixir-europe.org/">ELIXIR‚Äôs Training e-Support System</a></td>    </tr>    <tr>      <td>If appropriate, define access rules for your training materials</td>      <td>Online and free to use without registration</td>    </tr>    <tr>      <td>Use an interoperable format for your training materials</td>      <td>Content of the tutorials and slides written in Markdown. Metadata associated with tutorials stored in YAML, and workflows in JSON. All of this metadata is available <a href="https://training.galaxyproject.org/training-material/api/">from the GTN‚Äôs API</a></td>    </tr>    <tr>      <td>Make your training materials (re-)usable for trainers</td>      <td>Online. Rich metadata associated with each tutorial: title, contributor details, license, description, learning outcomes, audience, requirements, tags/keywords, duration, date of last revision. Strong technical support for each tutorial: <span class="notranslate">workflow</span>, data on Zenodo and also available as data libraries on Use<span class="notranslate">Galaxy</span>.*, tools installable via the <span class="notranslate">Galaxy</span> Tool Shed, list of possible <span class="notranslate">Galaxy</span> instances with the needed tools.</td>    </tr>    <tr>      <td>Make your training materials (re-)usable for trainees</td>      <td>Online and easy to follow hands-on tutorials. Rich metadata with ‚ÄúSpecific, Measurable, Attainable, Realistic and Time bound‚Äù (SMART) learning outcomes following Bloom‚Äôs taxonomy. Requirements and follow-up tutorials to build learning path. List of <span class="notranslate">Galaxy</span> instances offering needed tools, data on Zenodo and also available as data libraries on Use<span class="notranslate">Galaxy</span>.*. Support chat embedded in tutorial pages.</td>    </tr>    <tr>      <td>Make your training materials contribution friendly and citable</td>      <td>Open and collaborative infrastructure with contribution guidelines, a CONTRIBUTING file and a chat. Details to cite tutorials and give credit to contributors available at the end of each tutorial.</td>    </tr>    <tr>      <td>Keep your training materials up-to-date</td>      <td>Open, collaborative and transparent peer-review and curation process. Short time between updates.</td>    </tr>  </tbody></table></blockquote>




<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/gtn/issues_galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-information-should-i-include-when-reporting-a-problem"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What information should I include when reporting a problem?</h3>  <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>  <p><strong>What to include</strong></p>  <ol>  <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>  <li>Which server(s) did you run those commands on?</li>  <li>What account/use<span class="notranslate">rna</span>me did you use?</li>  <li>Where did it go wrong?</li>  <li>What were the stdout/stderr of the tool that failed? Include the text.</li>  <li>Did you try any workarounds? What results did those produce?</li>  <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>  <li>If there are job IDs, please include them as text so administrators don‚Äôt have to manually transcribe the job ID in your picture.</li></ol>  <p>It makes the process of answering ‚Äòbug reports‚Äô much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>  <p><strong>What does a GOOD bug report look like?</strong></p>  <p>The people who provide support for <span class="notranslate">Galaxy</span> are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>  <blockquote class="quote">  <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My use<span class="notranslate">rna</span>me is jane-doe.</p>  <p>Here is everything that I know:</p>  <ul>    <li>The dataset is green, the job did not fail</li>    <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>    <li>I have read it but I do not understand what X/Y means.</li>    <li>The job ID from the output information page is 123123abdef.</li>    <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>  </ul>  <p>Could you help me?</p></blockquote></blockquote>



 <br> 

<h3 id="data upload"> <a href="#data upload"> Data upload </a></h3>

<hr/>



<h4 class="faq-area"> Importing data from a data library <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>




<h4 class="faq-area"> Importing via links <a href="/training-material/faqs/galaxy/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>




<h4 class="faq-area"> Importaci√≥n por medio de enlaces <a href="/training-material/faqs/galaxy-es/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importaci√≥n-por-medio-de-enlaces"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importaci√≥n por medio de enlaces</h3>  <ul>  <li>Copia los enlaces</li>  <li>    <p>Abre el manejador de carga de datos de <span class="notranslate">Galaxy</span> (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> en la parte superior derecha del panel de herramientas)</p>  </li>  <li>Selecciona ‚ÄòPegar/Traer datos‚Äô <strong>Paste/Fetch Data</strong></li>  <li>    <p>Copia los enlaces en el campo de textos</p>  </li>  <li>    <p>Presiona ‚ÄòIniciar‚Äô <strong>Start</strong></p>  </li>  <li>    <p><strong>Close</strong> Cierra la ventana.</p>  </li>  <li><span class="notranslate">Galaxy</span> utiliza los URLs como nombres de forma predeterminada , as√≠ que los tendr√°s que cambiar a algunos que sean m√°s √∫tiles o informativos.the window</li></ul></blockquote>




<h4 class="faq-area"> How can I  upload data using EBI-SRA? <a href="/training-material/faqs/galaxy/dataupload_EBI-SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i--upload-data-using-ebi-sra"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I  upload data using EBI-SRA?</h3>  <ol>  <li>Search for your data directly in the tool and use the <span class="notranslate">Galaxy</span> links.</li>  <li>Be sure to check your sequence data for correct quality score formats and the metadata ‚Äúdatatype‚Äù assignment.</li></ol></blockquote>




<h4 class="faq-area"> Data retrieval with ‚ÄúNCBI SRA Tools‚Äù (fastq-dump) <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-data-retrieval-with-ncbi-sra-tools-fastq-dump"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Data retrieval with ‚ÄúNCBI SRA Tools‚Äù (fastq-dump)</h3>  <p>This section will guide you through downloading experimental metadata, organizing the metadata to short lists corresponding to conditions and replicates, and finally importing the data from NCBI SRA in collections reflecting the experimental design.</p>  <p><strong>Downloading metadata</strong></p>  <ul>  <li>It is critical to understand the condition/replicate structure of an experiment before working with the data so that it can be imported as collections ready for analysis. Direct your browser to <a href="https://www.ncbi.nlm.nih.gov/Traces/study/">SRA Run Selector</a> and in the search box enter GEO data set identifier (for example: GSE72018). Once the study appears, click the box to download the ‚ÄúRunInfo Table‚Äù.</li></ul>  <p><strong>Organizing metadata</strong></p>  <ul>  <li>The ‚ÄúRunInfo Table‚Äù provides the experimental condition and replicate structure of all of the samples. Prior to importing the data, we need to parse this file into individual files that contain the sample IDs of the replicates in each condition. This can be achieved by using a combination of the ‚Äògroup‚Äô, ‚Äòcompare two datasets‚Äô, ‚Äòfilter‚Äô, and ‚Äòcut‚Äô tools <strong>to end up with single column lists of sample IDs (SRRxxxxx) corresponding to each condition.</strong></li></ul>  <p><strong>Importing data</strong></p>  <ul>  <li>Provide the files with SRR IDs to NCBI SRA Tools (fastq-dump) to import the data from SRA to <span class="notranslate">Galaxy</span>. By organizing the replicates of each condition in separate lists, <strong>the data will be imported as ‚Äúcollections‚Äù that can be directly loaded to a <span class="notranslate">workflow</span> or analysis pipeline.</strong></li></ul></blockquote>




<h4 class="faq-area"> NCBI SRA sourced fastq data <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-ncbi-sra-sourced-fastq-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: NCBI SRA sourced fastq data</h3>  <p>In these FASTQ data:</p>  <ul>  <li>The quality score identifier (+) is sometimes not a match for the sequence identifier (@).</li>  <li>The forward and reverse <span class="notranslate">reads</span> may be interlaced and need to be separated into distinct datasets.</li>  <li>Both may be present in a dataset. Correct the first, then the second, as explained below.</li>  <li>Format problems of any kind can cause tool failures and/or unexpected results.</li>  <li>Fix the problems before running any other tools (including <strong>FastQC</strong>, <strong>Fastq Groomer</strong>, or other QA tools)</li></ul>  <p><strong>For inconsistent sequence (@) and quality (+) identifiers</strong></p>  <ul>  <li>    <p>Correct the format by running the tool <strong>Replace Text in entire line</strong> with these options:</p>    <ul>      <li>Find pattern: <code class="language-plaintext highlighter-rouge">^\+SRR.+</code></li>      <li>Replace with: <code class="language-plaintext highlighter-rouge">+</code></li>    </ul>  </li></ul>  <p>Note: If the quality score line is named like ‚Äú+ERR‚Äù instead (or other valid options), modify the pattern search to match.</p>  <p><strong>For interlaced forward and reverse <span class="notranslate">reads</span></strong></p>  <p><strong>Solution 1 (<span class="notranslate">reads</span> named /1 and /2)</strong></p>  <ul>  <li>Use the tool <strong>FASTQ de-interlacer on paired end <span class="notranslate">reads</span></strong></li></ul>  <p><strong>Solution 2 (<span class="notranslate">reads</span> named /1 and /2)</strong></p>  <ul>  <li>Create distinct datasets from an interlaced fastq dataset by running the tool <strong>Manipulate FASTQ <span class="notranslate">reads</span> on various attributes</strong> on the original dataset. It will run twice.</li></ul>  <p>Note: The solution does NOT use the <strong>FASTQ Splitter</strong> tool. The data to be manipulated are interlaced sequences. This is different in format from data that are joined into a single sequence.</p>  <ul>  <li>    <p>Use the Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/1</code> <span class="notranslate">reads</span>**</p>    <p><em>Match <span class="notranslate">Reads</span></em></p>    <ul>      <li>Match <span class="notranslate">Reads</span> by <code class="language-plaintext highlighter-rouge">Name/Identifier</code></li>      <li>Identifier Match Type <code class="language-plaintext highlighter-rouge">Regular <span class="notranslate">Expression</span></code></li>      <li>Match by <code class="language-plaintext highlighter-rouge">.+/2</code></li>    </ul>    <p><em>Manipulate <span class="notranslate">Reads</span></em></p>    <ul>      <li>Manipulate <span class="notranslate">Reads</span> by <code class="language-plaintext highlighter-rouge">Miscellaneous Actions</code></li>      <li>Miscellaneous Manipulation Type <code class="language-plaintext highlighter-rouge">Remove Read</code></li>    </ul>  </li>  <li>    <p>Use these Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/2</code> <span class="notranslate">reads</span>**</p>    <ul>      <li>Exact same settings as above except for this change: Match by <code class="language-plaintext highlighter-rouge">.+/1</code></li>    </ul>  </li></ul>  <p><strong>Solution 3 (<span class="notranslate">reads</span> named /1 and /3)</strong></p>  <ul>  <li>Use the same operations as in Solution 2 above, except change the first <strong>Manipulate FASTQ</strong> query term to be:</li>  <li>Match by <code class="language-plaintext highlighter-rouge">.+/3</code></li></ul>  <p><strong>Solution 4 (<span class="notranslate">reads</span> named without /N)</strong></p>  <ul>  <li>If your data has differently formatted sequence identifiers, the ‚ÄúMatch by‚Äù <span class="notranslate">expression</span> from <em>Solution 2</em> above can be modified to suit your identifiers.</li></ul>  <p>Alte<span class="notranslate">rna</span>tive identifiers such as:</p>  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M00946:180:000000000-ANFB2:1:1107:14919:14410 1:N:0:1<br>@M00946:180:000000000-ANFB2:1:1107:14919:14410 2:N:0:1<br></code></pre></div></div></blockquote>



 <br> 

<h3 id="datasets"> <a href="#datasets"> Datasets </a></h3>

<hr/>



<h4 class="faq-area"> Incomplete Dataset Download <a href="/training-material/faqs/galaxy/dataset_incomplete_download.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-incomplete-dataset-download"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Incomplete Dataset Download</h3>  <p>In case the dataset downloads incompletely:</p>  <ul>  <li>Use the <em>Google Chrome</em> web browser. Sometimes Chrome works better at supporting continuous data transfers.</li>  <li>Use the <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html">command-line option</a> instead. The data may really be too large to download OR your connection is slower. This can also be a faster way to download multiple datasets plus ensure a complete transfer (small or large data).</li></ul></blockquote>




<h4 class="faq-area"> How to find and correct tool errors related to Metadata? <a href="/training-material/faqs/galaxy/dataset_tool_errors_metadata.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding and Correcting Metadata
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-find-and-correct-tool-errors-related-to-metadata"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to find and correct tool errors related to Metadata?</h3>  <p>Tools can error when the wrong dataset attributes (metadata) are assigned. Some of these wrong assignments may be:</p>  <ul>  <li>Tool outputs, which are automatically assigned without user action.</li>  <li>Incorrect autodetection of datatypes, which need manual modification.</li>  <li>Undetected attributes, which require user action (example: assigning database to newly uploaded data).</li></ul>  <p>How to notice missing Dataset Metadata:</p>  <ul>  <li>Dataset will not be downloaded when using the disk icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span>.</li>  <li>Tools error when using a previously successfully used specific dataset.</li>  <li>Tools error with a message that ends with: <code class="language-plaintext highlighter-rouge">OSError: [Errno 2] No such file or directory</code>.</li></ul>  <p>Solution:</p>  <p>Click on the dataset‚Äôs pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to reach the <em>Edit Attributes</em> forms and do one of the following as applies:</p>  <ul>  <li><strong>Directly reset metadata</strong>    <ul>      <li>Find the tab for the metadata you want to change, make the change, and save.</li>    </ul>  </li>  <li><strong>Autodetect metadata</strong>    <ul>      <li>Click on the <em>Auto-detect button</em>. The dataset will turn yellow in the history while the job is processing.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Upload fastqsanger datasets via links <a href="/training-material/faqs/galaxy/dataset_upload_fastqsanger_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-fastqsanger-datasets-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload fastqsanger datasets via links</h3>  <ol>  <li>    <p>Click on <strong>Upload Data</strong> on the top of the left panel:</p>    <p><img src="/training-material/faqs/galaxy/images/upload_data_button.png" alt="UploadDataButton" /></p>  </li>  <li>    <p>Click on <strong>Paste/Fetch</strong>:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_button.png" alt="PasteFetchButton" /></p>  </li>  <li>    <p>Paste URL into text box that would appear:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_modal.png" alt="PasteFetchModal" /></p>  </li>  <li>    <p>Set <strong>Type (set all)</strong> to <code class="language-plaintext highlighter-rouge">fastqsanger</code> or, if your data is compressed as in URLs above (they have <code class="language-plaintext highlighter-rouge">.gz</code> extensions), to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_set_data_type.png" alt="ChangeTypeDropDown" />:</p>  </li></ol></blockquote>




<h4 class="faq-area"> Finding BAM dataset identifiers <a href="/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Quickly learn what the identifiers are in any **BAM** dataset that is the result from <span class="notranslate">mapping</span>
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-bam-dataset-identifiers"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding BAM dataset identifiers</h3>  <ol>  <li>Run <strong>Samtools: IdxStats</strong> on the aligned data (<code class="language-plaintext highlighter-rouge">bam</code> dataset).</li>  <li>The ‚Äúindex header‚Äù chromosome names and lengths will be listed in the output (along with read counts).</li>  <li>Compare the chromosome identifiers to the chromosome (aka ‚Äúchrom‚Äù) field in all other inputs: VCF, GTF, GFF(3), <span class="notranslate">BED</span>, Interval, etc.</li></ol>  <p><strong>Note:</strong></p>  <ul>  <li>The original <span class="notranslate">mapping</span> target may have been a built-in genome index, custom genome (transcriptome, exome, other) ‚Äì the same <code class="language-plaintext highlighter-rouge">bam</code> data will still be summarized.</li>  <li>This method will <em>not</em> work for ‚Äúsequence-only‚Äù <code class="language-plaintext highlighter-rouge">bam</code> datasets, as these usually have no header.</li></ul></blockquote>




<h4 class="faq-area"> Directly obtaining UCSC sourced *genome* identifiers <a href="/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-directly-obtaining-ucsc-sourced-genome-identifiers"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Directly obtaining UCSC sourced <em>genome</em> identifiers</h3>  <p><strong>Option 1</strong></p>  <ol>  <li>Go to <a href="http://genome.ucsc.edu/">UCSC Genome Browser</a>, navigate to ‚Äúgenomes‚Äù, then the species of interest.</li>  <li>On the home page for the genome build, immediately under the top navigation box, in the blue bar next to the full genome build name, you will find <strong>View sequences</strong> button.</li>  <li>Click on the <strong>View sequences</strong> button and it will take you to a detail page with a table listing out the contents.</li></ol>  <p><strong>Option 2</strong></p>  <ol>  <li>Use the tool <strong>Get Data -&gt; UCSC Main</strong>.</li>  <li>In the Table Browser, choose the target genome and build.</li>  <li>For ‚Äúgroup‚Äù choose the last option ‚ÄúAll Tables‚Äù.</li>  <li>For ‚Äútable‚Äù choose ‚ÄúchromInfo‚Äù.</li>  <li>Leave all other options at default and send the output to <span class="notranslate">Galaxy</span>.</li>  <li>This new dataset will load as a tabular dataset into your history.</li>  <li>It will list out the contents of the genome build, including the chromosome identifiers (in the first column).</li></ol></blockquote>




<h4 class="faq-area"> Adding a tag <a href="/training-material/faqs/galaxy/datasets_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tags can help you to better organize your history and track datasets.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>




<h4 class="faq-area"> Changing the datatype <a href="/training-material/faqs/galaxy/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select  your desired datatype    <ul>      <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>    </ul>  </li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Cambiar el tipo de datos <a href="/training-material/faqs/galaxy-es/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-cambiar-el-tipo-de-datos"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Cambiar el tipo de datos</h3>  <ul>  <li>Selecciona sobre el <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>icono del l√°piz</strong> para editar los atributos del conjunto de datos</li>  <li>Selecciona en la pesta√±a <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> en la parte superior del panel central</li>  <li>Selecciona  tu tipo de datos</li>  <li>Da clic en el bot√≥n <strong>Change datatype</strong></li></ul></blockquote>




<h4 class="faq-area"> Changing database/build (dbkey) <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can tell <span class="notranslate">Galaxy</span> which dbkey (e.g. reference genome) your dataset is associated with. This may be used by tools to automatically use the correct settings.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing database/build (dbkey)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Database/Build</strong> field</li>  <li>Select your desired database key from the dropdown list</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Mismatched Chromosome identifiers and how to avoid them <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-mismatched-chromosome-identifiers-and-how-to-avoid-them"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Mismatched Chromosome identifiers and how to avoid them</h3>  <ul>  <li>    <p>The methods listed here help to identify and correct errors or unexpected results linked to inputs having non-identical chromosome identifiers and/or different chromosome sequence content.</p>  </li>  <li>    <p><strong>If using a Custom Reference genome</strong>, the methods below also apply, but the first step is to make certain that the Custom Genome is formatted correctly. Improper formatting is the most common root cause of CG related errors.</p>  </li></ul>  <p>Method 1: <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html">Finding BAM dataset identifiers</a></p>  <p>Method 2: <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html">Directly obtaining UCSC sourced <em>genome</em> identifiers</a></p>  <p>Method 3: <a href="https://galaxyproject.org/support/chrom-identifiers/#adjusting-identifiers-or-input-source">Adjusting identifiers for UCSC sourced data used with other sourced data</a></p>  <p>Method 4: <a href="https://galaxyproject.org/support/chrom-identifiers/#any-mixed-sourced-data">Adjusting identifiers or input source for any mixed sourced data</a></p>  <p><strong>A Note on Built-in Reference Genomes</strong></p>  <ul>  <li>The default variant for all genomes is ‚ÄúFull‚Äù, defined as all primary chromosomes (or scaffolds/contigs) including mitochondrial plus associated unmapped, plasmid, and other segments.</li>  <li>When only one version of a genome is available for a tool, it represents the default ‚ÄúFull‚Äù variant.</li>  <li>    <p>Some genomes will have more than one variant available.</p>    <ul>      <li>The ‚ÄúCanonical Male‚Äù or sometimes simply ‚ÄúCanonical‚Äù variant contains the primary chromosomes for a genome. For example a human ‚ÄúCanonical‚Äù variant contains chr1-chr22, chrX, chrY, and chrM.</li>      <li>The ‚ÄúCanonical Female‚Äù variant contains the primary chromosomes excluding chrY.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Converting the file format <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Some datasets can be transformed into a different format. <span class="notranslate">Galaxy</span> has some built-in file conversion options depending on the type of data you have.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-converting-the-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Converting the file format</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <strong>Convert</strong> tab on the top</li>  <li>Select the appropriate datatype from the list</li>  <li>Click the <strong>Create dataset</strong> button to start the conversion.</li></ul></blockquote>




<h4 class="faq-area"> Creating a new file <a href="/training-material/faqs/galaxy/datasets_create_new_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> allows you to create new files from the upload menu. You can supply the contents of the file.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new file</h3>  <ul>  <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the file contents into the text field</p>  </li>  <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li></ul></blockquote>




<h4 class="faq-area"> Detecting the datatype (file format) <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-detecting-the-datatype-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Detecting the datatype (file format)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Click the <strong>Detect datatype</strong> button to have <span class="notranslate">Galaxy</span> try to autodetect it.</li></ul></blockquote>




<h4 class="faq-area"> Downloading datasets <a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading datasets</h3>  <ol>  <li>Click on the dataset in your history to expand it</li>  <li>Click on the Download icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span> to save the dataset to your computer.</li></ol></blockquote>




<h4 class="faq-area"> Downloading datasets using command line <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-datasets-using-command-line"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading datasets using command line</h3>  <p>From the terminal window on your computer, you can use <strong>wget</strong> or <strong>curl</strong>.</p>  <ol>  <li>Make sure you have <strong>wget</strong> or <strong>curl</strong> installed.</li>  <li>Click on the Dataset name, then click on the copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-link</span>. This is the direct-downloadable dataset link.</li>  <li>Once you have the link, use any of the following commands:    <ul>      <li>For <strong>wget</strong>        <blockquote>          <p><strong><code class="language-plaintext highlighter-rouge">wget '&lt;link&gt;'</code> <br /><code class="language-plaintext highlighter-rouge">wget -O '&lt;link&gt;'</code><br /><code class="language-plaintext highlighter-rouge">wget -O --no-check-certificate '&lt;link&gt;'  # ignore SSL certificate warnings</code> <br /><code class="language-plaintext highlighter-rouge">wget -c '&lt;link&gt;' # continue an interrupted download</code></strong></p>        </blockquote>      </li>      <li>For <strong>curl</strong>        <blockquote>          <p><strong><code class="language-plaintext highlighter-rouge">curl -o outfile '&lt;link&gt;'</code><br /><code class="language-plaintext highlighter-rouge">curl -o outfile --insecure '&lt;link&gt;'     # ignore SSL certificate warnings</code><br /><code class="language-plaintext highlighter-rouge">curl -C - -o outfile '&lt;link&gt;'           # continue an interrupted download</code></strong></p>        </blockquote>      </li>    </ul>  </li>  <li>For dataset collections and datasets within collections you have to supply your API key with the request    <ul>      <li>Sample commands for <strong>wget</strong> and <strong>curl</strong> respectively are:        <blockquote>          <p><strong><code class="language-plaintext highlighter-rouge">wget https://use<span class="notranslate">galaxy</span>.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>          <p><strong><code class="language-plaintext highlighter-rouge">curl -o myfile.txt https://use<span class="notranslate">galaxy</span>.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>        </blockquote>      </li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Finding Datasets <a href="/training-material/faqs/galaxy/datasets_finding_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding Datasets</h3>  <ul>  <li>To review all active Datasets in your account, go to <strong>User &gt; Datasets</strong>.</li></ul>  <p>Notes:</p>  <ul>  <li>Logging out of <span class="notranslate">Galaxy</span> while the Upload tool is still loading data can cause uploads to abort. This is most likely to occur when a dataset is loaded by browsing local files.</li>  <li>If you have more than one browser window open, each with a different <span class="notranslate">Galaxy</span> History loaded, the Upload tool will load data into the most recently used history.</li>  <li>Click on refresh icon <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> at the top of the History panel to display the current active History with the datasets.</li></ul></blockquote>




<h4 class="faq-area"> How to unhide "hidden datasets"? <a href="/training-material/faqs/galaxy/datasets_hidden.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-unhide-hidden-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to unhide ‚Äúhidden datasets‚Äù?</h3>  <p>If you have run a <span class="notranslate">workflow</span> with hidden datasets, in your History:</p>  <ul>  <li>Click the <strong>gear icon</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> ‚Üí Click <strong>Unhide Hidden Datasets</strong></li>  <li>Or use the toggle <code class="language-plaintext highlighter-rouge">hidden</code> to view them</li></ul>  <p>When using the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/histories_copy_dataset.html">Copy Datasets</a> feature, hidden datasets will not be available to transfer from the <strong>Source History</strong> list of datasets. To include them:</p>  <ol>  <li>Click the <strong>gear icon</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> ‚Üí Click <strong>Unhide Hidden Datasets</strong></li>  <li>Click the <strong>gear icon</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> ‚Üí Click <strong>Copy Datasets</strong></li></ol></blockquote>




<h4 class="faq-area"> Different dataset icons and their usage <a href="/training-material/faqs/galaxy/datasets_icons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Icons provide a visual experience for objects, actions, and ideas
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-different-dataset-icons-and-their-usage"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Different dataset icons and their usage</h3>  <p>Dataset icons and their usage:</p>  <ul>  <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> <strong>‚ÄúEye icon‚Äù</strong>: Display data of the job in the browser.</li>  <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>‚ÄúPencil icon‚Äù</strong>: Edit attributes of the job.</li>  <li><i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> <strong>‚Äú‚ÄòX‚Äô icon‚Äù</strong>: Delete the job.</li>  <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-info</span> <strong>‚ÄúInfo icon‚Äù</strong>: Job details and run information.</li>  <li><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> <strong>‚ÄúRefresh/Rerun icon‚Äù</strong>: Run this (selected) job again or examine original submitted form (filled in).</li>  <li><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> <strong>‚ÄúBug icon‚Äù</strong>: Review and optionally submit a bug report.</li></ul></blockquote>




<h4 class="faq-area"> Understanding job statuses <a href="/training-material/faqs/galaxy/datasets_job_status.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Job statuses will help you understand the stages of your work.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-job-statuses"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding job statuses</h3>  <p>The following job statuses will help you better understand the working stage of the process.</p>  <ul>  <li><strong>Green</strong>: The job was completed successfully.</li>  <li><strong>Yellow</strong>: The job is executing. Allow this to complete! Should they run longer, they will fail with a ‚Äúwall-time‚Äù error and turn <em>red</em>.</li>  <li><strong>Grey</strong>: The job is being evaluated to run (new dataset) or is queued. Allow this to complete.</li>  <li><strong>Red</strong>: The job has failed.</li>  <li><strong>Light Blue</strong>: The job is paused. This indicates either an input has a problem or that you have exceeded the disk quota set by the administrator of the <span class="notranslate">Galaxy</span> instance you are working on.</li>  <li><strong>Grey, Yellow, Grey again</strong>: The job is waiting to run due to admin re-run or an automatic fail-over to a longer-running cluster.</li>  <li><strong>Bright blue with moving arrow</strong>: May be found in earlier <span class="notranslate">Galaxy</span> versions. Applies to the ‚ÄúGet Data ‚Üí Upload File‚Äù tool only - the upload job is queuing or running.</li></ul>  <p><strong>It is <em>essential</em> to allow queued jobs to remain queued and not delete/re-run them.</strong></p></blockquote>




<h4 class="faq-area"> Moving datasets between <span class="notranslate">Galaxy</span> servers <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-moving-datasets-between-galaxy-servers"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Moving datasets between <span class="notranslate">Galaxy</span> servers</h3>  <p>On the origin <span class="notranslate">Galaxy</span> server:</p>  <ol>  <li>Click on the name of the dataset to expand the info.</li>  <li>Click on the <strong>Copy link icon</strong> <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-link</span>.</li></ol>  <p>On the destination <span class="notranslate">Galaxy</span> server:</p>  <ol>  <li>Click on <strong>Upload data &gt; Paste / Fetch Data</strong> and paste the link. Select attributes, such as genome assembly, if required. Hit the <strong>Start</strong> button.</li></ol>  <p>Note: The copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-link</span> cannot be used to move HTML datasets (but this can be downloaded using the download button <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span>) and SQLite datasets.</p></blockquote>




<h4 class="faq-area"> Datasets not downloading at all <a href="/training-material/faqs/galaxy/datasets_not_downloading_at_all.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-datasets-not-downloading-at-all"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Datasets not downloading at all</h3>  <ol>  <li>Check to see if pop-ups are blocked by your web browser. Where to check can vary by browser and extensions.</li>  <li>Double check your API key, if used. Go to <strong>User &gt; Preferences &gt; Manage API key.</strong></li>  <li>Check the sharing/permission status of the Datasets. Go to <strong>Dataset &gt; Pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> &gt; Edit attributes &gt; Permissions.</strong> If you do not see a ‚ÄúPermissions‚Äù tab, then you are not the owner of the data.</li></ol>  <p>Notes:</p>  <ul>  <li>If the data was shared with you by someone else from a Shared History, or was copied from a Published History, be aware that there are multiple levels of data sharing permissions.</li>  <li>All data are set to <strong>not shared</strong> by default.</li>  <li>Datasets sharing permissions for a new history can be set before creating a new history. Go to <strong>User &gt; Preferences &gt; Set Dataset Permissions for New Histories.</strong></li>  <li><strong>User &gt; Preferences &gt; Make all data private is a ‚Äúone click‚Äù option to unshare ALL data (Datasets, Histories).</strong> Note that once confirmed and all data is unshared, the action cannot be ‚Äúundone‚Äù in batch, even by an administrator. You will need to re-share data again and/or reset your global sharing preferences as wanted.</li>  <li>Only the data owner has control over sharing/permissions.</li>  <li>Any data you upload or create yourself is automatically owned by you with full access.</li>  <li>You may not have been granted full access if the data were shared or imported, and someone else is the data owner (your copy could be ‚Äúview only‚Äù).</li>  <li>After you have a fully shared copy of any shared/published data from someone else, then you become the owner of that data copy. If the other person or you make changes, it applies to each person‚Äôs copy of the data, individually and only.</li>  <li>Histories can be shared with included Datasets. Datasets can be downloaded/manipulated by others or viewed by others.</li>  <li>Share access to Datasets is distinct but it relates to Histories‚Äô access.</li></ul></blockquote>




<h4 class="faq-area"> Purging datasets <a href="/training-material/faqs/galaxy/datasets_purging_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-purging-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Purging datasets</h3>  <ol>  <li>All account Datasets can be reviewed under <strong>User &gt; Datasets</strong>.</li>  <li>To permanently delete: use the link from within the dataset, or use the Operations on Multiple Datasets functions, or use the Purge Deleted Datasets option in the History menu.</li></ol>  <p>Notes:</p>  <ul>  <li>Within a History, deleted/permanently deleted Datasets can be reviewed by toggling the deleted link at the top of the History panel, found immediately under the History name.</li>  <li>Both active (shown by default) and hidden (the other toggle link, next to the deleted link) datasets can be reviewed the same way.</li>  <li>Click on the far right ‚ÄúX‚Äù to delete a dataset.</li>  <li>Datasets in a deleted state are still part of your quota usage.</li>  <li>Datasets must be purged (permanently deleted) to not count toward quota.</li></ul></blockquote>




<h4 class="faq-area"> Quotas for datasets and histories <a href="/training-material/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-quotas-for-datasets-and-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Quotas for datasets and histories</h3>  <ul>  <li><em>Deleted datasets</em> and <em>deleted histories</em> containing datasets <strong>are considered when calculating quotas</strong>.</li>  <li><em>Permanently deleted datasets</em> and <em>permanently deleted histories</em> containing datasets <strong>are not considered</strong>.</li>  <li><em>Histories/datasets that are shared with you</em> are only <strong>partially considered</strong> unless you import them.</li></ul>  <p><strong>Note:</strong> To reduce quota usage, refer to <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/reduce_quota_usage.html">How can I reduce quota usage while still retaining prior work (data, tools, methods)?</a> FAQ.</p></blockquote>




<h4 class="faq-area"> Renaming a dataset <a href="/training-material/faqs/galaxy/datasets_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Upload few files (1-10) <a href="/training-material/faqs/galaxy/datasets_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-few-files-1-10"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload few files (1-10)</h3>  <ol>  <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>  <li>Click on <strong>Choose local file</strong> and select the files or drop the files in the <strong>Drop files here</strong> part</li>  <li>Click on <strong>Start</strong></li>  <li>Click on <strong>Close</strong></li></ol></blockquote>




<h4 class="faq-area"> Upload many files (>10) via FTP <a href="/training-material/faqs/galaxy/datasets_upload_ftp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-many-files-10-via-ftp"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload many files (&gt;10) via FTP</h3>  <ol>  <li>    <p>Make sure to have an FTP client installed</p>    <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>  </li>  <li>Establish FTP connection to the <span class="notranslate">Galaxy</span> server    <ol>      <li>Provide the <span class="notranslate">Galaxy</span> server‚Äôs FTP server name (e.g. <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.org</code>, <code class="language-plaintext highlighter-rouge">ftp.use<span class="notranslate">galaxy</span>.eu</code>)</li>      <li>Provide the <strong>use<span class="notranslate">rna</span>me</strong> (usually the email address) and the <strong>password</strong> on the <span class="notranslate">Galaxy</span> server</li>      <li>Connect</li>    </ol>  </li>  <li>    <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>    <p>The FTP transfer will start. We need to wait until they are done.</p>  </li>  <li>Open the Upload menu on the <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Choose FTP file</strong> on the bottom</li>  <li>Select files to import into the history</li>  <li>Click on <strong>Start</strong></li></ol></blockquote>




<h4 class="faq-area"> Working with deleted datasets <a href="/training-material/faqs/galaxy/datasets_working_with_deleted_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-working-with-deleted-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Working with deleted datasets</h3>  <p>Deleted datasets and histories can be recovered by users as they are retained in <span class="notranslate">Galaxy</span> for a time period set by the instance administrator. Deleted datasets can be undeleted or permanently deleted within a History. <em>Links to show/hide deleted (and hidden) datasets are at the top of the History panel.</em></p>  <ul>  <li>To review or adjust an individual dataset:    <ol>      <li>Click on the name to expand it.</li>      <li>If it is only deleted, but not permanently deleted, you‚Äôll see a message with links to recover or to purge.        <ul>          <li>Click on <em>Undelete</em> it to recover the dataset, making it active and accessible to tools again.</li>          <li>Click on <em>Permanently remove it from disk</em> to purge the dataset and remove it from the account quota calculation.</li>        </ul>      </li>    </ol>  </li>  <li>To review or adjust multiple datasets in batch:    <ol>      <li>Click on the checked box icon <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-selector</span> near the top right of the history panel to switch into ‚ÄúOperations on Multiple Datasets‚Äù mode.</li>      <li>Accordingly for each individual dataset, choose the selection box. Check the datasets you want to modify and choose your option (show, hide, delete, undelete, purge, and group datasets).</li>    </ol>  </li></ul></blockquote>




<h4 class="faq-area"> Working with very large fasta datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-working-with-very-large-fasta-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Working with very large fasta datasets</h3>  <ul>  <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.    <ul>      <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword ‚Äúqa-qc‚Äù for examples.</li>      <li>Search <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> with the keywords ‚Äúqa-qc‚Äù and ‚Äúfasta‚Äù for more help.</li>    </ul>  </li>  <li>Assembly result?    <ul>      <li>Consider filtering by length to remove <span class="notranslate">reads</span> that did not assemble.</li>      <li>Formatting criteria:        <ul>          <li>All sequence identifiers must be unique.</li>          <li>Some tools will require that there is no description line content, only identifiers, in the fasta title line (‚Äú&gt;‚Äù line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>        </ul>      </li>    </ul>  </li>  <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom genome</a>, transcriptome exome?    <ul>      <li>Only appropriate for smaller genomes (bacterial, viral, most insects).</li>      <li>Not appropriate for any mammalian genomes, or some plants/fungi.</li>      <li>Sequence identifiers must be an exact match with all other inputs or expect problems. See <strong>GFF GFT GFF3</strong>.</li>      <li>Formatting criteria:        <ul>          <li>All sequence identifiers must be unique.</li>          <li>ALL tools will require that there is no description content, only identifiers, in the fasta title line (‚Äú&gt;‚Äù line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>          <li>The only exception is when executing the <strong>MakeBLASTdb</strong> tool and when the input fasta is in NCBI BLAST format (see the tool form).</li>        </ul>      </li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Working with very large fastq datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-working-with-very-large-fastq-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Working with very large fastq datasets</h3>  <ul>  <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.    <ul>      <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword ‚Äúqa-qc‚Äù for examples.</li>      <li>Search <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> with the keywords ‚Äúqa-qc‚Äù and ‚Äúfastq‚Äù for more help.</li>    </ul>  </li>  <li>How to create a single smaller input. Search the tool panel with the keyword ‚Äúsubsample‚Äù for tool choices.</li>  <li>How to create multiple smaller inputs. Start with <strong>Split file to dataset collection</strong>, then merge the results back together using a tool specific for the datatype. Example: BAM results? Use <strong>MergeSamFiles</strong>.</li></ul></blockquote>




<h4 class="faq-area"> Working with GFF GFT GTF2 GFF3 reference annotation <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-working-with-gff-gft-gtf2-gff3-reference-annotation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Working with GFF GFT GTF2 GFF3 reference annotation</h3>  <ul>  <li>All annotation datatypes have a distinct format and content specification.    <ul>      <li>Data providers may release variations of any, and tools may produce variations.</li>      <li>GFF3 data may be labeled as GFF.</li>      <li>Content can overlap but is generally not understood by tools that are expecting just one of these specific formats.</li>    </ul>  </li>  <li>Best practices    <ul>      <li>The sequence identifiers must exactly match between reference annotation and reference genomes transcriptomes exomes.</li>      <li>Most tools expect GFT format unless the tool form specifically notes otherwise.        <ul>          <li>Get the GTF version from the data providers if it is available.</li>          <li>If only GFF3 is available, you can attempt to transform it with the tool <strong>gffread</strong>.</li>        </ul>      </li>      <li>Was GTF data detected as GFF during Upload? It probably has headers.  -Remove the headers (lines that start with a ‚Äú#‚Äù) with the <strong>Select</strong> tool using the option ‚ÄúNOT Matching‚Äù with the regular <span class="notranslate">expression</span>: ^#        <ul>          <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#detecting-the-datatype-file-format">Redetect the datatype</a>. It should be GTF once corrected.</li>        </ul>      </li>      <li>UCSC annotation        <ul>          <li>Find annotation under their Downloads area. The path will be similar to: <code class="language-plaintext highlighter-rouge">https://hgdownload.soe.ucsc.edu/goldenPath/&lt;database&gt;/bigZips/genes/</code></li>          <li>Copy the URL from UCSC and paste it into the Upload tool, allowing <span class="notranslate">Galaxy</span> to detect the datatype.</li>        </ul>      </li>    </ul>  </li></ul></blockquote>



 <br> 

<h3 id="datatypes"> <a href="#datatypes"> Datatypes </a></h3>

<hr/>



<h4 class="faq-area"> Using compressed fastq data as tool inputs <a href="/training-material/faqs/galaxy/datatypes_Using_compressed_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-compressed-fastq-data-as-tool-inputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using compressed fastq data as tool inputs</h3>  <ul>  <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastq</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastq.gz</code> is appropriate.</li>  <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastqsanger</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> is appropriate.</li>  <li>Using uncompressed <code class="language-plaintext highlighter-rouge">fastq</code> data is still an option with tools. The choice is yours.</li></ul>  <p><strong>TIP:</strong> Avoid labeling compressed data with an uncompressed datatype, and the reverse. Jobs using mismatched datatype versus actual format will fail with an error.</p></blockquote>




<h4 class="faq-area"> Best practices for loading fastq data into <span class="notranslate">Galaxy</span> <a href="/training-material/faqs/galaxy/datatypes_best_practices_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-best-practices-for-loading-fastq-data-into-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Best practices for loading fastq data into <span class="notranslate">Galaxy</span></h3>  <ul>  <li>As of release <code class="language-plaintext highlighter-rouge">17.09</code>, <code class="language-plaintext highlighter-rouge">fastq</code> data will have the datatype <code class="language-plaintext highlighter-rouge">fastqsanger</code> auto-detected when that quality score scaling is detected and ‚Äúautodetect‚Äù is used within the Upload tool. Compressed <code class="language-plaintext highlighter-rouge">fastq</code> data will be converted to uncompressed in the history.</li>  <li>To preserve <code class="language-plaintext highlighter-rouge">fastq</code> compression, directly assign the appropriate datatype (eg: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>).</li>  <li>If the data is close to or over 2 GB in size, be sure to use FTP.</li>  <li>If the data was already loaded as <code class="language-plaintext highlighter-rouge">fastq.gz</code>, don‚Äôt worry! Just test the data for correct format (as needed) and assign the <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html">metadata type</a>.</li></ul></blockquote>




<h4 class="faq-area"> How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment? <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-do-fastqgz-datasets-relate-to-the-fastqsanger-datatype-metadata-assignment"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How do <code class="language-plaintext highlighter-rouge">fastq.gz</code> datasets relate to the <code class="language-plaintext highlighter-rouge">.fastqsanger</code> datatype metadata assignment?</h3>  <p>Before assigning <code class="language-plaintext highlighter-rouge">fastqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>, be sure to confirm the format.</p>  <p><strong>TIP:</strong></p>  <ul>  <li>Using <em>non-fastqsanger</em> scaled quality values will cause scientific problems with tools that expected <code class="language-plaintext highlighter-rouge">fastqsanger</code> formatted input.</li>  <li>Even if the tool does not fail, get the format right from the start to avoid problems. Incorrect format is still one of the most common reasons for tool errors or unexpected results (within <span class="notranslate">Galaxy</span> or not).</li>  <li>For more information on <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datatypes_fastq_format.html">How to format fastq data for tools that require .fastqsanger format?</a></li></ul></blockquote>




<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/faqs/galaxy/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-compressed-fastq-files-gz"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Compressed FASTQ files, (<code class="language-plaintext highlighter-rouge">*.gz</code>)</h3>  <ul>  <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.    <ul>      <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>      <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>    </ul>  </li>  <li>Compression saves space (and therefore your quota).</li>  <li>Tools can accept the compressed versions of input files</li>  <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li></ul></blockquote>




<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-compressed-fastq-files-gz"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Compressed FASTQ files, (<code class="language-plaintext highlighter-rouge">*.gz</code>)</h3>  <ul>  <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.    <ul>      <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>      <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>    </ul>  </li>  <li>Compression saves space (and therefore your quota).</li>  <li>Tools can accept the compressed versions of input files</li>  <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li></ul></blockquote>




<h4 class="faq-area"> How to format fastq data for tools that require .fastqsanger format? <a href="/training-material/faqs/galaxy/datatypes_fastq_format.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-format-fastq-data-for-tools-that-require-fastqsanger-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to format fastq data for tools that require .fastqsanger format?</h3>  <ul>  <li>Most tools that accept FASTQ data expect it to be in a specific FASTQ version: <code class="language-plaintext highlighter-rouge">.fastqsanger</code>. The <code class="language-plaintext highlighter-rouge">.fastqsanger</code> datatype must be assigned to each FASTQ dataset.</li></ul>  <p>In order to do that:</p>  <ul>  <li>Watch the <strong><a href="http://vimeo.com/galaxyproject/fastqprep">FASTQ Prep Illumina</a></strong> video for a complete walk-through.</li>  <li>Run <strong>FastQC</strong> first to assess the type.    <ul>      <li>Run <strong>FASTQ Groomer</strong> if the data needs to have the quality scores rescaled.</li>      <li>If you are certain that the quality scores are already scaled to Sanger Phred+33 (the result of an Illumina 1.8+ pipeline), the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code> can be directly assigned. Click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the ‚ÄúDatatype‚Äù tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>    </ul>  </li>  <li>Run <strong>FastQC</strong> again on the entire dataset <em>if any changes were made to the quality scores</em> for QA.</li></ul>  <p><strong>Other tips</strong></p>  <ul>  <li>If you are not sure what type of FASTQ data you have (maybe it is not Illumina?), see the help directly on the <strong>FASTQ Groomer</strong> tool for information about types.    <ul>      <li>For <em>Illumina</em>, first run <strong>FastQC</strong> on a sample of your data (<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">how to read the full report</a>). The output report will note the quality score type interpreted by the tool. If not <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, run <strong>FASTQ Groomer</strong> on the entire dataset. If <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, just assign the datatype.</li>      <li>For <em>SOLiD</em>, run <strong>NGS: Fastq manipulation ‚Üí AB-SOLID DATA ‚Üí Convert</strong>, to create a <code class="language-plaintext highlighter-rouge">.fastqcssanger</code> dataset. If you have uploaded a color space fastq sequence with quality scores already scaled to Sanger Phred+33 (<code class="language-plaintext highlighter-rouge">.fastqcssanger</code>), first confirm by running <strong>FastQC</strong> on a sample of the data. Then if you want to double-encode the color space into psuedo-nucleotide space (required by certain tools), see the instructions on the tool form <strong>Fastq Manipulation</strong> for the conversion.</li>      <li>If your data is <em>FASTA</em>, but you want to use tools that require FASTQ input, then using the tool <strong>NGS: QC and manipulation ‚Üí Combine FASTA and QUAL</strong>. This tool will create ‚Äúplaceholder‚Äù quality scores that fit your data. On the output, click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the ‚ÄúDatatype‚Äù tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/faqs/galaxy/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-fastq-files-fastq-vs-fastqsanger-vs-"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: FASTQ files: <code class="language-plaintext highlighter-rouge">fastq</code> vs <code class="language-plaintext highlighter-rouge">fastqsanger</code> vs ..</h3>  <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for a more detailed explanation of encoding schemes.</p>  <p>Nowadays, the most commonly used encoding scheme is sanger. In <span class="notranslate">Galaxy</span>, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>  <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>  <p><strong>Tip:</strong> There are 2 <span class="notranslate">Galaxy</span> datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>  <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p></blockquote>




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-fastq-files-fastq-vs-fastqsanger-vs-"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: FASTQ files: <code class="language-plaintext highlighter-rouge">fastq</code> vs <code class="language-plaintext highlighter-rouge">fastqsanger</code> vs ..</h3>  <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for a more detailed explanation of encoding schemes.</p>  <p>Nowadays, the most commonly used encoding scheme is sanger. In <span class="notranslate">Galaxy</span>, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>  <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>  <p><strong>Tip:</strong> There are 2 <span class="notranslate">Galaxy</span> datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>  <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p></blockquote>




<h4 class="faq-area"> Identifying and formatting Tabular Datasets <a href="/training-material/faqs/galaxy/datatypes_tabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Format help for Tabular/<span class="notranslate">BED</span>/Interval Datasets
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-identifying-and-formatting-tabular-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Identifying and formatting Tabular Datasets</h3>  <p>A Tabular datatype is human readable and has tabs separating data columns. Please note that tabular data is different from comma separated data (.csv) and the common datatypes are: <code class="language-plaintext highlighter-rouge">.<span class="notranslate">bed</span></code>, <code class="language-plaintext highlighter-rouge">.gtf</code>, <code class="language-plaintext highlighter-rouge">.interval</code>, or <code class="language-plaintext highlighter-rouge">.txt</code>.</p>  <ol>  <li>Click the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to reach the <strong><em>Edit Attributes</em></strong> form.    <ol>      <li>Change the datatype (3rd tab) and save.</li>      <li>Label columns (1st tab) and save.</li>      <li>Metadata will be assigned, then the dataset can be used.</li>    </ol>  </li>  <li>If the required input is a <span class="notranslate">BED</span> or Interval datatype, adjusting (<code class="language-plaintext highlighter-rouge">.tab</code> ‚Üí <code class="language-plaintext highlighter-rouge">.<span class="notranslate">bed</span></code>, <code class="language-plaintext highlighter-rouge">.tab</code> ‚Üí <code class="language-plaintext highlighter-rouge">.interval</code>) maybe possible using a combination of <strong><em>Text Manipulation</em></strong> tools, to create a dataset that matches required specifications.</li>  <li>Some tools require that <span class="notranslate">BED</span> format be followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.    <ul>      <li>These tools will fail, if they are run with malformed <span class="notranslate">BED</span> datasets or non-specific column assignments.</li>      <li><strong>Solution</strong>: reorganize the data to be in <span class="notranslate">BED</span> format and rerun.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Understanding Datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-datatypes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding Datatypes</h3>  <ul>  <li>Allow <span class="notranslate">Galaxy</span> to detect the datatype during Upload, and adjust from there if needed.</li>  <li>Tool forms will filter for the appropriate datatypes it can use for each input.</li>  <li><a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_change_datatype.html">Directly changing</a> a datatype can lead to errors. Be intentional and consider <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> instead when possible.</li>  <li>Dataset content can also be adjusted (tools: Data manipulation) and the expected <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#detecting-the-datatype-file-format">datatype detected</a>. Detected datatypes are the most reliable in most cases.</li>  <li>If a tool does not accept a dataset as valid input, it is not in the correct format with the correct datatype.</li>  <li>Once a dataset‚Äôs content matches the datatype, and that dataset is repeatedly used (example: Reference annotation) use that same dataset for all steps in an analysis or expect problems. This may mean rerunning prior tools if you need to make a correction.</li>  <li>Tip: Not sure what datatypes a tool is expecting for an input?    <ol>      <li>Create a new empty history</li>      <li>Click on a tool from the tool panel</li>      <li>The tool form will list the accepted datatypes per input</li>    </ol>  </li>  <li><em>Warning</em>: In some cases, tools will transform a dataset to a new datatype at runtime for you.    <ul>      <li>This is generally helpful, and best reserved for smaller datasets.</li>      <li>Why? This can also unexpectedly create hidden datasets that are near duplicates of your original data, only in a different format.</li>      <li>For large data, that can quickly consume working space (quota).</li>      <li>Deleting/purging any hidden datasets can lead to errors if you are still using the original datasets as an input.</li>      <li>Consider <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> to the expected datatype yourself when data is large.</li>      <li>Then test the tool directly on converted data. If it works, purge the original to recover space.</li>    </ul>  </li></ul></blockquote>



 <br> 

<h3 id="diffs"> <a href="#diffs"> Diffs </a></h3>

<hr/>



<h4 class="faq-area"> How to read a Diff <a href="/training-material/topics/admin/faqs/diffs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-read-a-diff"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to read a Diff</h3>  <p>If you haven‚Äôt worked with diffs before, this can be something quite new or different.</p>  <p>If we have two files, let‚Äôs say a grocery list, in two files. We‚Äôll call them ‚Äòa‚Äô and ‚Äòb‚Äô.</p>  <blockquote class="code-2col">  <blockquote class="code-in">    <h3 data-toc-skip id="code-in-old"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Old</h3>    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ cat old<br>üçé<br>üçê<br>üçä<br>üçã<br>üçí<br>ü•ë<br></code></pre></div>    </div>  </blockquote>  <blockquote class="code-out">    <h3 data-toc-skip id="code-out-new"><i class="fas fa-laptop-code" aria-hidden="true"></i><span class="visually-hidden">code-out</span> New</h3>    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ cat new<br>üçé<br>üçê<br>üçä<br>üçã<br>üçç<br>ü•ë<br></code></pre></div>    </div>  </blockquote></blockquote>  <p>We can see that they have some different entries. We‚Äôve removed üçí because they‚Äôre awful, and replaced them with an üçç</p>  <p>Diff lets us compare these files</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff old new<br>5c5<br>&lt; üçí<br><span class="nt">---</span><br><span class="o">&gt;</span> üçç<br></code></pre></div></div>  <p>Here we see that üçí is only in a, and üçç is only in b. But otherwise the files are identical.</p>  <p>There are a couple different formats to diffs, one is the ‚Äòunified diff‚Äô</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff <span class="nt">-U2</span> old new<br><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:06:36.340962616 +0100<br>@@ <span class="nt">-3</span>,4 +3,4 @@<br> üçä<br> üçã<br>-üçí<br>+üçç<br> ü•ë<br></code></pre></div></div>  <p>This is basically what you see in the training materials which gives you a lot of context about the changes:</p>  <ul>  <li><code class="language-plaintext highlighter-rouge">--- old</code> is the ‚Äòold‚Äô file in our view</li>  <li><code class="language-plaintext highlighter-rouge">+++ new</code> is the ‚Äònew‚Äô file</li>  <li>@@ these lines tell us where the change occurs and how many lines are added or removed.</li>  <li>Lines starting with a - are removed from our ‚Äònew‚Äô file</li>  <li>Lines with a + have been added.</li></ul>  <p>So when you go to apply these diffs to your files in the training:</p>  <ol>  <li>Ignore the header</li>  <li>Remove lines starting with - from your file</li>  <li>Add lines starting with + to your file</li></ol>  <p>The other lines (üçä/üçã and ü•ë) above just provide ‚Äúcontext‚Äù, they help you know where a change belongs in a file, but <strong>should not be edited</strong> when you‚Äôre making the above change. Given the above diff, you would find a line with a üçí, and replace it with a üçç</p>  <h4 id="added--removed-lines">Added &amp; Removed Lines</h4>  <p>Removals are very easy to spot, we just have removed lines</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:10:14.370722802 +0100<br>@@ <span class="nt">-4</span>,3 +4,2 @@<br> üçã<br> üçí<br>-ü•ë<br></code></pre></div></div>  <p>And additions likewise are very easy, just add a new line, between the other lines in your file.</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:11:11.422135393 +0100<br>@@ <span class="nt">-1</span>,3 +1,4 @@<br> üçé<br>+üçç<br> üçê<br> üçä<br></code></pre></div></div>  <h4 id="completely-new-files">Completely new files</h4>  <p>Completely new files look a bit different, there the ‚Äúold‚Äù file is <code class="language-plaintext highlighter-rouge">/dev/null</code>, the empty file in a Linux machine.</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff <span class="nt">-U2</span> /dev/null old<br><span class="nt">---</span> /dev/null	2022-02-15 11:47:16.100000270 +0100<br>+++ old	2022-02-16 14:06:19.697132568 +0100<br>@@ <span class="nt">-0</span>,0 +1,6 @@<br>+üçé<br>+üçê<br>+üçä<br>+üçã<br>+üçí<br>+ü•ë<br></code></pre></div></div>  <p>And removed files are similar, except with the new file being /dev/null</p>  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ /dev/null	2022-02-15 11:47:16.100000270 +0100<br>@@ <span class="nt">-1</span>,6 +0,0 @@<br>-üçé<br>-üçê<br>-üçä<br>-üçã<br>-üçí<br>-ü•ë<br></code></pre></div></div></blockquote>



 <br> 

<h3 id="estimation of strandedness"> <a href="#estimation of strandedness"> Estimation of strandedness </a></h3>

<hr/>



<h4 class="faq-area"> When is the "infer experiment" tool used in practice? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-when-is-the-infer-experiment-tool-used-in-practice"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> When is the ‚Äúinfer experiment‚Äù tool used in practice?</h3>  <p>Often you are already aware whether the <span class="notranslate">RNA</span>-seq data is stranded or not in the first place because you sequenced it yourself or ordered it from a company.</p>  <p>But it can happen in cases where you get the data from someone else, that this information is lost and you need to find out.</p></blockquote>




<h4 class="faq-area"> In 'infer experiments' I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-in-infer-experiments-i-get-unequal-numbers-but-in-the-igv-it-looks-like-it-is-unstranded-what-does-this-mean"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> In ‚Äòinfer experiments‚Äô I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean?</h3>  <p>It‚Äôs also often the case that elimination of the second strand is not perfect, and there are genuine cases of bidirectional transcription in the genome. 70 / 30 % as in your report is not a good result for a stranded library. You can treat this as a stranded library in your analysis, but for instance you couldn‚Äôt make the conclusion that a given gene is actually transcribed from the reverse strand. Likely that the library preparation didn‚Äôt work perfectly. This can depend on many factors, one is that you need to completely digest your <span class="notranslate">DNA</span> using a high quality <span class="notranslate">DNa</span>se before doing the reverse transcription.</p></blockquote>



 <br> 

<h3 id="features"> <a href="#features"> Features </a></h3>

<hr/>



<h4 class="faq-area"> Using the Scratchbook to view multiple datasets <a href="/training-material/faqs/galaxy/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-the-scratchbook-to-view-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using the Scratchbook to view multiple datasets</h3>  <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in <span class="notranslate">Galaxy</span>:</p>  <ol>  <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> on the top menu bar.    <ul>      <li>You should see a little checkmark on the icon now</li>    </ul>  </li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> to view the output    <ul>      <li>You should see the output in a window overlayed over <span class="notranslate">Galaxy</span></li>      <li>You can resize this window by dragging the bottom-right corner</li>    </ul>  </li>  <li><strong>Click</strong> outside the file to exit the Scratchbook</li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a second dataset from your history    <ul>      <li>You should now see a second window with the new dataset</li>      <li>This makes it easier to compare the two outputs</li>    </ul>  </li>  <li>Repeat this for as many files as you would like to compare</li>  <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> by clicking on the icon again</li></ol></blockquote>




<h4 class="faq-area"> Uso del cuaderno de apuntes para ver varios conjuntos de datos <a href="/training-material/faqs/galaxy-es/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-uso-del-cuaderno-de-apuntes-para-ver-varios-conjuntos-de-datos"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Uso del cuaderno de apuntes para ver varios conjuntos de datos</h3>  <p>Si deseas ver dos o m√°s conjuntos de datos al mismo tiempo, puedes usar la funci√≥n <strong>Scratchbook</strong> en <span class="notranslate">Galaxy</span>:    1. <strong>Haz clic</strong> en el icono <em>Scratchbook</em> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> en la barra de men√∫ superior.       - Deber√≠a aparecer ver una peque√±a marca de verificaci√≥n en el icono    2. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> un conjunto de datos haciendo clic en el icono de ojo <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> para ver el resultado.       - Deber√≠as ver la salida en una ventana emergente sobre <span class="notranslate">Galaxy</span>       - Puedes cambiar el tama√±o de esta ventana arrastrando la esquina inferior derecha    3. <strong>Haz clic</strong> fuera del archivo para salir del Scratchbook    4. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> un segundo conjunto de datos de tu historial       - Ahora deber√≠as poder ver una segunda ventana con el nuevo conjunto de datos       - Esto hace que sea m√°s f√°cil comparar las dos salidas.    5. Repite estos pasos para todos los archivos que desees comparar.    6. Puedes desactivar <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> haciendo clic en el icono nuevamente.</p></blockquote>




<h4 class="faq-area"> Why not use Excel? <a href="/training-material/faqs/galaxy/why_not_excel.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Excel is a fantastic tool and a great place to build simple analysis models, but when it comes to scaling, <span class="notranslate">Galaxy</span> wins every time.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-not-use-excel"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why not use Excel?</h3>  <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>  <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel won‚Äôt help you here. It doesn‚Äôt track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>  <p><span class="notranslate">Galaxy</span>, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable <span class="notranslate">workflow</span> from your analysis that others (or yourself) can use on other datasets.</p>  <p>Another challenge with sp<span class="notranslate">reads</span>heet programs is that they don‚Äôt scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but you‚Äôll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p></blockquote>



 <br> 

<h3 id="further reading"> <a href="#further reading"> Further reading </a></h3>

<hr/>



<h4 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-read-more-about-this-analysis"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I read more about this analysis?</h3>  <p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p></blockquote>




<h4 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-where-can-i-read-more-about-this-analysis"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Where can I read more about this analysis?</h3>  <p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p></blockquote>



 <br> 

<h3 id="galaxy"> <a href="#galaxy"> <span class="notranslate">Galaxy</span> </a></h3>

<hr/>



<h4 class="faq-area"> How many mules? <a href="/training-material/topics/admin/faqs/galaxy-how-many-mules.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-many-mules"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How many mules?</h3>  <p>Start with 2 and add more as needed. If you notice that your jobs seem to inexplicably sit for a long time before being dispatched to the cluster, or after they have finished on the cluster, you may need additional handlers.</p></blockquote>



 <br> 

<h3 id="galaxy admin interface"> <a href="#galaxy admin interface"> <span class="notranslate">Galaxy</span> admin interface </a></h3>

<hr/>



<h4 class="faq-area"> Install tools via the Admin UI <a href="/training-material/topics/admin/faqs/install_tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-install-tools-via-the-admin-ui"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Install tools via the Admin UI</h3>  <ol>  <li>Open <span class="notranslate">Galaxy</span> in your browser and type `` in the tool search box on the left. If ‚Äú‚Äù is among the search results, you can skip the following steps.</li>  <li>Access the Admin menu from the top bar (you need to be logged-in with an email specified in the <code class="language-plaintext highlighter-rouge">admin_users</code> setting)</li>  <li>Click ‚ÄúInstall and Uninstall‚Äù, which can be found on the left, under ‚ÄúTool Management‚Äù</li>  <li>Enter `` in the search interface</li>  <li>Click on the first hit, having <code class="language-plaintext highlighter-rouge">devteam</code> as owner</li>  <li>Click the ‚ÄúInstall‚Äù button for the latest revision</li>  <li>Enter ‚Äú‚Äù as the target section and click ‚ÄúOK‚Äù.</li></ol></blockquote>



 <br> 

<h3 id="gene"> <a href="#gene"> Gene </a></h3>

<hr/>



<h4 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/transcriptomics/faqs/dimension_reduction.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</h3>  <p>The actual data has tens of thousands of genes, and so tens of thousands of variables to consider. Even after selecting for the most variable genes and the most high quality genes, we can still be left with &gt; 1000 genes. Performing clustering on a dataset with 1000s of variables is possible, but computationally expensive. It is therefore better to perform dimension reduction to reduce the number of variables to a latent representation of these variables. These latent variables are ideally more than 10 but less than 50 to capture the variability in the data to perform clustering upon.</p></blockquote>




<h4 class="faq-area"> What exactly is a ‚ÄòGene profile‚Äô? <a href="/training-material/topics/transcriptomics/faqs/gene_profile.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-exactly-is-a-gene-profile"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What exactly is a ‚ÄòGene profile‚Äô?</h3>  <p>Think of it like a fingerprint that some cells exhibit and others don‚Äôt. It‚Äôs a small collection of genes which are up or down regulated in relation to one another. Their differences are not absolute, but relative. So if CellA has 100 counts of Gene1 and 50 counts of Gene2, this creates a relation of 2:1 between Gene1 and Gene2. If CellB has a 20 counts of Gene1 and 10 counts of Gene2, then they share the same relation. If CellA and CellB share other relations with other genes than this might be enough to say that they share a Gene profile, and will therefore likely cluster together as they describe the same cell type.</p></blockquote>




<h4 class="faq-area"> Why do we only consider highly variable genes? <a href="/training-material/topics/transcriptomics/faqs/variable_genes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-only-consider-highly-variable-genes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we only consider highly variable genes?</h3>  <p>The non-variable genes are likely housekeeping genes, which are expressed everywhere and are not so useful for distinguishing one cell type from another. However background genes are important to the analysis and are used to generate a background baseline model for measuring the variability of the other genes.</p></blockquote>



 <br> 

<h3 id="historias"> <a href="#historias"> Historias </a></h3>

<hr/>



<h4 class="faq-area"> Para la creaci√≥n de un historial nuevo <a href="/training-material/faqs/galaxy-es/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Los historiales son una parte importante de <span class="notranslate">Galaxy</span>, la mayor√≠a de la gente utiliza un historial para cada an√°lisis nuevo. Aseg√∫rate siempre de darle buenos nombres a tus historiales, de tal forma que despu√©s puedas encontrar f√°cilmente tus resultados.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-para-la-creaci√≥n-de-un-historial-nuevo"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Para la creaci√≥n de un historial nuevo</h3>  <p>Haz click sobre el icono <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> en la parte superior del panel de historiales.</p></blockquote>




<h4 class="faq-area"> Cambiando el nombre de un historial <a href="/training-material/faqs/galaxy-es/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-cambiando-el-nombre-de-un-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Cambiando el nombre de un historial</h3>  <ol>  <li>Haz clic sobre  <strong>Unnamed history</strong> (o el nombre que tenga el historial sobre el que est√°s trabajando) (<strong>Haz clic para cambiar el nombre del historial</strong>) en la parte superior de tu panel de historial</li>  <li>Escribe el nombre nuevo</li>  <li>Pulsa <kbd>Enter</kbd></li></ol></blockquote>



 <br> 

<h3 id="histories"> <a href="#histories"> Histories </a></h3>

<hr/>



<h4 class="faq-area"> Copy a dataset between histories <a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Sometimes you may want to use a dataset in multiple histories. You do not need to re-upload the data, but you can copy datasets from one history to another.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-copy-a-dataset-between-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Copy a dataset between histories</h3>  <p>There 3 ways to copy datasets between histories</p>  <ol>  <li>    <p>From the original history</p>    <ol>      <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>      <li>Click on <strong>Copy Dataset</strong></li>      <li>        <p>Select the desired files</p>      </li>      <li>        <p>Give a relevant name to the ‚ÄúNew history‚Äù</p>      </li>      <li>Click on the new history name in the green box that have just appear to switch to this history</li>    </ol>  </li>  <li>    <p>From the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-columns</span> <strong>View all histories</strong></p>    <ol>      <li>Click on <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-columns</span> <strong>View all histories</strong> on the top right</li>      <li>Switch to the history in which the dataset should be copied</li>      <li>Drag the dataset to copy from its original history</li>      <li>Drop it in the target history</li>    </ol>  </li>  <li>    <p>From the target history</p>    <ol>      <li>Click on <strong>User</strong> in the top bar</li>      <li>Click on <strong>Datasets</strong></li>      <li>Search for the dataset to copy</li>      <li>Click on it</li>      <li>Click on <strong>Copy to History</strong></li>    </ol>  </li></ol></blockquote>




<h4 class="faq-area"> Creating a new history <a href="/training-material/faqs/galaxy/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Histories are an important part of <span class="notranslate">Galaxy</span>, most people use a new history for every new analysis. Always make sure to give your histories good names, so you can easily find your results back later.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>




<h4 class="faq-area"> Downloading histories <a href="/training-material/faqs/galaxy/histories_download_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading histories</h3>  <ol>  <li>Click on the gear icon <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> on the top of the history panel.</li>  <li>Select ‚ÄúExport History to File‚Äù from the History menu.</li>  <li>Click on the ‚ÄúClick here to generate a new archive for this history‚Äù text.</li>  <li>Wait for the <span class="notranslate">Galaxy</span> server to prepare history for download.</li>  <li>Click on the generated link to download the history.</li></ol></blockquote>




<h4 class="faq-area"> Find all Histories and purge (aka permanently delete) <a href="/training-material/faqs/galaxy/histories_find_purge.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-find-all-histories-and-purge-aka-permanently-delete"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Find all Histories and purge (aka permanently delete)</h3>  <ol>  <li>Login to your <span class="notranslate">Galaxy</span> account.</li>  <li>On the top navigation bar Click on <strong>User</strong>.</li>  <li>On the drop down menu that appears Click on <strong>Histories</strong>.</li>  <li>Click on <strong>Advanced Search</strong>, additional fields will be displayed.</li>  <li>Next to the <strong>Status</strong> field, click <strong>All</strong>, a list of all histories will be displayed.</li>  <li>Check the box next to <strong>Name</strong> in the displayed list to select all histories.</li>  <li>Click <strong>Delete Permanently</strong> to purge all histories.</li>  <li>A pop up dialogue box will appear letting you know history contents will be removed and cannot be undone, then click <strong>OK</strong> to confirm.</li></ol></blockquote>




<h4 class="faq-area"> Finding Histories <a href="/training-material/faqs/galaxy/histories_finding_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding Histories</h3>  <ol>  <li>To review all histories in your account, go to <strong>User &gt; Histories</strong> in the top menu bar.</li>  <li>At the top of the History listing, click on <strong>Advanced Search</strong>.</li>  <li>Set the status to all to view all of your active, deleted, and permanently deleted (purged) histories.</li>  <li>Histories in all states are listed for registered accounts. Meaning one will always find their data here if it ever appears to be ‚Äúlost‚Äù.</li>  <li>Note: Permanently deleted (purged) Histories may be fully removed from the server at any time. The data content inside the History is always removed at the time of purging (by a double-confirmed user action), but the purged History artifact may still be in the listing. Purged data content cannot be restored, even by an administrator.</li></ol></blockquote>




<h4 class="faq-area"> Importing a history <a href="/training-material/faqs/galaxy/histories_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a history</h3>  <ol>  <li>Open the link to the shared history</li>  <li>Click on the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>Import history</em> button on the top right</li>  <li>Enter a title for the new history</li>  <li>Click on <strong>Import</strong></li></ol></blockquote>




<h4 class="faq-area"> Renaming a history <a href="/training-material/faqs/galaxy/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>




<h4 class="faq-area"> Searching your history <a href="/training-material/faqs/galaxy/histories_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-searching-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Searching your history</h3>  <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>  <ol>  <li>    <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>    <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>  </li>  <li><strong>Type</strong> a search term in this box    <ul>      <li>For example a tool name, or sample name</li>    </ul>  </li>  <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-clear</span> next to the search box</li></ol></blockquote>




<h4 class="faq-area"> Finding and working with "Histories shared with me" <a href="/training-material/faqs/galaxy/histories_shared.html"><i class="fas fa-external-link-alt"></i></a> </h4>
How to find and work on histories shared with you
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-and-working-with-histories-shared-with-me"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding and working with ‚ÄúHistories shared with me‚Äù</h3>  <p><strong>To find histories shared with me:</strong></p>  <ol>  <li>Log into your account.</li>  <li>Select <strong>User</strong>, in the drop-down menu, select <strong>Histories shared with me</strong>.</li></ol>  <p><strong>To work with shared histories:</strong></p>  <ul>  <li>Import the History into your account via copying it to work with it.</li>  <li>Unshare Histories that you no longer want shared with you or that you have already made a copy of.</li></ul>  <p><strong>Note:</strong> Shared Histories (when copied into your account or not) do count in portion toward your total account data quota usage. More details on histories shared concerning account quota usage can be found in <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#unsharing-unwanted-histories">this link</a>.</p></blockquote>




<h4 class="faq-area"> Sharing your History <a href="/training-material/faqs/galaxy/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can share your work in <span class="notranslate">Galaxy</span>. There are various ways you can give access one of your histories to other users.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sharing-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sharing your History</h3>  <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>  <ol>  <li><strong>Share via link</strong>    <ul>      <li>Open the <strong>History Options</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> menu (gear icon) at the top of your history panel        <ul>          <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History accessible</strong></li>          <li>A <strong>Share Link</strong> will appear that you give to others</li>        </ul>      </li>      <li>Anybody who has this link can view and copy your history</li>    </ul>  </li>  <li><strong>Publish your history</strong>    <ul>      <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>      <li>Anybody on this <span class="notranslate">Galaxy</span> server will see your history listed under the <em>Shared Data</em> menu</li>    </ul>  </li>  <li><strong>Share only with another user</strong>.    <ul>      <li>Click the <strong>Share with a user</strong> button at the bottom</li>      <li>Enter an email address for the user you want to share with</li>      <li>Your history will be shared only with this user.</li>    </ul>  </li>  <li><strong>Finding histories others have shared with me</strong>    <ul>      <li>Click on <strong>User</strong> menu on the top bar</li>      <li>Select <strong>Histories shared with me</strong></li>      <li>Here you will see all the histories others have shared with you directly</li>    </ul>  </li></ol>  <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <em>History options</em> icon in your history and clicking <em>Copy</em></p></blockquote>




<h4 class="faq-area"> Compartiendo un historial <a href="/training-material/faqs/galaxy-es/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Puedes compartir tu trabajo en <span class="notranslate">Galaxy</span>. Hay varias formas de dar acceso a tus historiales a otros usuarios.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-compartiendo-un-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Compartiendo un historial</h3>  <p>Compartir tu historial permite a otros importar y acceder a los conjuntos de datos, par√°metros y pasos de tu historial.</p>  <ol>  <li><strong>Compartir a trav√©s de un enlace</strong>    <ul>      <li>Abre el men√∫ <strong>Opciones de historial</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (icono de engranaje) en la parte superior del panel de historial        <ul>          <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Hacer que el historial sea accesible</strong></li>          <li>Aparecer√° un <strong>Compartir enlace</strong> que puedes dar a otros usuarios.</li>        </ul>      </li>      <li>Cualquiera que tenga este enlace puede ver y copiar tu historial.</li>    </ul>  </li>  <li><strong>Publica tu historial</strong>    <ul>      <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Hacer que el historial est√© disponible p√∫blicamente en Historias publicadas</strong></li>      <li>Cualquiera en este servidor <span class="notranslate">Galaxy</span> podr√° ver tu historial en el men√∫ <em>Datos compartidos</em></li>    </ul>  </li>  <li><strong>Comparte solo con otro usuario</strong>.    <ul>      <li>Haz clic en el bot√≥n <strong>Compartir con un usuario</strong> en la parte inferior</li>      <li>Ingresa una direcci√≥n de correo electr√≥nico del usuario con el que deseas compartir</li>      <li>Tu historial se compartir√° solo con este usuario.</li>    </ul>  </li>  <li><strong>Encontrar historiales que otros han compartido conmigo</strong>    <ul>      <li>Haz clic en el men√∫ <strong>Usuario</strong> en la barra superior</li>      <li>Selecciona <strong>Historiales compartidos conmigo</strong></li>      <li>Aqu√≠ ver√°s todos los historiales que otros han compartido contigo directamente** Nota: ** Si deseas realizar cambios en tu historial sin afectar la versi√≥n compartida, crea una copia mediante al √≠cono <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <em>Opciones de historial</em> en tu historial y haciendo clic en <em>Copiar</em></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Transfer entire histories from one <span class="notranslate">Galaxy</span> server to another <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-transfer-entire-histories-from-one-galaxy-server-to-another"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Transfer entire histories from one <span class="notranslate">Galaxy</span> server to another</h3>  <ol>  <li>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> in the history panel of the <em>sender</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Export to File</strong></li>  <li>Select either exporting history <strong>to a link</strong> or <strong>to a remote file</strong></li>  <li>Click on the link text to generate a new archive for the history <em>if</em> exporting to a link</li>  <li>Wait for the link to generate</li>  <li>Copy the link address or click on the generated link to download the history archive</li>  <li>Click on <strong>User</strong> on the top menu of the <em>receiver</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Histories</strong> to view saved histories</li>  <li>Click on <strong>Import history</strong> in the grey button on the top right</li>  <li>Select the appropriate importing method based on the choices made in steps 3 and 6    <ul>      <li>Choose <strong>Export URL from another <span class="notranslate">galaxy</span> instance</strong> if link address was copied in step 6</li>      <li>Select <strong>Upload local file from your computer</strong> if history archive was downloaded in step 6</li>      <li>Choose <strong>Select a remote file</strong> if history was exported to a remote file in step 3</li>    </ul>  </li>  <li>Click the link text to check out your histories if import is successful</li></ol>  <p>If history being transferred is too large, you may:</p>  <ol>  <li>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> in the history panel of the <em>sender</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click <strong>Copy Datasets</strong> to move just the important datasets into a new history</li>  <li>Create the archive from that smaller history</li></ol></blockquote>




<h4 class="faq-area"> Undeleting history <a href="/training-material/faqs/galaxy/histories_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Undelete your deleted histories
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-undeleting-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Undeleting history</h3>  <ul>  <li>Click on <strong>User</strong> then select <strong>Histories</strong></li>  <li>Click on <strong>Advanced search</strong> on the top left side below  <strong>Saved Histories</strong></li>  <li>On <strong>Status</strong> click <strong>Deleted</strong></li>  <li>Select the history you want to undelete using the <strong>checkbox</strong> on the left side</li>  <li>Click <strong>Undelete</strong> button below the deleted histories</li></ul></blockquote>




<h4 class="faq-area"> Unsharing unwanted histories <a href="/training-material/faqs/galaxy/histories_unsharing_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-unsharing-unwanted-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Unsharing unwanted histories</h3>  <ul>  <li>All account Histories owned by others but shared with you can be reviewed under <strong>User &gt; Histories shared with me</strong>.</li>  <li>The other person does not need to unshare a history with you. <em>Unshare histories yourself</em> on this page using the pull-down menu per history.</li>  <li>Dataset and History privacy options, including sharing, can be set under <strong>User &gt; Preferences</strong>.</li></ul>  <p>Three key features to work with shared data are:</p>  <ul>  <li><strong>View</strong> is a review feature. The data cannot be worked with, but many details, including tool and dataset metadata/parameters, are included.</li>  <li><strong>Copy</strong> those you want to work with. This will increase your quota usage. This will also allow you to manipulate the datasets or the history independently from the original owner. All History/Dataset functions are available if the other person granted full access to the datasets to you.</li>  <li><strong>Unshare</strong> any on the list not needed anymore. After a history is copied, you will still have your version of the history, even if later unshared or the other person who shared it with you changes their version later. Meaning, that <em>each account‚Äôs version of a History and the Datasets in it are distinct</em> (unless the Datasets were not shared, you will still only be able to ‚Äúview‚Äù but not work with or download them).</li></ul>  <p><strong>Note:</strong> ‚ÄúHistories shared with me‚Äù result in only a tiny part of your quota usage. Unsharing will not significantly reduce quota usage unless hundreds (or more!) or many significant histories are shared. If you share a History with someone else, that does not increase or decrease your quota usage.</p></blockquote>



 <br> 

<h3 id="history"> <a href="#history"> History </a></h3>

<hr/>



<h4 class="faq-area"> My jobs are not running / I cannot see the history overview menu <a href="/training-material/topics/introduction/faqs/login.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-jobs-are-not-running--i-cannot-see-the-history-overview-menu"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My jobs are not running / I cannot see the history overview menu</h3>  <p>Please make sure you are logged in. At the top menu bar, you should see a section labeled ‚ÄúUser‚Äù. If you see ‚ÄúLogin/Register‚Äù here you are not logged in.</p></blockquote>



 <br> 

<h3 id="igv"> <a href="#igv"> Igv </a></h3>

<hr/>



<h4 class="faq-area"> Add genome and annotations to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-genome-and-annotations-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add genome and annotations to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Expand the FASTA dataset with the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>  <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>  <li>    <p>Close the window</p>    <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>  </li>  <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The annotation track should appear. Be careful that all files have the same genome ID</p>  </li></ol></blockquote>




<h4 class="faq-area"> Add genome and annotations to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/visualisation/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-genome-and-annotations-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add genome and annotations to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Expand the FASTA dataset with the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>  <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>  <li>    <p>Close the window</p>    <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>  </li>  <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The annotation track should appear. Be careful that all files have the same genome ID</p>  </li></ol></blockquote>




<h4 class="faq-area"> Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-mapped-reads-track-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Check if the reference genome is available on the IGV instance</li>  <li>Expand the BAM dataset with the mapped <span class="notranslate">reads</span> in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The mapped <span class="notranslate">reads</span> track should appear. Be sure that all files have the same genome ID</p>  </li></ol></blockquote>




<h4 class="faq-area"> Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/visualisation/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-mapped-reads-track-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Check if the reference genome is available on the IGV instance</li>  <li>Expand the BAM dataset with the mapped <span class="notranslate">reads</span> in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The mapped <span class="notranslate">reads</span> track should appear. Be sure that all files have the same genome ID</p>  </li></ol></blockquote>



 <br> 

<h3 id="inputs"> <a href="#inputs"> Inputs </a></h3>

<hr/>



<h4 class="faq-area"> Do I need to create collections to run MaxQuant analysis or can I use single sample inputs? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-do-i-need-to-create-collections-to-run-maxquant-analysis-or-can-i-use-single-sample-inputs"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Do I need to create collections to run MaxQuant analysis or can I use single sample inputs?</h3>  <p>Collections are not necessary to run MaxQuant but they make the history more clean and easier to navigate. The multiple datasets options allows to select multiple files that are not part of a collection and will give the same result as with a collection as input.</p></blockquote>




<h4 class="faq-area"> Do we need a contaminant FASTA for MQ in <span class="notranslate">galaxy</span>? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-do-we-need-a-contaminant-fasta-for-mq-in-galaxy"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Do we need a contaminant FASTA for MQ in <span class="notranslate">galaxy</span>?</h3>  <p>Normally MaxQuant has a default contaminant fasta that we don‚Äôt have to input ourselves.MaxQuant in <span class="notranslate">galaxy</span> comes with the option to add contaminants automatically (one does not need to add contaminants to the fasta file)</p></blockquote>




<h4 class="faq-area"> Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant. <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-do-you-need-to-merge-the-databases-because-you-can-select-multiple-fasta-files-in-maxquant"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant.</h3>  <p>For MaxQuant one does not need to merge the databases, also MaxQuant offers the function to add common contaminants to the provided fasta.</p></blockquote>



 <br> 

<h3 id="instructors"> <a href="#instructors"> Instructors </a></h3>

<hr/>



<h4 class="faq-area"> What are the best practices for teaching with <span class="notranslate">Galaxy</span>? <a href="/training-material/faqs/gtn/instructors_best_practices.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-are-the-best-practices-for-teaching-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What are the best practices for teaching with <span class="notranslate">Galaxy</span>?</h3>  <p>We started to collect some best practices for instructors inside our <a href="/training-material/topics/instructors/">Good practices slides</a></p></blockquote>




<h4 class="faq-area"> What <span class="notranslate">Galaxy</span> instance should I use for my training? <a href="/training-material/faqs/gtn/instructors_choosing_server.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-galaxy-instance-should-i-use-for-my-training"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What <span class="notranslate">Galaxy</span> instance should I use for my training?</h3>  <p>To teach the hands-on tutorials you need a <span class="notranslate">Galaxy</span> server to run the examples on.</p>  <p>Each tutorial is annotated with the information on which <a href="https://galaxyproject.org/public-galaxy-servers/">public <span class="notranslate">Galaxy</span> servers</a> it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server‚Äôs admins to confirm that the server can handle the workload for a workshop. For example, the <a href="https://usegalaxy.eu/">use<span class="notranslate">galaxy</span>.eu</a></p>  <p>If your organization/consortia/community has its own <span class="notranslate">Galaxy</span> server, then you may want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same as your participants will be using after the training. They‚Äôll also be able to revisit any analysis they did during the training. If you pursue this option you‚Äôll need to work with your organization‚Äôs <span class="notranslate">Galaxy</span> Admins to confirm that</p>  <ul>  <li>the server can support a room full of people all doing the same analysis at the same time.</li>  <li>all tools and reference datasets needed in the tutorial are locally installed.  To learn how to setup a <span class="notranslate">Galaxy</span> instance for a tutorial, you can follow our <a href="/training-material/topics/instructors/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</li>  <li>all participants will be able to create/use accounts on the system.</li></ul>  <p>Some training topics have a Docker image that can be installed and run on all participants‚Äô laptops.  These images contain <span class="notranslate">Galaxy</span> instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.</p>  <p>Finally, you can also run your tutorials on cloud-based infrastructures.  <span class="notranslate">Galaxy</span> is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://launch.genome.edu.au/launch">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.</p></blockquote>




<h4 class="faq-area"> How do I get help? <a href="/training-material/faqs/gtn/instructors_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-do-i-get-help"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How do I get help?</h3>  <p>The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the <span class="notranslate">Galaxy</span> Training Network <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>. Anyone can view the discussion, but you‚Äôll need to login (using your GitHub or Twitter account) to add to the discussion.</p>  <p>If you have questions about <span class="notranslate">Galaxy</span> in general (that are not training-centric) then there are <a href="https://galaxyproject.org/support/">several support options</a>.</p></blockquote>




<h4 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/instructors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-do-i-start"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where do I start?</h3>  <p>Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.</p></blockquote>



 <br> 

<h3 id="interactive tools"> <a href="#interactive tools"> Interactive tools </a></h3>

<hr/>



<h4 class="faq-area"> Launch JupyterLab <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch JupyterLab</h3>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-jupyterlab-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch JupyterLab in <span class="notranslate">Galaxy</span></h3>  <p>Currently JupyterLab in <span class="notranslate">Galaxy</span> is available on <a href="https://live.usegalaxy.eu">Live.use<span class="notranslate">Galaxy</span>.eu</a>, <a href="https://usegalaxy.org">use<span class="notranslate">galaxy</span>.org</a> and <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a>.</p>  <blockquote class="notranslate hands_on">    <h3 data-toc-skip id="hands_on-hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run JupyterLab</h3>    <ol>      <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Tested with interactive_tool_jupyter_notebook"><strong>Interactive Jupyter Notebook</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_jupyter_notebook</span></span> :</li>      <li>Click Execute</li>      <li>The tool will start running and will stay running permanently</li>      <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>      <li>Click on your JupyterLab instance</li>    </ol>  </blockquote></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-try-jupyterlab-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch Try JupyterLab if not available on <span class="notranslate">Galaxy</span></h3>  <p>If JupyterLab is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Open a Terminal in Jupyter <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-open-a-terminal-in-jupyter"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Open a Terminal in Jupyter</h3>  <p>This tutorial will let you accomplish almost everything from this view, running code in the cells below directly in the training material. You can choose between running the code here, or opening up a terminal tab in which to run it.Here are some instructions for how to do this on various environments.</p>  <p><strong>Jupyter</strong> on Use<span class="notranslate">Galaxy</span>.* and <strong>MyBinder.org</strong></p>  <ol>  <li>    <p>Use the File ‚Üí New ‚Üí Terminal menu to launch a terminal.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-launch.png" alt="screenshot of jupyterlab showing the File menu expanded to show new and terminal option." /></p>  </li>  <li>    <p>Disable ‚ÄúSimple‚Äù mode in the bottom left hand corner, if it activated.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-simple.png" alt="screenshot of jupyterlab showing a toggle labelled simple" /></p>  </li>  <li>    <p>Drag one of the terminal or notebook tabs to the side to have the training materials and terminal side-by-side</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-final.png" alt="screenshot of jupyterlab with notebook and terminal side-by-side." /></p>  </li></ol>  <p><strong>CoCalc</strong></p>  <ol>  <li>    <p>Use the <em>Split View</em> functionality of cocalc to split your view into two portions.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-split.png" alt="screenshot of cocalc button to split views" /></p>  </li>  <li>    <p>Change the view of one panel to a terminal</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-terminal.png" alt="screenshot of cocalc swapping view port to that of a terminal" /></p>  </li></ol></blockquote>




<h4 class="faq-area"> Open interactive tool <a href="/training-material/faqs/galaxy/interactive_tools_open.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-open-interactive-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Open interactive tool</h3>  <ol>  <li>Go to User &gt; Active InteractiveTools</li>  <li>Wait for the  to be running (Job Info)</li>  <li>Click on</li></ol></blockquote>




<h4 class="faq-area"> Knitting RMarkdown documents in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-knitting-rmarkdown-documents-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Knitting RMarkdown documents in RStudio</h3>  <p>One of the other nice features of RMarkdown documents is making lovely presentation-quality worthy documents. You can take, for example, a tutorial and produce a nice report like output as HTML, PDF, or <code class="language-plaintext highlighter-rouge">.doc</code> document that can easily be shared with colleagues or students.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-meta-output.png" alt="Screenshot of the metadata with html_notebook and word_document being visible and a number of options controlling their output. TOC, standing for table of contents, has been set to true for both." /></p>  <p>Now you‚Äôre <strong>ready to preview the document</strong>:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of preview dropdown with options like preview, knit to html, knit to pdf, knit to word" /></p>  <p><strong>Click Preview</strong>. A window will popup with a preview of the rendered verison of this document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of rendered document with the table of contents on left, title is in a large font, and there are coloured boxes similar to GTN tutorials offering tips and more information" /></p>  <p>The preview is really similar to the GTN rendering, no cells have been executed, and no output is embedded yet in the preview document. But if you have run cells (e.g. the first few loading a library and previewing the <code class="language-plaintext highlighter-rouge">msleep</code> dataset:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview-output.png" alt="screenshot of the rendered document with a fancy table browser embedded as well as the output of each step" /></p>  <p><strong>When you‚Äôre ready to distribute</strong> the document, you can instead use the <code class="language-plaintext highlighter-rouge">Knit</code> button. This runs every cell in the entire document fresh, and then compiles the outputs together with the rendered markdown to produce a nice result file as HTML, PDF, or Word document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-knit.png" alt="screenshot of the console with 'chunks' being knitted together" /></p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-pdf--word-require-a-latex-installation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: PDF + Word require a LaTeX installation</h3>  <p>You might need to install additional packages to compile the PDF and Word document versions</p></blockquote>  <p>And at the end you can see a pretty document rendered with all of the output of every step along the way. This is a fantastic way to e.g. distribute read-only lesson materials to students, if you feel they might struggle with using an RMarkdown document, or just want to read the output without <em>doing</em> it themselves.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-pdf.png" alt="screenshot of a PDF document showing the end of the tutorial where a pretty plot has been rendered and there is some text for conclusions and citations" /></p></blockquote>




<h4 class="faq-area"> Launch RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch RStudio</h3>  <p>Depending on which server you are using, you may be able to run RStudio directly in <span class="notranslate">Galaxy</span>. If that is not available, RStudio Cloud can be an alte<span class="notranslate">rna</span>tive.</p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>  <p>Currently RStudio in <span class="notranslate">Galaxy</span> is only available on <a href="https://usegalaxy.eu">Use<span class="notranslate">Galaxy</span>.eu</a> and <a href="https://usegalaxy.org">Use<span class="notranslate">Galaxy</span>.org</a></p>  <ol>    <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>    <li>Click Execute</li>    <li>The tool will start running and will stay running permanently</li>    <li>Click on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started.</li>  </ol></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>  <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>    <li>Create a new project</li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Launch RStudio <a href="/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch RStudio</h3>  <p>Depending on which server you are using, you may be able to run RStudio directly in <span class="notranslate">Galaxy</span>. If that is not available, RStudio Cloud can be an alte<span class="notranslate">rna</span>tive.</p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>  <p>Currently RStudio in <span class="notranslate">Galaxy</span> is only available on <a href="https://usegalaxy.eu">Use<span class="notranslate">Galaxy</span>.eu</a> and <a href="https://usegalaxy.org">Use<span class="notranslate">Galaxy</span>.org</a></p>  <ol>    <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>    <li>Click Execute</li>    <li>The tool will start running and will stay running permanently</li>    <li>Click on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started.</li>  </ol></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>  <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>    <li>Create a new project</li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Launch RStudio <a href="/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch RStudio</h3>  <p>Depending on which server you are using, you may be able to run RStudio directly in <span class="notranslate">Galaxy</span>. If that is not available, RStudio Cloud can be an alte<span class="notranslate">rna</span>tive.</p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>  <p>Currently RStudio in <span class="notranslate">Galaxy</span> is only available on <a href="https://usegalaxy.eu">Use<span class="notranslate">Galaxy</span>.eu</a> and <a href="https://usegalaxy.org">Use<span class="notranslate">Galaxy</span>.org</a></p>  <ol>    <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>    <li>Click Execute</li>    <li>The tool will start running and will stay running permanently</li>    <li>Click on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started.</li>  </ol></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>  <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>    <li>Create a new project</li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Learning with RMarkdown in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-learning-with-rmarkdown-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Learning with RMarkdown in RStudio</h3>  <p>Learning with RMarkdown is a bit different than you might be used to. Instead of copying and pasting code from the GTN into a document you‚Äôll instead be able to run the code directly as it was written, inside RStudio! You can now focus just on the code and reading within RStudio.</p>  <ol>  <li>    <p><strong>Load the notebook</strong> if you have not already, following the tip box at the top of the tutorial</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-download-file.png" alt="Screenshot of the Console in RStudio. There are three lines visible of not-yet-run R code with the download.file statements which were included in the setup tip box." /></p>  </li>  <li>    <p><strong>Open it</strong> by clicking on the <code class="language-plaintext highlighter-rouge">.Rmd</code> file in the file browser (bottom right)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-file-menu.png" alt="Screenshot of Files tab in RStudio, here there are three files listed, a data-science-r-dplyr.Rmd file, a css and a bib file." /></p>  </li>  <li>    <p>The RMarkdown document will appear in the document viewer (top left)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-document-view.png" alt="Screenshot of an open document in RStudio. There is some yaml metadata above the tutorial showing the title of the tutorial." /></p>  </li></ol>  <p>You‚Äôre now ready to view the RMarkdown notebook! Each notebook starts with a lot of metadata about how to build the notebook for viewing, but you can ignore this for now and <strong>scroll down to the content of the tutorial</strong>.</p>  <p>You‚Äôll see codeblocks scattered throughout the text, and these are all runnable snippets that appear like this in the document:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-run-cell.png" alt="Screenshot of the RMarkdown document in the viewer, a cell is visible between markdown text reading library tidyverse. It is slightly more grey than the background region, and it has a run button at the right of the cell in a contextual menu." /></p>  <p>And you have a few options for how to run them:</p>  <ol>  <li>Click the green arrow</li>  <li><kbd>ctrl+enter</kbd></li>  <li>    <p>Using the menu at the top to run all</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-run-all.png" alt="Screenshot of the run dropdown menu in R, the first item is run selected lines showing the mentioned shortcut above, the second is run next chunk, and then it also mentions a 'run all chunks below' and 'restart r and run all chunks' option." /></p>  </li></ol>  <p>When you run cells, the output will appear below in the Console. RStudio essentially copies the code from the RMarkdown document, to the console, and runs it, just as if you had typed it out yourself!</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-cell-output.png" alt="Screenshot of a run cell, its output is included below in the RMarkdown document and the same output is visible below in the console. It shows a log of loading the tidyverse library." /></p>  <p>One of the best features of RMarkdown documents is that they include a very nice table browser which makes previewing results a lot easier! Instead of needing to use <code class="language-plaintext highlighter-rouge">head</code> every time to preview the result, you get an interactive table browser for any step which outputs a table.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-table.png" alt="Screenshot of the table browser. Below a code chunk is a large white area with two images, the first reading 'r console' and the second reading 'tbl_df'. The tbl_df is highlighted like it is active. Below that is a pretty-printed table with bold column headers like name and genus and so on. At the right of the table is a small arrow indicating you can switch to seeing more columns than just the initial three. At the bottom of the table is 1-10 of 83 rows written, and buttons for switching between each page of results." /></p></blockquote>




<h4 class="faq-area"> Stop RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Stop RStudio</h3>  <p>When you have finished your R analysis, it‚Äôs time to stop RStudio.</p>  <ol>  <li>First, save your work into <span class="notranslate">Galaxy</span>, to ensure reproducibility:    <ol>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>    </ol>  </li>  <li>Once you have saved your data, you can proceed in 2 different ways:    <ul>      <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a ‚Äúin progress state‚Äù, so yellow, OR</li>      <li>Clicking on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the ‚ÄúStop‚Äù button at the bottom.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Stop RStudio <a href="/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Stop RStudio</h3>  <p>When you have finished your R analysis, it‚Äôs time to stop RStudio.</p>  <ol>  <li>First, save your work into <span class="notranslate">Galaxy</span>, to ensure reproducibility:    <ol>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>    </ol>  </li>  <li>Once you have saved your data, you can proceed in 2 different ways:    <ul>      <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a ‚Äúin progress state‚Äù, so yellow, OR</li>      <li>Clicking on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the ‚ÄúStop‚Äù button at the bottom.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Stop RStudio <a href="/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Stop RStudio</h3>  <p>When you have finished your R analysis, it‚Äôs time to stop RStudio.</p>  <ol>  <li>First, save your work into <span class="notranslate">Galaxy</span>, to ensure reproducibility:    <ol>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>    </ol>  </li>  <li>Once you have saved your data, you can proceed in 2 different ways:    <ul>      <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a ‚Äúin progress state‚Äù, so yellow, OR</li>      <li>Clicking on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the ‚ÄúStop‚Äù button at the bottom.</li>    </ul>  </li></ol></blockquote>



 <br> 

<h3 id="introduction"> <a href="#introduction"> Introduction </a></h3>

<hr/>



<h4 class="faq-area"> Ways to use <span class="notranslate">Galaxy</span> <a href="/training-material/faqs/gtn/galaxy_usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-ways-to-use-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Ways to use <span class="notranslate">Galaxy</span></h3>  <p>All ways to use <span class="notranslate">Galaxy</span> are included in the <a href="https://galaxyproject.org/use"><span class="notranslate">Galaxy</span> Directory listing</a>.</p>  <p>Having one account at several public <span class="notranslate">Galaxy</span> servers expands your access to distinct data storage and computational resources, plus common and domain-specific analysis tools.</p>  <p>When running your own private <span class="notranslate">Galaxy</span> server for routine analysis, publishing results at a public <span class="notranslate">Galaxy</span> server allows for worldwide access by others when you share your data: Histories, Workflows, and related assets.</p>  <p>Tips:</p>  <ul>  <li><strong>Teaching with <span class="notranslate">Galaxy</span></strong> We strongly recommend using <span class="notranslate">Galaxy</span>‚Äôs Training Infrastructure as a Service (<a href="https://usegalaxy-eu.github.io/tiaas">TIaaS</a>) for synchronous class work.</li>  <li><strong>Public <span class="notranslate">Galaxy</span></strong> servers are appropriate for many analysis projects or for when sharing data or results publicly is a goal. These are also a great choice when learning on your own with <a href="https://training.galaxyproject.org/">GTN</a> tutorials.</li>  <li><strong>Private <span class="notranslate">Galaxy</span></strong> servers are more appropriate when working with very large data, time sensitive projects, and ongoing research projects that require more resources than the public <span class="notranslate">Galaxy</span> servers can support. These two options are scientist friendly as they require very little to no server administration.    <ul>      <li><a href="https://galaxyproject.org/use/aws/">GVL Cloudman</a> is a single or multi-user choice and AWS <a href="https://aws.amazon.com/grants/">offers grants</a>.</li>      <li><a href="https://galaxyproject.org/use/anvil/">AnVIL</a> is a single-user choice sponsored by NHGRI and is a pay-for-use Google Cloud platform.</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> What is <span class="notranslate">Galaxy</span>? <a href="/training-material/faqs/gtn/galaxy_what_is_it.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-is-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What is <span class="notranslate">Galaxy</span>?</h3>  <p><a href="https://galaxyproject.org/"><span class="notranslate">Galaxy</span></a> is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.</p></blockquote>




<h4 class="faq-area"> How can I load data? <a href="/training-material/faqs/gtn/get_data_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-load-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I load data?</h3>  <ul>  <li>Load by ‚Äúbrowsing‚Äù for a local file. Some servers will support load data that is 2 GB or larger. If you are having problems with this method, try FTP.</li>  <li>Load using an <strong>HTTP URL</strong> or <strong>FTP URL</strong>.</li>  <li>Load a few lines of plain text.</li>  <li>Load using FTP. Either line command or with a desktop client.</li></ul></blockquote>




<h4 class="faq-area"> What is this website? <a href="/training-material/faqs/gtn/gtn-what-is-it.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-is-this-website"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What is this website?</h3>  <p>This website is a collection of hands-on tutorials that are designed to be interactive and are built around <span class="notranslate">Galaxy</span>:</p>  <p><img src="/training-material/shared/images/interactive_training.png" alt="Interactive training" title="The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance" /></p>  <p>This material is developed and maintained by the <a href="https://galaxyproject.org/">worldwide <span class="notranslate">Galaxy</span> community</a>. You can learn more about this effort by reading our <a href="https://doi.org/10.1016/j.cels.2018.05.012">article</a>.</p></blockquote>




<h4 class="faq-area"> How can I advertise the training materials on my posters? <a href="/training-material/faqs/gtn/gtn_advertise.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-advertise-the-training-materials-on-my-posters"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I advertise the training materials on my posters?</h3>  <p>We provide some QR codes and logos in the <a href="https://github.com/galaxyproject/training-material/tree/main/assets/images">images folder</a>.</p></blockquote>




<h4 class="faq-area"> What audiences are the tutorials for? <a href="/training-material/faqs/gtn/gtn_audience.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-audiences-are-the-tutorials-for"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What audiences are the tutorials for?</h3>  <p>There are two distinct audiences for these materials.</p>  <ol>  <li><strong>Self-paced individual learners.</strong> These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.</li>  <li><strong>Instructors.</strong> They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.</li></ol></blockquote>




<h4 class="faq-area"> How can I cite the GTN? <a href="/training-material/faqs/gtn/gtn_citing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-cite-the-gtn"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I cite the GTN?</h3>  <p>We wrote an <a href="https://doi.org/10.1016/j.cels.2018.05.012">article</a> about our efforts.</p>  <p>To cite individual tutorials, please find citation information at the end of the tutorial.</p></blockquote>




<h4 class="faq-area"> How is the content licensed? <a href="/training-material/faqs/gtn/gtn_license.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-is-the-content-licensed"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How is the content licensed?</h3>  <p>The content of this website is licensed under the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 License</a>.</p></blockquote>




<h4 class="faq-area"> What are the tutorials for? <a href="/training-material/faqs/gtn/gtn_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-are-the-tutorials-for"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What are the tutorials for?</h3>  <p>These tutorials can be used for learning and teaching how to use <span class="notranslate">Galaxy</span> for general data analysis, and for learning/teaching specific domains such as assembly and differential gene <span class="notranslate">expression</span> analysis with <span class="notranslate">RNA</span>-Seq data.</p></blockquote>



 <br> 

<h3 id="learners"> <a href="#learners"> Learners </a></h3>

<hr/>



<h4 class="faq-area"> How can I get help? <a href="/training-material/faqs/gtn/learners_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-get-help"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I get help?</h3>  <p>If you have questions about this training material, you can reach us using the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>.  You‚Äôll need a <a href="https://github.com/">GitHub</a> or <a href="https://twitter.com/">Twitter</a> account to post questions.  If you have questions about <span class="notranslate">Galaxy</span> outside the context of training, see the <a href="https://galaxyproject.org/support/"><span class="notranslate">Galaxy</span> Support page</a>.</p></blockquote>




<h4 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/learners_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-do-i-start"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where do I start?</h3>  <p>If you are new to <span class="notranslate">Galaxy</span> then start with one of the introductory topics.  These introduce you to concepts that are useful in <span class="notranslate">Galaxy</span>, no matter what domain you are doing analysis in.</p>  <p>If you are already familiar with <span class="notranslate">Galaxy</span> basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.</p>  <p>If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in <span class="notranslate">Galaxy</span>, then many tutorials include <em>Instructions for the impatient</em> sections.</p></blockquote>




<h4 class="faq-area"> Where can I run the hands-on tutorials? <a href="/training-material/faqs/gtn/learners_running_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-can-i-run-the-hands-on-tutorials"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where can I run the hands-on tutorials?</h3>  <p>To run the hands-on tutorials you need a <span class="notranslate">Galaxy</span> server to run them on.</p>  <p>Each tutorial is annotated with information about which <a href="https://galaxyproject.org/public-galaxy-servers/">public <span class="notranslate">Galaxy</span> servers</a> it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.</p>  <p>If your organization/consortia/community has its own <span class="notranslate">Galaxy</span> server, then you may  want to run tutorials on that. You will need to confirm that all necessary tools and reference genomes are available on your server and possible install missing tools and data. To learn how to do that, you can follow our <a href="/training-material/topics/instructors/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</p>  <p>Some topics have a <a href="https://www.docker.com/">Docker</a> image that can be installed and run on participants‚Äô laptops.  These Docker images contain <span class="notranslate">Galaxy</span> instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to <a href="https://docs.docker.com/install/">install Docker</a>.</p>  <p>Finally, you can also run your tutorials on cloud-based infrastructures.  <span class="notranslate">Galaxy</span> is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://www.gvl.org.au/">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.</p>  <p>If you are already familiar with, and have an account on <a href="https://aws.amazon.com/">Amazon Web Services</a> then you can also launch a <span class="notranslate">Galaxy</span> server there using <a href="https://launch.usegalaxy.org/">CloudLaunch</a>.</p></blockquote>




<h4 class="faq-area"> How do I use this material? <a href="/training-material/faqs/gtn/learners_using_materials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-do-i-use-this-material"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How do I use this material?</h3>  <p>Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.</p>  <p>Most of your learning will happen in the next step - the hands-on tutorials.  This is where you‚Äôll become familiar with using the <span class="notranslate">Galaxy</span> interface and experiment with different ways to use <span class="notranslate">Galaxy</span> and the tools in <span class="notranslate">Galaxy</span>.</p></blockquote>



 <br> 

<h3 id="mapping"> <a href="#mapping"> <span class="notranslate">Mapping</span> </a></h3>

<hr/>



<h4 class="faq-area"> Is it possible to visualize the <span class="notranslate">RNA</span> STAR bam file using the JBrowse tool? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-is-it-possible-to-visualize-the-rna-star-bam-file-using-the-jbrowse-tool"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Is it possible to visualize the <span class="notranslate">RNA</span> STAR bam file using the JBrowse tool?</h3>  <p>Yes, that should work.</p></blockquote>




<h4 class="faq-area"> <span class="notranslate">RNA</span>star: Why do we set 36 for 'Length of the genomic sequence around annotated junctions'? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-rnastar-why-do-we-set-36-for-length-of-the-genomic-sequence-around-annotated-junctions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> <span class="notranslate">RNA</span>star: Why do we set 36 for ‚ÄòLength of the genomic sequence around annotated junctions‚Äô?</h3>  <p><span class="notranslate">RNA</span> STAR is using the gene model to create the database of splice junctions, and that these don‚Äôt ‚Äúneed‚Äù to have a length longer than the <span class="notranslate">reads</span> (37bp).</p></blockquote>



 <br> 

<h3 id="markdown"> <a href="#markdown"> Markdown </a></h3>

<hr/>



<h4 class="faq-area"> How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-create-a-tutorial-skeleton-from-a-galaxy-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>?</h3>  <p>There are two ways to do this:</p>  <ol>  <li>Use <a href="https://planemo.readthedocs.io/en/latest/">planemo</a> on your local machine. Please see the tutorial named <a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">‚ÄúCreating a new tutorial‚Äù</a> for detailed instructions.</li>  <li>Use our <a href="https://ptdk.herokuapp.com">web service</a></li></ol></blockquote>



 <br> 

<h3 id="other"> <a href="#other"> Other </a></h3>

<hr/>



<h4 class="faq-area"> Are there any upcoming events focused on <span class="notranslate">Galaxy</span> Training? <a href="/training-material/faqs/gtn/galaxy_events.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-are-there-any-upcoming-events-focused-on-galaxy-training"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Are there any upcoming events focused on <span class="notranslate">Galaxy</span> Training?</h3>  <p>Yes, always! Have a look at the <a href="https://galaxyproject.org/events/"><span class="notranslate">Galaxy</span> Community Events Calendar</a> for what coming up right now.</p></blockquote>




<h4 class="faq-area"> Sustainability of the training-material and metadata <a href="/training-material/faqs/gtn/gtn_sustainability.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sustainability-of-the-training-material-and-metadata"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sustainability of the training-material and metadata</h3>  <p>This repository is hosted on <a href="https://github.com/">GitHub</a> using git as a <a href="https://en.wikipedia.org/wiki/Distributed_version_control">DVCS</a>. Therefore the community is hosting backups of thisrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.In addition we mirror snapshops of this repo on <a href="http://zenodo.org">Zenodo</a>.</p></blockquote>



 <br> 

<h3 id="outputs"> <a href="#outputs"> Outputs </a></h3>

<hr/>



<h4 class="faq-area"> Does MaxQuant give as output possibility the PSMs and PEPs? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-does-maxquant-give-as-output-possibility-the-psms-and-peps"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Does MaxQuant give as output possibility the PSMs and PEPs?</h3>  <p>Many output options, evidence &amp; msms contain e.g. PSM or feature level info</p></blockquote>



 <br> 

<h3 id="proteogenomics general"> <a href="#proteogenomics general"> Proteogenomics general </a></h3>

<hr/>



<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-example-histories-for-the-proteogenomics-tutorials"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Example histories for the proteogenomics tutorials</h3>  <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>  <p><span class="notranslate">Galaxy</span> EU (use<span class="notranslate">galaxy</span>.eu):</p>  <ul>  <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li></ul>  <p><span class="notranslate">Galaxy</span> Main (use<span class="notranslate">galaxy</span>.org):</p>  <ul>  <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li></ul></blockquote>




<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-example-histories-for-the-proteogenomics-tutorials"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Example histories for the proteogenomics tutorials</h3>  <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>  <p><span class="notranslate">Galaxy</span> EU (use<span class="notranslate">galaxy</span>.eu):</p>  <ul>  <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li></ul>  <p><span class="notranslate">Galaxy</span> Main (use<span class="notranslate">galaxy</span>.org):</p>  <ul>  <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li></ul></blockquote>




<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-example-histories-for-the-proteogenomics-tutorials"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Example histories for the proteogenomics tutorials</h3>  <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>  <p><span class="notranslate">Galaxy</span> EU (use<span class="notranslate">galaxy</span>.eu):</p>  <ul>  <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li></ul>  <p><span class="notranslate">Galaxy</span> Main (use<span class="notranslate">galaxy</span>.org):</p>  <ul>  <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>  <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>  <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li></ul></blockquote>




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can I use these workflows on datasets generated from our laboratory?</h3>  <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p></blockquote>




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can I use these workflows on datasets generated from our laboratory?</h3>  <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p></blockquote>




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Can I use these workflows on datasets generated from our laboratory?</h3>  <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p></blockquote>




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</h3>  <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>  <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>  <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in <span class="notranslate">Galaxy</span>).</p></blockquote>




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</h3>  <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>  <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>  <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in <span class="notranslate">Galaxy</span>).</p></blockquote>




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</h3>  <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>  <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>  <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in <span class="notranslate">Galaxy</span>).</p></blockquote>




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-kind-of-variants-are-seen-in-the-output"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What kind of variants are seen in the output?</h3>  <p>From this <span class="notranslate">workflow</span> we can see insertions, deletions, SNVs, or we will know whether it‚Äôs an intron, exon, splice junction etc.</p></blockquote>




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-kind-of-variants-are-seen-in-the-output"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What kind of variants are seen in the output?</h3>  <p>From this <span class="notranslate">workflow</span> we can see insertions, deletions, SNVs, or we will know whether it‚Äôs an intron, exon, splice junction etc.</p></blockquote>




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-what-kind-of-variants-are-seen-in-the-output"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> What kind of variants are seen in the output?</h3>  <p>From this <span class="notranslate">workflow</span> we can see insertions, deletions, SNVs, or we will know whether it‚Äôs an intron, exon, splice junction etc.</p></blockquote>



 <br> 

<h3 id="recommended browser"> <a href="#recommended browser"> Recommended browser </a></h3>

<hr/>



<h4 class="faq-area"> UCSC - I fetched data from a remote website but now I‚Äôm logged out of <span class="notranslate">Galaxy</span> and my data is gone? <a href="/training-material/topics/introduction/faqs/fetched_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-ucsc---i-fetched-data-from-a-remote-website-but-now-im-logged-out-of-galaxy-and-my-data-is-gone"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: UCSC - I fetched data from a remote website but now I‚Äôm logged out of <span class="notranslate">Galaxy</span> and my data is gone?</h3>  <p>This is a known bug with Chrome + <span class="notranslate">Galaxy</span>, we‚Äôre working on it <a href="https://github.com/galaxyproject/galaxy/issues/11374"><span class="notranslate">galaxy</span>project/<span class="notranslate">galaxy</span>#11374</a>. For now we can recommend using Firefox (known to work) or trying another browser.</p></blockquote>



 <br> 

<h3 id="reference genomes"> <a href="#reference genomes"> Reference genomes </a></h3>

<hr/>



<h4 class="faq-area"> How to use Custom Reference Genomes? <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-use-custom-reference-genomes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to use Custom Reference Genomes?</h3>  <p>A reference genome contains the nucleotide sequence of the chromosomes, scaffolds, transcripts, or contigs for single species. It is representative of a specific genome build or release. There are two options to use reference genomes in <span class="notranslate">Galaxy</span>: <em>native</em> (provided by the server administrators and used by most of the tools) and <em>custom</em> (uploaded by users in FASTA format).</p>  <p>There are five basic steps to use a Custom Reference Genome:</p>  <ol>  <li>Obtain a FASTA copy of the target genome.</li>  <li>Use FTP to upload the genome to <span class="notranslate">Galaxy</span> and load into a history as a dataset.</li>  <li>Clean up the format with the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>  <li>Make sure the chromosome identifiers are a match for other inputs.</li>  <li>Set a tool form‚Äôs options to use a custom reference genome from the history and select the loaded genome.</li></ol></blockquote>




<h4 class="faq-area"> Sorting Reference Genome <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sorting-reference-genome"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sorting Reference Genome</h3>  <p>Certain tools expect that reference genomes are sorted in <a href="https://en.wikipedia.org/wiki/Lexicographic_order">lexicographical order</a>. These tools are often <span class="notranslate">downstream</span> of the initial <span class="notranslate">mapping</span> tools, which means that a large investment in a project has already been made, before a problem with sorting pops up in conclusion layer tools. How to avoid? Always sort your FASTA reference genome dataset at the beginning of a project. Many sources only provide sorted genomes, but double checking is your own responsibility, and super easy in <span class="notranslate">Galaxy</span>!</p>  <ol>  <li>Convert Formats -&gt; <strong>FASTA-to-Tabular</strong></li>  <li>Filter and Sort -&gt; <strong>Sort</strong>    on column: c1     with flavor: Alphabetical    everything in: Ascending order</li>  <li>Convert Formats -&gt; <strong>Tabular-to-FASTA</strong></li></ol>  <p>Note: The above sorting method is for most tools, but not all. In particular, GATK tools have a tool-specific sort order requirement.</p></blockquote>




<h4 class="faq-area"> Troubleshooting Custom Genome fasta <a href="/training-material/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-troubleshooting-custom-genome-fasta"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Troubleshooting Custom Genome fasta</h3>  <p>If a custom genome/transcriptome/exome dataset is producing errors, double check the format and that the chromosome identifiers between <strong>ALL</strong> inputs. Clicking on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> will often provide a description of the problem. This does not automatically submit a bug report, and it is not always necessary to do so, but it is a good way to get some information about <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting.html">why a job is failing</a>.</p>  <ul>  <li>    <p>Custom genome not assigned as FASTA format</p>    <ul>      <li><strong>Symptoms include</strong>: Dataset not included in custom genome ‚ÄúFrom history‚Äù pull down menu on tool forms.</li>      <li><strong>Solution</strong>: Check datatype assigned to dataset and assign <strong>fasta</strong> format.</li>      <li><strong>How</strong>: Click on the dataset‚Äôs pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to reach the ‚ÄúEdit Attributes‚Äù form, and in the Datatypes tab &gt; <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">redetect the datatype</a>.</li>      <li>If <code class="language-plaintext highlighter-rouge">fasta</code> is not assigned, there is a format problem to correct.</li>    </ul>  </li>  <li>    <p>Incomplete Custom genome file load</p>    <ul>      <li><strong>Symptoms include</strong>: Tool errors result the first time you use the Custom genome.</li>      <li><strong>Solution</strong>: Use <strong>Text Manipulation ‚Üí Select last lines from a dataset</strong> to check last 10 lines to see if file is truncated.</li>      <li><strong>How</strong>: Reload the dataset (switch to <a href="https://galaxyproject.org/ftp-upload/">FTP</a> if not using already). Check your FTP client logs to make sure the load is complete.</li>    </ul>  </li>  <li>    <p>Extra spaces, extra lines, inconsistent line wrapping, or any deviation from strict FASTA format</p>    <ul>      <li><strong>Symptoms include: <span class="notranslate">RNA</span>-seq tools (Cufflinks, Cuffcompare, Cuffmerge, Cuffdiff)</strong> fails with error <code class="language-plaintext highlighter-rouge">Error: sequence lines in a FASTA record must have the same length!.</code></li>      <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within <span class="notranslate">Galaxy</span>, then re-run.</li>      <li><strong>How</strong>:        <ul>          <li><strong>Quick re-formatting</strong> Run the tool through the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>          <li><strong>Optional Detailed re-formatting</strong> Start with <strong>FASTA manipulation ‚Üí FASTA Width formatter</strong> with a value between 40-80 (60 is common) to reformat wrapping. Next, use Filter and Sort ‚Üí Select with ‚Äú&gt;‚Äù to examine identifiers. Use a combination of <strong>Convert Formats ‚Üí FASTA-to-Tabular, Text Manipulation</strong> tools, then <strong>Tabular-to-FASTA</strong> to correct.</li>          <li><strong>With either of the above</strong>, finish by using <strong>Filter and Sort ‚Üí Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use ‚ÄúNOT matching‚Äù to remove these lines and output a properly format fasta dataset).</li>        </ul>      </li>    </ul>  </li>  <li>    <p>Inconsistent line wrapping, common if merging chromosomes from various Genbank records (e.g. primary chroms with mito)</p>    <ul>      <li><strong>Symptoms include</strong>: Tools (<strong>SAMTools, Extract Genomic <span class="notranslate">DNA</span></strong>, but rarely alignment tools) may complain about unexpected line lengths/missing identifiers. Or they may just fail for what appears to be a cluster error.</li>      <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within <span class="notranslate">Galaxy</span>.</li>      <li><strong>How</strong>: Use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Finish by using <strong>Filter and Sort ‚Üí Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use ‚ÄúNOT matching‚Äù to remove these lines and output a properly format fasta dataset).</li>    </ul>  </li>  <li>    <p>Unsorted fasta genome file</p>    <ul>      <li><strong>Symptoms include</strong>: Tools such as <strong>Extract Genomic <span class="notranslate">DNA</span></strong> report problems with sequence lengths.</li>      <li><strong>Solution</strong>: First try sorting and re-formatting in <span class="notranslate">Galaxy</span> then re-run.</li>      <li><strong>How</strong>: To sort, follow instructions for <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html">Sorting</a> a Custom Genome.</li>    </ul>  </li>  <li>    <p>Identifier and Description in ‚Äú&gt;‚Äù title lines used inconsistently by tools in the same analysis</p>    <ul>      <li><strong>Symptoms include</strong>: Will generally manifest as a false genome-mismatch problem.</li>      <li><strong>Solution</strong>: Remove the description content and re-run all tools/workflows that used this input. <span class="notranslate">Mapping</span> tools will usually not fail, but <span class="notranslate">downstream</span> tools will. When this comes up, it usually means that an analysis needs to be started over from the <span class="notranslate">mapping</span> step to correct the problems. No one enjoys redoing this work. Avoid the problems by formatting the genome, by double checking that the same reference genome was used for all steps, and by making certain the ‚Äòidentifiers‚Äô are a match between all planned inputs (including reference annotation such as GTF data) <strong>before using your custom genome</strong>.</li>      <li><strong>How</strong>: To drop the title line description content, use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Next, double check that the chromosome identifiers are an exact match between all inputs.</li>    </ul>  </li>  <li>    <p>Unassigned database</p>    <ul>      <li><strong>Symptoms include</strong>: Tools report that no build is available for the assigned reference genome.</li>      <li><strong>Solution</strong>: This occurs with tools that require an assigned database metadata attribute. <strong>SAMTools and Picard</strong> often require this assignment.</li>      <li><strong>How</strong>: Create a <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom Build</a> and assign it to the dataset.</li>    </ul>  </li></ul></blockquote>



 <br> 

<h3 id="rule-builder"> <a href="#rule-builder"> Rule-builder </a></h3>

<hr/>



<h4 class="faq-area"> Flatten a list of list of paired datasets into a list of paired datasets <a href="/training-material/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-flatten-a-list-of-list-of-paired-datasets-into-a-list-of-paired-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Flatten a list of list of paired datasets into a list of paired datasets</h3>  <p>Sometimes you find yourself with a <code class="language-plaintext highlighter-rouge">list:list:paired</code>, i.e. a collection of collection of paired end data, and you really want a <code class="language-plaintext highlighter-rouge">list:paired</code>, a flatter collection of paired end data. This is easy to resolve with <span class="tool" data-tool="__APPLY_RULES__" title="Tested with __APPLY_RULES__"><strong>Apply rules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>APPLY_RULES</strong></span></span> :</p>  <ol>  <li>Open <span class="tool" data-tool="__APPLY_RULES__" title="Tested with __APPLY_RULES__"><strong>Apply rules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>APPLY_RULES</strong></span></span></li>  <li>Select your collection</li>  <li>Click <strong>Edit</strong></li></ol>  <p>You‚Äôll now be in the Apply rules editing interface. There are three columns (if it‚Äôs a <code class="language-plaintext highlighter-rouge">list:list:paired</code>)</p>  <ol>  <li>The outermost <code class="language-plaintext highlighter-rouge">list</code> identifier(s)</li>  <li>The next <code class="language-plaintext highlighter-rouge">list</code> identifier(s)</li>  <li>The paired-end indicator</li></ol>  <p>Flattening this top level list, so it‚Äôs just a <code class="language-plaintext highlighter-rouge">list:paired</code> requires a few changes:</p>  <ol>  <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Concatenate Columns</code>    <ul>      <li><em>‚ÄúFrom Column‚Äù</em>: <code class="language-plaintext highlighter-rouge">A</code></li>      <li><em>‚ÄúFrom Column‚Äù</em>: <code class="language-plaintext highlighter-rouge">B</code>This creates a column with the top list identifier, and the inner list identifier, which will be our new list identifier for the flattened list.</li>    </ul>  </li>  <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>    <ul>      <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Paired-end Indicator</code>        <ul>          <li><em>‚ÄúPaired-end Indicator‚Äù</em>: <code class="language-plaintext highlighter-rouge">C</code></li>        </ul>      </li>      <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>        <ul>          <li><em>‚ÄúList Identifier(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">D</code></li>        </ul>      </li>      <li>Click Apply</li>    </ul>  </li>  <li>Click <strong>Save</strong></li>  <li>Click <strong>Execute</strong></li></ol>  <p>The tool will execute and reshape your list, congratulations!</p></blockquote>



 <br> 

<h3 id="sequencing"> <a href="#sequencing"> Sequencing </a></h3>

<hr/>



<h4 class="faq-area"> Illumina MiSeq sequencing <a href="/training-material/faqs/galaxy/sequencing_illumina_miseq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Illumina MiSeq sequencing</h3>  <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the namesuggests, fluorescent labels are measured for every base that bind at aspecific moment at a specific place on a flow cell. These flow cells arecovered with oligos (small single strand <span class="notranslate">DNA</span> strands). In the librarypreparation the <span class="notranslate">DNA</span> strands are cut into small <span class="notranslate">DNA</span> fragments (differs perkit/device) and specific pieces of <span class="notranslate">DNA</span> (adapters) are added, which arecomplementary to the oligos. Using bridge amplification large amounts ofclusters of these <span class="notranslate">DNA</span> fragments are made. The reverse string is washed away,making the clusters single stranded. Fluorescent bases are added one by one,which emit a specific light for different bases when added. This is happeningfor whole clusters, so this light can be detected and this data is basecalled(translation from light to a nucleotide) to a nucleotide sequence (Read). Forevery base a quality score is determined and also saved per read. Thisprocess is repeated for the reverse strand on the same place on the flowcell, so the forward and reverse <span class="notranslate">reads</span> are from the same <span class="notranslate">DNA</span> strand. Theforward and reversed <span class="notranslate">reads</span> are linked together and should always be processedtogether!</p>  <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p></blockquote>




<h4 class="faq-area"> Nanopore sequencing <a href="/training-material/faqs/galaxy/sequencing_nanopore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Nanopore sequencing</h3>  <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>  <ol>  <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>  <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>  <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li></ol>  <p><img src="/training-material/topics/metagenomics/images/plasmid-metagenomics-nanopore/sequence_method.jpg" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://nanoporetech.com/sites/default/files/s3/white-papers/WGS_Assembly_white_paper.pdf?submissionGuid=40a7546b-9e51-42e7-bde9-b5ddef3c3512&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>  <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change inelectrical current is measured over the membrane of a flow cell. Whennucleotides pass the pores in the flow cell the current change is translated(basecalled) to nucleotides by a basecaller. A schematic overview is given inthe picture above.</p>  <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software ispresent on the devices. With basecalling the electrical signals are translatedto bases (A,T,G,C) with a quality score per base. The sequenced <span class="notranslate">DNA</span> strand willbe basecalled and this will form one read. Multiple <span class="notranslate">reads</span> will be stored in afastq file.</p></blockquote>



 <br> 

<h3 id="support"> <a href="#support"> Support </a></h3>

<hr/>



<h4 class="faq-area"> I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why? <a href="/training-material/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">  <h3 data-toc-skip id="-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why?</h3>  <p>This is okay! Many aspects of the tutorial can potentially affect the exact results you obtain. For example, the reference genome version used and versions of tools. It‚Äôs less important to get the exact results shown in the tutorial, and more important to understand the concepts so you can apply them to your own data.</p></blockquote>




<h4 class="faq-area"> Where do I get more support? <a href="/training-material/faqs/galaxy/support.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-do-i-get-more-support"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where do I get more support?</h3>  <p>If you need support for using <span class="notranslate">Galaxy</span>, running your analysis or completing a tutorial, please try one of the following options:</p>  <ul>  <li><strong>Gitter Chat</strong>: You can get help on Gitter chat platform, on various channels.    <ul>      <li><a href="https://gitter.im/galaxyproject/Lobby"><span class="notranslate">Galaxy</span> General Support</a></li>      <li><a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Training Support</a></li>      <li><a href="https://gitter.im/usegalaxy-eu/Lobby"><span class="notranslate">Galaxy</span> EU Server</a></li>    </ul>  </li>  <li>    <p><strong><span class="notranslate">Galaxy</span> Help Forum</strong>: You can also have a look at the <a href="https://help.galaxyproject.org"><span class="notranslate">Galaxy</span> Help Forum</a>. Your question may already have been answered here before. If not, you can post your question here.</p>  </li>  <li><strong>Contact Server Admins:</strong> If you think there is a problem with the <span class="notranslate">Galaxy</span> server, or you would like to make a request, <a href="/training-material/faqs/galaxy/support_admins.html">contact the <span class="notranslate">Galaxy</span> server administrators</a>.</li></ul></blockquote>




<h4 class="faq-area"> Contacting <span class="notranslate">Galaxy</span> Administrators <a href="/training-material/faqs/galaxy/support_admins.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-contacting-galaxy-administrators"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Contacting <span class="notranslate">Galaxy</span> Administrators</h3>  <p>If you suspect there is something wrong with the server, or would like to request a tool to be installed, you should contact the server administrators for the <span class="notranslate">Galaxy</span> you are on.</p>  <ul>  <li><strong>Tool error?</strong> Please follow these <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting steps</a></li>  <li>Each <span class="notranslate">Galaxy</span> server has different contact procedures, here are the contact options for the 3 biggest servers:    <ul>      <li><strong><span class="notranslate">Galaxy</span> US:</strong> <a href="mailto:help@galaxyproject.org">Email</a>, <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a></li>      <li><strong><span class="notranslate">Galaxy</span> EU:</strong> <a href="https://gitter.im/usegalaxy-eu/Lobby">Gitter channel</a>, <a href="https://usegalaxy.eu/tiaas/">Request TIaaS</a></li>      <li><strong><span class="notranslate">Galaxy</span> AU:</strong> <a href="mail:help@genome.edu.au">Email</a>, <a href="https://request.usegalaxy.org.au/">Request a tool</a>, <a href="https://docs.google.com/forms/d/e/1FAIpQLSeiw6ajmkezLCwbXc3OFQEU3Ai9hGnBd967u9YbQ8ANPgvatA/viewform">Request Data Quota</a></li>      <li>Other <span class="notranslate">Galaxy</span> servers? Check the homepage for more information.</li>    </ul>  </li></ul></blockquote>



 <br> 

<h3 id="tools"> <a href="#tools"> Tools </a></h3>

<hr/>



<h4 class="faq-area"> Regular <span class="notranslate">Expression</span>s 101 <a href="/training-material/faqs/galaxy/analysis_regular_expressions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-regular-expressions-101"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Regular <span class="notranslate">Expression</span>s 101</h3>  <p>Regular <span class="notranslate">expression</span>s are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of <span class="notranslate">Galaxy</span>.</p>  <h4 id="finding">Finding</h4>  <p>Below are just a few examples of basic <span class="notranslate">expression</span>s:</p>  <table>  <thead>    <tr>      <th>Regular <span class="notranslate">expression</span></th>      <th>Matches</th>    </tr>  </thead>  <tbody>    <tr>      <td><code class="language-plaintext highlighter-rouge">abc</code></td>      <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>      <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>      <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>      <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>      <td>any lowercase letter</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>      <td>any letter (upper or lower case)</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>      <td>numbers 0-9</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\d</code></td>      <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\D</code></td>      <td>any non-digit character</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\w</code></td>      <td>any alphanumeric character</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\W</code></td>      <td>any non-alphanumeric character</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\s</code></td>      <td>any whitespace</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\S</code></td>      <td>any non-whitespace character</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">.</code></td>      <td><em>any</em> character</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">\.</code></td>      <td>¬†</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>      <td>between <em>x</em> and <em>y</em> repetitions</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">^</code></td>      <td>the beginning of the line</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">$</code></td>      <td>the end of the line</td>    </tr>  </tbody></table>  <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular <span class="notranslate">expression</span>. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>  <p><strong>Examples</strong></p>  <table>  <thead>    <tr>      <th>Regular <span class="notranslate">expression</span></th>      <th>matches</th>    </tr>  </thead>  <tbody>    <tr>      <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>      <td>4 digits (e.g. a year)</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>      <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>      <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">^$</code></td>      <td>empty line</td>    </tr>  </tbody></table>  <h4 id="replacing">Replacing</h4>  <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values.</p>  <table>  <thead>    <tr>      <th>Regular <span class="notranslate">expression</span></th>      <th>Input</th>      <th>Captures</th>    </tr>  </thead>  <tbody>    <tr>      <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>      <td><code class="language-plaintext highlighter-rouge">chr14</code></td>      <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>    </tr>    <tr>      <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>      <td>24 July 1984</td>      <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>    </tr>  </tbody></table>  <p>An <span class="notranslate">expression</span> like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement <span class="notranslate">expression</span>, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn‚Äôt stop after the first match.</p>  <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>  <p>Note: In <span class="notranslate">Galaxy</span>, you are often asked to provide the find and replacement <span class="notranslate">expression</span>s separately, so you don‚Äôt have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>  <p>There is a lot more you can do with regular <span class="notranslate">expression</span>s, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>  <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular <span class="notranslate">expression</span>s.</p>  <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular <span class="notranslate">expression</span>s, it even provides an explanation of a regular <span class="notranslate">expression</span> if you provide one.</p></blockquote>




<h4 class="faq-area"> Changing the tool version <a href="/training-material/faqs/galaxy/tools_change_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-tool-version"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the tool version</h3>  <p>Tools are frequently updated to new versions. Your <span class="notranslate">Galaxy</span> may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>  <p><strong>Switching to a different version</strong> of a tool:</p>  <ul>  <li>Open the tool</li>  <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>  <li>Select the desired version from the dropdown list</li></ul></blockquote>




<h4 class="faq-area"> Organizing the tool panel <a href="/training-material/faqs/galaxy/tools_favorite.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-organizing-the-tool-panel"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Organizing the tool panel</h3>  <p><span class="notranslate">Galaxy</span> servers can have a lot of tools available, which can make it challenging to find the tool you are looking for. To help find your favourite tools, you can:</p>  <ul>  <li><strong>Keep a list of your favourite tools</strong> to find them back easily later.    <ul>      <li><strong>Adding tools to your favourites</strong>        <ul>          <li>Open a tool</li>          <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-star</span> next to the tool name to add it to your favourites</li>        </ul>      </li>      <li><strong>Viewing your favourite tools</strong>        <ul>          <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-star</span> at the top of the <span class="notranslate">Galaxy</span> tool panel (above the tool search bar)</li>          <li>This will filter the toolbox to show all your starred tools</li>        </ul>      </li>    </ul>  </li>  <li><strong>Change the tool panel view</strong>    <ul>      <li>Click on the <i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-panelview</span> icon at the top of the <span class="notranslate">Galaxy</span> tool panel (above the tool search bar)</li>      <li>Here you can view the tools by EDAM ontology terms        <ul>          <li>EDAM Topics (e.g. biology, ecology)</li>          <li>EDAM Operations (e.g. quality control, variant analysis)</li>          <li>You can always get back to the default view by choosing <strong>‚ÄúFull Tool Panel‚Äù</strong></li>        </ul>      </li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Where is the tool help? <a href="/training-material/faqs/galaxy/tools_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding tool support
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-is-the-tool-help"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where is the tool help?</h3>  <p>There is documentation available on the tool form itself which mentions the following information:</p>  <ul>  <li>Parameters</li>  <li>Expected format for input dataset(s)</li>  <li>Links to publications and <span class="notranslate">ToolShed</span> source repositories</li>  <li>Tool and wrapper version(s)</li>  <li>3rd party author web sites and documentation</li></ul>  <p><strong>Scroll down on the tool form</strong> to locate:</p>  <ul>  <li>Information about expected inputs/outputs</li>  <li>Expanded definitions</li>  <li>Sample data</li>  <li>Example use cases</li>  <li>Graphics</li></ul></blockquote>




<h4 class="faq-area"> Viewing tool logs (`stdout` and `stderr`) <a href="/training-material/faqs/galaxy/tools_logs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-viewing-tool-logs-stdout-and-stderr"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Viewing tool logs (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>)</h3>  <p>Most tools create log files as output, which can contain useful information about how the tool ran (<code class="language-plaintext highlighter-rouge">stdout</code>, or <em>standard output</em>), and what went wrong (<code class="language-plaintext highlighter-rouge">stderr</code>, or <em>standard error</em>).</p>  <p>To view these log files in <span class="notranslate">Galaxy</span>:</p>  <ul>  <li>Expand one of the outputs of the tool in your history</li>  <li>Click on <strong>View details</strong> <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span></li>  <li>Scroll to the <strong>Job Information</strong> section    <ul>      <li>Here you will find links to the log files (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>).</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> If a Tool is Missing <a href="/training-material/faqs/galaxy/tools_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-if-a-tool-is-missing"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: If a Tool is Missing</h3>  <p>To use the tools installed and available on the <span class="notranslate">Galaxy</span> server:</p>  <ol>  <li>At the top of the left tool panel, type in a tool name or datatype into the tool search box.</li>  <li>Shorter keywords find more choices.</li>  <li>Tools can also be directly browsed by category in the tool panel.</li></ol>  <p>If you can‚Äôt find a tool you need for a tutorial on <span class="notranslate">Galaxy</span>, please:</p>  <ol>  <li><strong>Check that you are using a compatible <span class="notranslate">Galaxy</span> server</strong>    <ul>      <li>Navigate to the overview box at the top of the tutorial</li>      <li>Find the <strong>‚ÄúSupporting Materials‚Äù</strong> section</li>      <li>Check <strong>‚ÄúAvailable on these Galaxies‚Äù</strong></li>      <li>If your server is not listed here, the tutorial is not supported on your <span class="notranslate">Galaxy</span> server</li>      <li>You can create an account on one of the supporting Galaxies  <img src="/training-material/faqs/galaxy/images/supporting_galaxies.png" alt="screenshot of overview box with available Galaxies section" /></li>    </ul>  </li>  <li>Use the <strong>Tutorial mode</strong> feature    <ul>      <li>Open your <span class="notranslate">Galaxy</span> server</li>      <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside <span class="notranslate">Galaxy</span>.</li>      <li>Navigate to your tutorial</li>      <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>      <li><strong>Note:</strong> this does not work for all tutorials (yet)  <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>    </ul>  </li>  <li><strong>Still not finding the tool?</strong>    <ul>      <li>Ask help in <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a>.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Re-running a tool <a href="/training-material/faqs/galaxy/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-re-running-a-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Re-running a tool</h3>  <ol>  <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>  <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> the tool</li></ol>  <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p></blockquote>




<h4 class="faq-area"> Para volver a ejecutar una herramienta <a href="/training-material/faqs/galaxy-es/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-para-volver-a-ejecutar-una-herramienta"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Para volver a ejecutar una herramienta</h3>  <ol>  <li><strong>Expande</strong> uno de los conjuntos de datos de la salida de la herramienta haciendo clic sobre √©l</li>  <li><strong>Selecciona</strong> volver a ejecutar <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> de la herramienta</li></ol>  <p>Esto es de utilidad si quieres volver a correr la herramienta variando ligeramente los valores de los par√°metros, o si deseas verificar la configuraci√≥n de par√°metros que utilizaste.</p></blockquote>




<h4 class="faq-area"> Selecting a dataset collection as input <a href="/training-material/faqs/galaxy/tools_select_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-selecting-a-dataset-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a dataset collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>




<h4 class="faq-area"> Select multiple datasets <a href="/training-material/faqs/galaxy/tools_select_multiple_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>




<h4 class="faq-area"> Sorting Tools <a href="/training-material/faqs/galaxy/tools_sorting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sorting-tools"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sorting Tools</h3>  <p>Sometimes input errors are caused because of non-sorted inputs. Try using these:</p>  <ul>  <li><strong>Picard SortSam:</strong> Sort SAM/BAM by coordinate or queryname.</li>  <li><strong>Samtools Sort:</strong> Alte<span class="notranslate">rna</span>te for SAM/BAM, best when used for coordinate sorting only.</li>  <li><strong>SortBED order the intervals:</strong> Best choice for <span class="notranslate">BED</span>/Interval.</li>  <li><strong>Sort data in ascending or descending order:</strong> Alte<span class="notranslate">rna</span>te choice for Tabular/<span class="notranslate">BED</span>/Interval/GTF.</li>  <li><strong>VCFsort:</strong> Best choice for VFC.</li>  <li><strong>Tool Form Options for Sorting:</strong> Some tools have an option to sort inputs during job execution. Whenever possible, sort inputs before using tools, especially if jobs fail for not having enough memory resources.</li></ul></blockquote>




<h4 class="faq-area"> Tool doesn't recognize input datasets <a href="/training-material/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-tool-doesnt-recognize-input-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Tool doesn‚Äôt recognize input datasets</h3>  <p>The expected input datatype assignment is explained on the tool form. Review the input select areas and the help section below the Execute button.</p>  <p>Understanding datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">FAQ</a>.</p>  <p>No datasets or collections available? Solutions:</p>  <ol>  <li>Upload or Copy an appropriate dataset for the input into the active history.    <ul>      <li>To load new datasets, review the <strong>Upload</strong> tool and more choices under <strong>Get Data</strong> within <span class="notranslate">Galaxy</span>.</li>      <li>To copy datasets from a different history into the active history see <a href="/training-material/faqs/galaxy/histories_copy_dataset.html">this FAQ</a>.</li>      <li>To use datasets loaded into a shared Data Library see <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html">this FAQ</a>.</li>    </ul>  </li>  <li>Resolve a datatype assignment incompatibility between the dataset and the tool.    <ul>      <li>To redetect a datatype see <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">this FAQ</a>.</li>      <li>To convert a datatype see <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">this FAQ</a>.</li>      <li>To change a datatype see <a href="/training-material/faqs/galaxy/datasets_change_datatype.html">this FAQ</a>.</li>    </ul>  </li>  <li>Individual datasets and dataset collections are selected differently on tool forms.    <ul>      <li>To select a collection input on a tool form see <a href="/training-material/faqs/galaxy/tools_select_collection.html">this FAQ</a>.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using tutorial mode <a href="/training-material/faqs/galaxy/tutorial_mode.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-tutorial-mode"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using tutorial mode</h3>  <p>Tools are frequently updated to new versions. Your <span class="notranslate">Galaxy</span> may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step‚Ä¶ this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>  <ul>  <li>Open your <span class="notranslate">Galaxy</span> server</li>  <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside <span class="notranslate">Galaxy</span>.</li>  <li>Navigate to your tutorial</li>  <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>  <li><strong>Note:</strong> this does not work for all tutorials (yet)  <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li></ul></blockquote>



 <br> 

<h3 id="user interface"> <a href="#user interface"> User interface </a></h3>

<hr/>



<h4 class="faq-area"> I can‚Äôt find the ‚ÄúAnalyze Data‚Äù button <a href="/training-material/topics/introduction/faqs/analyze_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-cant-find-the-analyze-data-button"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I can‚Äôt find the ‚ÄúAnalyze Data‚Äù button</h3>  <p>The <span class="notranslate">Galaxy</span> interface has changed a bit recently, ‚ÄúAnalyze Data‚Äù was always the home button, and now looks like a home icon.</p></blockquote>




<h4 class="faq-area"> My <span class="notranslate">Galaxy</span> looks different than in the tutorial/video <a href="/training-material/topics/introduction/faqs/galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-galaxy-looks-different-than-in-the-tutorialvideo"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My <span class="notranslate">Galaxy</span> looks different than in the tutorial/video</h3>  <p><span class="notranslate">Galaxy</span> gets frequent updates, different servers will be running different versions. This is nothing to worry about, just let us know if you can‚Äôt find how to perform a task in your <span class="notranslate">Galaxy</span>.</p></blockquote>



 <br> 

<h3 id="user preferences"> <a href="#user preferences"> User preferences </a></h3>

<hr/>



<h4 class="faq-area"> Getting your API key <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-getting-your-api-key"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Getting your API key</h3>  <ol>  <li>In your browser, open your <span class="notranslate">Galaxy</span> homepage</li>  <li>Log in, or register a new account, if it‚Äôs the first time you‚Äôre logging in</li>  <li>Go to <code class="language-plaintext highlighter-rouge">User -&gt; Preferences</code> in the top menu bar, then click on <code class="language-plaintext highlighter-rouge">Manage API key</code></li>  <li>If there is no current API key available, click on <code class="language-plaintext highlighter-rouge">Create a new key</code> to generate it</li>  <li>Copy your API key to somewhere convenient, you will need it throughout this tutorial</li></ol></blockquote>




<h4 class="faq-area"> Does your account usage quota seem incorrect? <a href="/training-material/faqs/galaxy/user_preferences_account_usage_quota.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-does-your-account-usage-quota-seem-incorrect"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Does your account usage quota seem incorrect?</h3>  <ol>  <li>Log out of <span class="notranslate">Galaxy</span>, then back in again. This refreshes the disk usage calculation displayed in the Masthead usage (summary) and under <em>User &gt; Preferences</em> (exact).</li></ol>  <p>Note:</p>  <ul>  <li>Your account usage quota can be found at the bottom of your user preferences page.</li></ul></blockquote>




<h4 class="faq-area"> Forgot Password <a href="/training-material/faqs/galaxy/user_preferences_forgot_password.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-forgot-password"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Forgot Password</h3>  <ol>  <li>Go to the <span class="notranslate">Galaxy</span> server you are using.</li>  <li>Click on <strong>Login or Register</strong>.</li>  <li>Enter your email on the Public Name or Email Address entry box.</li>  <li>Click on the link under the password entry box titled <strong>Forgot password? Click here to reset your password.</strong></li>  <li>An email will be sent with a password reset link. This email may be in your email Spam or Trash folders, depending on your filters.</li>  <li>Click on the reset link in the email or copy and paste it into a web browser window.</li>  <li>Enter your new password and click on <strong>Save new password</strong>.</li></ol></blockquote>




<h4 class="faq-area"> Changing user preferences <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-user-preferences"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing user preferences</h3>  <ol>  <li>Make sure you are logged in to <span class="notranslate">Galaxy</span>.</li>  <li>Go to <strong>User &gt; Preferences</strong> in the top menu bar.</li>  <li>To change email and public name, click on <strong>Manage Information</strong> and to change password, click on <strong>Change Password</strong>.</li>  <li>Make the changes and click on the <strong>Save</strong> button at the bottom.</li>  <li>To change email successfully, verify your account by email through the activation link sent by <span class="notranslate">Galaxy</span>.</li></ol>  <p>Note: Don‚Äôt open another account if your email changes, update the existing account email instead. Creating a new account will be detected as a duplicate and will get your account disabled and deleted.</p></blockquote>



 <br> 

<h3 id="utilities"> <a href="#utilities"> Utilities </a></h3>

<hr/>



<h4 class="faq-area"> Got lost along the way? <a href="/training-material/topics/admin/faqs/missed-something.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-got-lost-along-the-way"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Got lost along the way?</h3>  <p>If you missed any steps, you can compare against the <a href="https://github.com/hexylena/git-gat/tree/step-">reference files</a>, or see what changed since <a href="https://github.com/hexylena/git-gat/compare/step--1...step-#files_bucket">the previous tutorial</a>.</p>  <p>If you‚Äôre using <code class="language-plaintext highlighter-rouge">git</code> to track your progress, remember to add your changes and commit with a good commit message!</p></blockquote>



 <br> 

<h3 id="visualisation"> <a href="#visualisation"> Visualisation </a></h3>

<hr/>



<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/faqs/galaxy/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/epigenetics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/genome-annotation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/proteomics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/galaxy-interface/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/sequence-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/visualisation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/introduction/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/variant-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent versions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>



 <br> 

<h3 id="workflows"> <a href="#workflows"> Workflows </a></h3>

<hr/>



<h4 class="faq-area"> Annotate a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_annotate.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-annotate-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Annotate a <span class="notranslate">workflow</span></h3>  <ul>  <li>Open the <span class="notranslate">workflow</span> editor for the <span class="notranslate">workflow</span></li>  <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>Edit Attributes</strong> on the top right</li>  <li>Write a description of the <span class="notranslate">workflow</span> in the <strong>Annotation</strong> box</li>  <li>Add a tag (which will help to search for the <span class="notranslate">workflow</span>) in the <strong>Tags</strong> section</li></ul></blockquote>




<h4 class="faq-area"> Creating a new <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can create a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> from scratch in the <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> editor.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new <span class="notranslate">workflow</span></h3>  <ol>  <li>Click <strong><span class="notranslate">Workflow</span></strong> on the top bar</li>  <li>Click the new <span class="notranslate">workflow</span> <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-new</span> button</li>  <li>Give it a clear and memorable name</li>  <li>Clicking <strong>Save</strong> will take you directly into the <span class="notranslate">workflow</span> editor for that <span class="notranslate">workflow</span></li>  <li>Need more help? Please see the <em>How to make a <span class="notranslate">workflow</span> subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li></ol></blockquote>




<h4 class="faq-area"> Opening the <span class="notranslate">workflow</span> editor <a href="/training-material/faqs/galaxy/workflows_edit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-opening-the-workflow-editor"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Opening the <span class="notranslate">workflow</span> editor</h3>  <ol>  <li>In the top menu bar, click on <strong>Workflows</strong></li>  <li>Click on the name of the <span class="notranslate">workflow</span> you want to edit<img src="/training-material/topics/galaxy-interface/images/edit_workflow.png" alt="Workflow drop down menu showing Edit option" title="When you click on the name of a workflow it will open a drop down menu whose first option is Edit. Select the Edit option to edit the workflow." /></li>  <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-edit</span> <strong>Edit</strong> from the dropdown menu to open the <span class="notranslate">workflow</span> in the <span class="notranslate">workflow</span> editor</li></ol></blockquote>




<h4 class="faq-area"> Extracting a <span class="notranslate">workflow</span> from your history <a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> can automatically create a <span class="notranslate">workflow</span> based on the analysis you have performed in a history. This means that once you have done an analysis manually once, you can easily extract a <span class="notranslate">workflow</span> to repeat it on different data.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extracting-a-workflow-from-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extracting a <span class="notranslate">workflow</span> from your history</h3>  <ol>  <li>    <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> button.</p>    <p>This will make the creation of the <span class="notranslate">workflow</span> easier.</p>  </li>  <li>    <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract <span class="notranslate">workflow</span></strong>.</p>    <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>    <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the <span class="notranslate">workflow</span>.</p>  </li>  <li>    <p>Replace the <strong><span class="notranslate">Workflow</span> name</strong> to something more descriptive.</p>  </li>  <li>    <p><strong>Rename</strong> each <span class="notranslate">workflow</span> input in the boxes at the top of the second column.</p>  </li>  <li>    <p>If there are any steps that shouldn‚Äôt be included in the <span class="notranslate">workflow</span>, you can <strong>uncheck</strong> them in the first column of boxes.</p>  </li>  <li>    <p>Click on the <strong>Create <span class="notranslate">Workflow</span></strong> button near the top.</p>    <p>You will get a message that the <span class="notranslate">workflow</span> was created.</p>  </li></ol></blockquote>




<h4 class="faq-area"> Extraer un flujo de trabajo de tu historial <a href="/training-material/faqs/galaxy-es/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> puede crear autom√°ticamente un flujo de trabajo basado en un an√°lisis almacenado en tu historial. Esto significa que una vez que hayas realizado un an√°lisis manualmente, puedes extraer f√°cilmente un flujo de trabajo para repetirlo con diferentes datos.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extraer-un-flujo-de-trabajo-de-tu-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extraer un flujo de trabajo de tu historial</h3>  <ol>  <li>Elimina cualquier trabajo fallido o no deseado de tu historial.</li>  <li>Haz clic en <strong>Opciones de historial</strong> (icono de engranaje <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span>) en la parte superior del panel de historial.</li>  <li>Selecciona <strong>Extraer flujo de trabajo</strong></li>  <li>Verifica los pasos, ingresa un nombre para tu flujo de trabajo y presiona el bot√≥n <strong>Crear flujo de trabajo</strong>.</li></ol></blockquote>




<h4 class="faq-area"> Hiding intermediate steps <a href="/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-hiding-intermediate-steps"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Hiding intermediate steps</h3>  <p>When a <span class="notranslate">workflow</span> is executed, the user is usually primarily interested in the final product and not in all intermediate steps.By default all the outputs of a <span class="notranslate">workflow</span> will be shown, but we can explicitly tell <span class="notranslate">Galaxy</span> which outputs to show and which to hide for a given <span class="notranslate">workflow</span>.This behaviour is controlled by the little checkbox in front of every output dataset:</p>  <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p></blockquote>




<h4 class="faq-area"> Importing a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> URL‚Äù</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> File‚Äù</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>




<h4 class="faq-area"> Importing a <span class="notranslate">workflow</span> using the search <a href="/training-material/faqs/galaxy/workflows_import_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow-using-the-search"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span> using the search</h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>Import</strong> icon at the top-right of the screen</li>  <li>    <p>Click on <strong>search form</strong> in <strong>Import a <span class="notranslate">Workflow</span> from Configured GA4GH Tool Registry Servers (e.g. Dockstore)</strong></p>  </li>  <li>    <p>Select the relevant TRS Server</p>  </li>  <li>    <p>Type the query</p>  </li>  <li>    <p>Expand the correct <span class="notranslate">workflow</span></p>  </li>  <li>    <p>Click on the wanted version</p>    <p>The <span class="notranslate">workflow</span> will be imported in your workflows</p>  </li></ul></blockquote>




<h4 class="faq-area"> Setting parameters at run-time <a href="/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-setting-parameters-at-run-time"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Setting parameters at run-time</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the parameter you want users to provide every time they run the <span class="notranslate">workflow</span></li>  <li>Click on the arrow in front of the name <i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-runtime-toggle</span> to toggle to <em>set at runtime</em></li></ol></blockquote>




<h4 class="faq-area"> Make a <span class="notranslate">workflow</span> public <a href="/training-material/faqs/galaxy/workflows_publish.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-make-a-workflow-public"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Make a <span class="notranslate">workflow</span> public</h3>  <ul>  <li>Click on <strong><span class="notranslate">Workflow</span></strong> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows</li>  <li>Click on the interesting <span class="notranslate">workflow</span></li>  <li>Click on <strong>Share</strong></li>  <li>Click on <strong>Make <span class="notranslate">Workflow</span> accessible</strong>. This makes the <span class="notranslate">workflow</span> <em>publicly accessible</em> but <em>unlisted</em>.</li>  <li>To also list the <span class="notranslate">workflow</span> in the <em>Shared Data</em> section (in the top menu bar) of <span class="notranslate">Galaxy</span>, click <strong>Make <span class="notranslate">Workflow</span> publicly available in Published Workflows</strong></li></ul></blockquote>




<h4 class="faq-area"> Renaming <span class="notranslate">workflow</span> outputs <a href="/training-material/faqs/galaxy/workflows_rename_output.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-workflow-outputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming <span class="notranslate">workflow</span> outputs</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.    <ul>      <li>        <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>        <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>      </li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Viewing a <span class="notranslate">workflow</span> report <a href="/training-material/faqs/galaxy/workflows_report_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When creating a <span class="notranslate">workflow</span> in <span class="notranslate">Galaxy</span>, you can also define an output report page that should be created. Here you can display certain outputs of the pipeline (e.g. output files, tables, images, etc.) and other information about the run.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-viewing-a-workflow-report"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Viewing a <span class="notranslate">workflow</span> report</h3>  <ul>  <li>Go to <strong>User</strong> on the top menu bar of <span class="notranslate">Galaxy</span>.</li>  <li>Click on <strong><span class="notranslate">Workflow</span> invocations</strong>    <ul>      <li>Here you will find a list of all the workflows you have run</li>    </ul>  </li>  <li>Click on the name of a <span class="notranslate">workflow</span> invocation to expand it<img src="/training-material/faqs/galaxy/images/workflow_invocations_menu.png" alt="workflow invocations list" /></li>  <li>Click on <strong>View Report</strong> to go to the <span class="notranslate">workflow</span> report page</li>  <li><strong>Note:</strong> The report can also be downloaded in PDF format by clicking on the <i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-report-download</span> icon.</li></ul></blockquote>




<h4 class="faq-area"> Running a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_run.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>









<h2 id="bibliography">References</h2>
<ol class="bibliography"><li id="Devenyi_2018">Devenyi, G. A., R. Emonet, R. M. Harris, K. L. Hertweck, D. Irving <i>et al.</i>, 2018 <b>Ten simple rules for collaborative lesson development</b> (S. Markel, Ed.). PLOS Computational Biology 14: e1005963. <a href="https://doi.org/10.1371/journal.pcbi.1005963">10.1371/jou<span class="notranslate">rna</span>l.pcbi.1005963</a></li>
<li id="Garcia2020">Garcia, L., B. Batut, M. L. Burke, M. Kuzak, F. Psomopoulos <i>et al.</i>, 2020 <b>Ten simple rules for making training materials FAIR</b> (S. Markel, Ed.). PLOS Computational Biology 16: e1007854. <a href="https://doi.org/10.1371/journal.pcbi.1007854">10.1371/jou<span class="notranslate">rna</span>l.pcbi.1007854</a></li></ol>



<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div><span class="notranslate">Galaxy</span> Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        
            <footer>
    <div class="container">
        <br><br>
        <p>
            The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
            provides researchers with online training materials, connects them with local trainers, and helps promoting open data analysis practices worldwide.
        </p>
        <p>
        The content of the tutorials is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>. The website and infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>.
        </p>

        <p>This is revision <a href="https://github.com/galaxyproject/training-material/commit/886decceb6c0558ba6b920bcebb885554ce262fc">886decceb6c0558ba6b920bcebb885554ce262fc</a></p>
    </div>
</footer>

        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
