<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Avian influenza viral strain analysis from gene segment sequencing data</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" />
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/" />

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material" />
<meta name="DC.type" content="text" />
<meta name="DC.title" content="Avian influenza viral strain analysis from gene segment sequencing data" />
<meta name="DC.publisher" content="Galaxy Training Network" />
<meta name="DC.date" content="2023-01-23 14:28:11 +0000" />
<meta name="DC.creator" content="Wolfgang Maier" />

        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:site_name" content="Galaxy Training Network" />
        <meta property="og:title" content="Galaxy Training: Avian influenza viral strain analysis from gene segment s..." />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-01-23 14:28:11 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Avian influenza viral strain analysis from gene segment sequencing data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Avian influenza viral strain analysis from gene segment sequencing data' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html",
  "timeRequired": "PT4H",
  "keywords": "variant-analysis, virology, public health, work-in-progress",
  "description": "The questions this  addresses are:\n - With reassortment of gene segments being a common event in avian influenza virus (AIV) evolution, does it make sense to use a reference-based mapping approach for constructing consensus genome sequences for AIV samples?\n - Is it possible to reuse existing tools and workflows developed for the analysis of sequencing data from other viruses?\n - How can we obtain meaningful phylogenetic insight from AIV consensus sequences?\n\n\nThe objectives are:\n - Determine how reassortment impacts reference-based mapping approaches\n - Use a collection of per-segment reference sequences to construct a hybrid reference genome that is sufficiently close to a sequenced sample to be useful as a reference for mapping\n - Construct a sample consensus genome from mapped reads\n - Generate per-segment phylogenetic trees of AIV consensus sequences\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Intermediate"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc">
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/aiv-analysis/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                    <button class="btn" data-theme="default">Default</button>
                    <button class="btn" data-theme="night">Night</button>
                    <button class="btn" data-theme="midnight">Midnight</button>
                    <button class="btn" data-theme="rainbow">Rainbow</button>
                    <button class="btn" data-theme="blm">✊🏿</button>
                    <button class="btn" data-theme="progress" >🏳️‍🌈</button>
                    <button class="btn" data-theme="halloween">🎃</button>
                    <button class="btn" data-theme="straya">🇦🇺</button>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-05-01/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" title="Version 2022-05-01">2022-05-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-04-01/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" title="Version 2022-04-01">2022-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<section class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Avian influenza viral strain analysis from gene segment sequencing data</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    

    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/wm75?s=27" alt="Avatar">Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div class="box-title" aria-label="overview box">Overview</div>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>With reassortment of gene segments being a common event in avian influenza virus (AIV) evolution, does it make sense to use a reference-based mapping approach for constructing consensus genome sequences for AIV samples?</p>
</li>
        
        <li><p>Is it possible to reuse existing tools and workflows developed for the analysis of sequencing data from other viruses?</p>
</li>
        
        <li><p>How can we obtain meaningful phylogenetic insight from AIV consensus sequences?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Determine how reassortment impacts reference-based mapping approaches</p>
</li>
        
        <li><p>Use a collection of per-segment reference sequences to construct a hybrid reference genome that is sufficiently close to a sequenced sample to be useful as a reference for mapping</p>
</li>
        
        <li><p>Construct a sample consensus genome from mapped reads</p>
</li>
        
        <li><p>Generate per-segment phylogenetic trees of AIV consensus sequences</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            

            

            

            
            
            
                 <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/faqs/" title="Frequently Asked Questions" alt="Tutorial FAQs">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
  </li>


            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jan 23, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 id="introduction">Introduction</h1>

<p>Of the four species of influenza viruses (Influenza A-D), <em>Influenza A</em> is the most virulent in human hosts and subtypes of it have been responsible for all historic flu pandemics.</p>

<p>The different influenza species infect distinct animal hosts (though all of them can infect humans and pigs) and the most important natural reservoir for <em>Influenza A</em> are birds (in particular, wild aquatic birds), in which it causes <strong>Avian influenza</strong>.</p>

<p>Importantly, all flu pandemics of the last century have been triggered by <strong>reassortment</strong> events between avian and human <em>Influenza A</em> strains (although at least the 2009 swine flu pandemic involved an additional mixing event with an <em>Influenza A</em> strain from pigs).</p>

<p>Reassortment events are a key trait of <em>Influenza A</em> made possible by its wide range of natural hosts combined with two molecular characteristics of the species:</p>

<ol>
  <li>the linear negative-sense single-stranded <span class="notranslate">RNA</span> genomes of influenza viruses (and of all viruses of the <em>Orthomyxoviridae</em> family) are segmented, i.e. consist of several (eight in the case of <em>Influenza A</em> and <em>B</em>) distinct pieces of <span class="notranslate">RNA</span>, which typically encode just one, sometimes two viral proteins.</li>
  <li>the mutation rate in the genome is high in influenza viruses, and particularly high in <em>Influenza A</em>, since their <span class="notranslate">RNA</span> polymerase lacks exonuclease activity necessary for proof-reading.</li>
</ol>

<p>Together these characteristics enable <em>Influenza A</em> to evolve relatively rapidly in non-human hosts, then re-adapt to humans through exchange of segments (reassortment) in a host co-infected with a human and, <em>e.g.</em>, an avian strain. If such a reassortment introduces new variants of the two most antigenic proteins of influenza, HA (hemagglutinin) and NA (neuraminidase), this antigenic shift provides the reassorted strain with immune escape potential, which, if it happens in a genetic background compatible with efficient transmission in humans, can trigger an unusually strong wave of influenza or even a new pandemic.</p>

<p>In order to estimate the likelihood of an epidemic or pandemic influenza event in the near future it is, therefor, important to monitor closely the genome composition of circulating Avian Influenza strains in wild and domestic birds and the huge technological advances over the last decade make it possible to use next-generation sequencing for this purpose.</p>

<p>At the same time, the segmented nature of their genomes combined with high genetic variability, especially in the HA and NA genes, requires rather specialized bioinformatics workflows for the analysis of data from influenza viruses compared to other viruses with similar genome size (like, <em>e.g.</em> SARS-CoV-2):</p>

<p>The viral surface proteins HA and (to a lesser extent) NA are the main targets of the host antibody response and are, thus, under constant selection pressure to mutate into forms capable of evading an existing host immune response. As a consequence, these segments have evolved into a much richer panel of sequence variants than the other segments, to the point that sequences of the HA segment can, at present, be classified into 18 distinct subfamilies, H1-H18, while there are 11 recognized subfamilies of NA, and <em>Influenza A</em> strains are subtyped (as, for example, H5N1, H3N2, etc.) according to the combination of HA- and NA-encoding segments they are carrying.</p>

<p>Importantly, the sequence diversity of HA and (again to a lesser extent) of NA segments is big enough to prevent a naive approach of <span class="notranslate">mapping</span> sequenced <span class="notranslate">reads</span> to one specific agreed-upon <em>Influenza A</em> reference sequence. While this would work for the other six segments, <span class="notranslate">mapping</span> software would regularly fail to find enough plausible <span class="notranslate">mapping</span>s for sequenced <span class="notranslate">reads</span> of HA and NA origin to continue analysis with. This is why, in this tutorial we are going to explore an alte<span class="notranslate">rna</span>tive approach, which is also <span class="notranslate">mapping</span>-based but chooses a suitable reference for each segment dynamically based on the input sequencing data.</p>

<blockquote class="agenda">
  <div id="agenda" class="box-title" aria-label="agenda box: ">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#prepare-analysis-history-and-data" id="markdown-toc-prepare-analysis-history-and-data">Prepare analysis history and data</a>    <ol>
      <li><a href="#get-reference-and-sequenced-samples-data" id="markdown-toc-get-reference-and-sequenced-samples-data">Get reference and sequenced samples data</a></li>
      <li><a href="#inspect-the-reference-data" id="markdown-toc-inspect-the-reference-data">Inspect the reference data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
  <li><a href="#per-segment-subtyping-and-hybrid-reference-construction" id="markdown-toc-per-segment-subtyping-and-hybrid-reference-construction">Per-segment subtyping and hybrid reference construction</a></li>
  <li><a href="#mapping-to-a-hybrid-reference" id="markdown-toc-mapping-to-a-hybrid-reference"><span class="notranslate">Mapping</span> to a hybrid reference</a></li>
  <li><a href="#consensus-sequence-construction" id="markdown-toc-consensus-sequence-construction">Consensus sequence construction</a></li>
  <li><a href="#placing-segments-on-a-phylogenetic-tree" id="markdown-toc-placing-segments-on-a-phylogenetic-tree">Placing segments on a phylogenetic tree</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="prepare-analysis-history-and-data">Prepare analysis history and data</h1>

<p>Any analysis should get its own <span class="notranslate">Galaxy</span> history. So let’s start by creating a new one:</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-prepare-the-galaxy-history" class="box-title" aria-label="hands-on box: Prepare the Galaxy history"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Prepare the <span class="notranslate">Galaxy</span> history</div>

  <ol>
    <li>
      <p>Create a new history for this analysis</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-creating-a-new-history" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
    </li>
    <li>
      <p>Rename the history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-renaming-a-history" class="box-title"><button type="button" aria-controls="tip-renaming-a-history-contents" aria-expanded="true" aria-label="Toggle tip box: Renaming a history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="get-reference-and-sequenced-samples-data">Get reference and sequenced samples data</h2>

<p>In this tutorial you are going to work on a single avian influenza sample sequenced in paired-end mode on the Illumina platform, <em>i.e.</em> we are going to download two datasets of sequenced <span class="notranslate">reads</span> for that sample.</p>

<p>In addition, we are going to base the analysis on a small collection of multiple reference sequences for each influenza gene segment. We prepared this collection for you from public INSAFlu data.</p>

<p>If you have your own curated collection of reference sequences, you should be able to use it for following this tutorial without any problem. Note, however, that the analysis results referred to in many of the questions will be different if you are exchanging the reference collection.
If you want to learn how you can create a <span class="notranslate">Galaxy</span> collection from your own references that is structured like the default we are suggesting here, you may want to follow the dedicated tutorial on <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">Using dataset collections</a>.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-get-the-data" class="box-title" aria-label="hands-on box: Get the data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Get the data</div>

  <ol>
    <li><span class="tool" data-tool="upload1" title="Tested with upload1"><strong>Upload</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span>  the forward and reverse <span class="notranslate">reads</span> of your sequenced sample to your history and turn them into a <em>Paired Collection</em>.
      <ol>
        <li>In the <em>“Download from web or upload from disk”</em> dialogue, switch to the <code class="language-plaintext highlighter-rouge">Collection</code> tab,
then configure the drop-down select boxes on that tab like this:
          <ul>
            <li><em>“Collection Type”</em>: <code class="language-plaintext highlighter-rouge">Pair</code></li>
            <li><em>“File Type”</em>: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
          </ul>
        </li>
        <li>
          <p>Click on <code class="language-plaintext highlighter-rouge">Paste/Fetch data</code> and copy the following links over to the empty text box</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://use<span class="notranslate">galaxy</span>.eu/api/datasets/4838ba20a6d86765171e4f201205515c/display?to_ext=fastqsanger.gz
https://use<span class="notranslate">galaxy</span>.eu/api/datasets/4838ba20a6d867657d09280baae4e1ec/display?to_ext=fastqsanger.gz
</code></pre></div>          </div>

          <p>and press <code class="language-plaintext highlighter-rouge">Start</code>.</p>
        </li>
        <li>Wait for the <code class="language-plaintext highlighter-rouge">Build</code> button to become enabled, click it, and, in the next dialogue, give a suitable <strong>Name</strong> to the new collection.</li>
      </ol>
    </li>
    <li>Import the reference data collection into your history
      <ol>
        <li>Open the <a href="https://usegalaxy.eu/u/wolfgang-maier/h/influenza-resources">shared history</a> with the pre-prepared INSAFlu reference data</li>
        <li>Click on <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>“Import”</em> in the upper right corner of the page</li>
        <li>Choose a name for the new history (or go with the default), confirm by clicking Import and wait for the page to reload</li>
        <li>In the history panel (which should now show the newly imported history),
click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon in the bar just above the history’s datasets,
select <code class="language-plaintext highlighter-rouge">Copy Datasets</code>, and wait for the center panel to refresh</li>
        <li>In the center panel,
          <ul>
            <li>make sure the imported history is selected as the <em>“Source History”</em></li>
            <li>select the history you created for this tutorial as the <em>“Destination History”</em></li>
            <li>under <em>“Source History”</em> select the <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code> collection for copying</li>
            <li>click <em>“Copy History Items”</em></li>
          </ul>
        </li>
        <li>Switch to the tutorial history</li>
      </ol>
    </li>
  </ol>

</blockquote>

<h2 id="inspect-the-reference-data">Inspect the reference data</h2>

<blockquote class="notranslate hands_on">
  <div id="hands-on-what-is-in-the-reference-collection" class="box-title" aria-label="hands-on box: What is in the reference collection?"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: What is in the reference collection?</div>

  <ol>
    <li>Step inside the uploaded reference collection by clicking on it.</li>
    <li>Expand individual elements of the collection by clicking on them to reveal more details about them.</li>
    <li>
      <p>View the content of any element by clicking its <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span>.</p>

      <p>You can then scroll through the data and use your browser’s search function to look for particular sequences.</p>

      <blockquote class="question">
        <div id="question-questions" class="box-title" aria-label="question box: Questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question: Questions</div>

        <ol>
          <li>How many elements are in the collection? What do they represent?</li>
          <li>How many sequences are stored in each element?</li>
          <li>Are all subtypes of HA and NA represented in the sequences?</li>
          <li>Are there sequences of non-A Influenza species?</li>
        </ol>

        <blockquote class="solution">
          <div id="solution-answers" class="box-title"><button type="button" aria-controls="solution-answers-contents" aria-expanded="true" aria-label="Toggle solution box: Answers"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answers<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The collection consists of eight elements and each element provides reference sequences for one genome segment.</li>
            <li>Each element has data for 56 sequences (<span class="notranslate">Galaxy</span> knows how to interpret the fasta format of the sequencing data and counts sequences for you; you can see the count for any of the elements by expanding it).</li>
            <li>The collection has reference sequence data for H1-H18 (with the exception of H11) and for N1-N11. For many of those subtypes, however, there is only one reference, i.e., within-subtype variation is not captured well by the collection.</li>
            <li>There are 6 <em>Influenza B</em> references in the collection. Since <em>Influenza B</em> is not normally seen in birds and we are going to analyze an avian influenza sample here, these can be considered controls: if we get an <em>Influenza B</em> assignment for the sample at any step in our analysis, we know something is very suspicious. This also means that we only have 50 <em>informative</em> references in the collection.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-info</span> There are lots of text processing and filtering tools for <span class="notranslate">Galaxy</span>, of which this tutorial is going to introduce just a small subset. You could use them to extract statistics about which subtypes are found how many times in the collection of references instead of relying on scrolling and searching. If you are interested and have the time, you may want to try the <a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">Data Manipulation Olympics</a> to learn more about available such tools and how to combine them.</p>

<h1 id="quality-control">Quality control</h1>

<p>As a very first step, we would like to make sure that we base our analysis only on the high-quality parts of the sequenced <span class="notranslate">reads</span>.</p>

<p>NGS <span class="notranslate">reads</span> often have lower base calling quality near their ends, so here we are going to trim low-quality stretches of bases from both ends. In addition, we are going to discard <span class="notranslate">reads</span> shorter than 30 bases after trimming. This is reasonable since our next step will involve extracting all possible 21-mers from the <span class="notranslate">reads</span> and there are very few possibilities for a read shorter than 30 bases.</p>

<p>The tool <strong>fastp</strong> lets us perform these tasks and obtain a nice quality report for our <span class="notranslate">reads</span> before and after processing in one go, but many other options exist to perform sequenced <span class="notranslate">reads</span> quality control and trimming/filtering in <span class="notranslate">Galaxy</span> and the dedicated tutorial on <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">quality control</a> introduces more of them.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-qc-and-read-trimming-filtering-with-fastp" class="box-title" aria-label="hands-on box: QC and read trimming/filtering with fastp"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: QC and read trimming/filtering with fastp</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><em>“Single-end or paired <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code></li>
        <li><em>“Select paired collection(s)”</em>: the uploaded paired collection of sequenced <span class="notranslate">reads</span></li>
        <li>In <em>“Filter Options”</em> under <em>“Length filtering options”</em>:
          <ul>
            <li><em>“Length required”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em> under <em>“Per read cutting by quality options”</em>:
          <ul>
            <li><em>“Cut by quality in front (5’)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Cut by quality in tail (3’)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Cutting mean quality”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
      </ul>

      <p>Running the tool will result in two outputs. One is a new paired collection with the processed <span class="notranslate">reads</span>, the other one is a report of initial quality, the processing actions performed and their effect on key quality metrics.</p>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> <strong>Inspect</strong> the report and try to answer the following questions:</p>

      <blockquote class="question">
        <div id="question-questions-1" class="box-title" aria-label="question box: Questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question: Questions</div>

        <ol>
          <li>Which set of <span class="notranslate">reads</span> (forward or reverse <span class="notranslate">reads</span>) did profit most quality-wise from our low-quality base trimming?</li>
          <li>What percentage of <span class="notranslate">reads</span> got discarded completely with our settings, and what percentage of bases?</li>
          <li>Why, according to the Filtering result, have some <span class="notranslate">reads</span> been discarded because of too many Ns?</li>
        </ol>

        <blockquote class="solution">
          <div id="solution-answers-1" class="box-title"><button type="button" aria-controls="solution-answers-1-contents" aria-expanded="true" aria-label="Toggle solution box: Answers"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answers<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The forward <span class="notranslate">reads</span> were of worse quality in particular near their 3’-ends than the reverse <span class="notranslate">reads</span> and, consequently, were affected more strongly by trimming (see the different “quality” plots in the report).</li>
            <li>Less than 2% of <span class="notranslate">reads</span> got discarded completely (the “Filtering result” section of the report says 98.03% of all <span class="notranslate">reads</span> were retained). In contrast, around 10% of all bases got discarded (compare total bases before and after filtering) so most <span class="notranslate">reads</span> got trimmed but not discarded.</li>
            <li><strong>fastp</strong> is a tool with many hidden default actions in its various sections. Inside the <em>Filter Options</em> check <em>Quality filtering options</em> and the various defaults mentioned in the parameters help therein. One of these defaults is to discard <span class="notranslate">reads</span> with more than 5 Ns. These defaults are often reasonable, but it’s good to be aware of them.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="per-segment-subtyping-and-hybrid-reference-construction">Per-segment subtyping and hybrid reference construction</h1>

<p>Our goal at this step is to find best matches between our sequencing data and the reference sequences of each genome segment. This will:</p>
<ul>
  <li>give us preliminary subtyping results with regard to the HA and NA segments for the sequenced sample</li>
  <li>suggest the best reference to map the sequencing data to for each segment.</li>
</ul>

<p>We are then going to combine the best reference of each segment into a hybrid reference genome to use for <span class="notranslate">mapping</span> our sequenced <span class="notranslate">reads</span> against.</p>

<p>To identify the best matching reference segments, we are going to run the tool <strong>VAPOR</strong> asking it to report the stats for any hits it identifies.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-exploring-best-matching-reference-segment-scores" class="box-title" aria-label="hands-on box: Exploring best-matching reference segment scores"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Exploring best-matching reference segment scores</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3"><strong>VAPOR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+<span class="notranslate">galaxy</span>3</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Reference sequences”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li><em>“Type of sequencing data”</em>: <code class="language-plaintext highlighter-rouge">Paired-end as collection</code></li>
        <li><em>“Paired collection of sequenced <span class="notranslate">reads</span>”</em>: quality-trimmed <span class="notranslate">reads</span>; output of <strong>fastp</strong></li>
        <li><em>“Desired output”</em>: <code class="language-plaintext highlighter-rouge">Return scores of best matches</code></li>
        <li><em>“Limit number of reported matches to”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li>In <em>“Optional arguments”</em>:
          <ul>
            <li><em>“Read kmer filtering threshold”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
            <li><em>“Minimum k-mer proportion”</em>: <code class="language-plaintext highlighter-rouge">0.0</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> Explore the output collection produced by the tool</p>

      <blockquote class="question">
        <div id="question-questions-2" class="box-title" aria-label="question box: Questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question: Questions</div>

        <ol>
          <li>Is there a difference between the results for segment 4 (encoding HA), segment 6 (encoding NA) and those for the remaining segments?</li>
          <li>According to the tool, what is the likely subtype with regard to HA and NA of the sample?</li>
        </ol>

        <blockquote class="solution">
          <div id="solution-answers-2" class="box-title"><button type="button" aria-controls="solution-answers-2-contents" aria-expanded="true" aria-label="Toggle solution box: Answers"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answers<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The values for <code class="language-plaintext highlighter-rouge">% of query bases in <span class="notranslate">reads</span></code> and for <code class="language-plaintext highlighter-rouge">Total score</code> are drastically lower for the HA and somewhat lower for the NA segment than for any of the other segments. This could be due to the structure of the collection of references we have used, but also fits very well with the higher selection pressure on NA and, in particular on HA, as the main antigenic proteins of the virus, which leads to higher variability in these segments than in the others.</li>
            <li>
              <p>The best match (assigned an almost 2x higher score than the second-best match) found for the HA segment is from an H4N6 strain. The top two matches with regard to the NA segment (again assigned ~ 2x higher scores than the next best matches) are from strains with the N6 subtype of NA.</p>

              <p>The most likely subtype of the sample, thus, appears to be H4N6.</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Now that we have established that things <em>may</em> make sense, we can use the output of <strong>VAPOR</strong> to extract the actual sequence of the top hit for each reference segment. We then concatenate these best matches into a hybrid reference genome for <span class="notranslate">mapping</span>.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-obtaining-sequences-of-top-hits-identified-by-vapor" class="box-title" aria-label="hands-on box: Obtaining sequences of top hits identified by VAPOR"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Obtaining sequences of top hits identified by VAPOR</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>  to extract just the name of the sequence from each line of VAPOR’s output
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to process”</em>: collection of score outputs of <strong>VAPOR</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^.+\t&gt;(.+)$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">$1</code></li>
            <li><em>“Find-Pattern is a regular <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Show beginning1" title="Tested with Show beginning1"><strong>Select first lines from a dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Show beginning1</span></span>  to get the first line, i.e. the best match from VAPOR’s output
      <ul>
        <li><em>“Select first”</em>: <code class="language-plaintext highlighter-rouge">1</code> lines</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“from”</em>: output of <strong>Replace</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.3.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.3.1"><strong>seqtk_subseq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.3.1</span></span>  to extract the reference sequences based on their names reported by VAPOR
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input FASTA/Q file”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li><em>“Select source of sequence choices”</em>: <code class="language-plaintext highlighter-rouge">FASTA/Q ID list</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input ID list”</em>: the collection output of <strong>Select first</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</span></span>  to combine the best-matching sequence of each segment into one dataset
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>: the selected sequences collection produced by <strong>seqtk_subseq</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>At this point, you may want to <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> inspect the output of the last step to see if it is an eight-segments reference genome as expected, and if the segments correspond to the top hits found by VAPOR.</p>

<h1 id="mapping-to-a-hybrid-reference"><span class="notranslate">Mapping</span> to a hybrid reference</h1>

<p>If things went well, the hybrid reference we just obtained should be close enough across all segments to our sample to allow successful <span class="notranslate">mapping</span> of <span class="notranslate">reads</span>. Before we start the <span class="notranslate">mapping</span> we may want to truncate the segment names in our hybrid reference genome though because currently these names still reflect the full origin of the segment sequences, but from now on we are fine with just the segment ID. We can use the <strong>Replace</strong> tool from before again to truncate the names.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-shortening-sequence-titles" class="box-title" aria-label="hands-on box: Shortening sequence titles"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Shortening sequence titles</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: the hybrid reference genome; output of <strong>Collapse Collection</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;([^|]+).+$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">&gt;$1</code></li>
            <li><em>“Find-Pattern is a regular <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Having polished the titles of the segments in our hybrid reference genome we are finally ready for <span class="notranslate">mapping</span>, which we will carry out with <strong>BWA-MEM</strong>, clean up a bit with <strong>Samtools view</strong> and produce a quality report for with <strong>QualiMap BamQC</strong>.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-read-mapping-and-quality-control" class="box-title" aria-label="hands-on box: Read mapping and quality control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Read <span class="notranslate">mapping</span> and quality control</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: hybrid reference genome with shortened names; output of <strong>Replace</strong></li>
        <li><em>“Single or Paired-end <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code></li>
        <li><em>“Select a paired collection”</em>: quality-trimmed <span class="notranslate">reads</span>; output of <strong>fastp</strong> in QC and Trimming</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SAM/BAM/CRAM data set”</em>: mapped <span class="notranslate">reads</span> BAM dataset; output of <strong>BWA-MEM</strong></li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of <span class="notranslate">reads</span></code></li>
        <li>In <em>“Configure filters”</em>:
          <ul>
            <li><em>“Filter by quality”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>“Require that these flags are set”</em>: <code class="language-plaintext highlighter-rouge">Read is paired</code> and <code class="language-plaintext highlighter-rouge">Read is mapped in a proper pair</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3"><strong>QualiMap BamQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+<span class="notranslate">galaxy</span>3</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mapped <span class="notranslate">reads</span> input dataset”</em>: filtered mapped <span class="notranslate">reads</span> BAM dataset; output of <strong>Samtools view</strong></li>
        <li><em>“Skip duplicate <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Unselect all</code></li>
        <li>In <em>“Settings affecting specific plots”</em>:
          <ul>
            <li><em>“Number of bins to use in across-reference plots”</em>: <code class="language-plaintext highlighter-rouge">40</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> Study the report generated with QualiMap</p>

      <blockquote class="question">
        <div id="question-questions-3" class="box-title" aria-label="question box: Questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question: Questions</div>

        <ol>
          <li>What is the coverage of each segment by the sequenced <span class="notranslate">reads</span>, and is it uniform?</li>
          <li>Look for a plot showing read <span class="notranslate">mapping</span> quality across the reference. What can you conclude?</li>
        </ol>

        <blockquote class="solution">
          <div id="solution-answers-3" class="box-title"><button type="button" aria-controls="solution-answers-3-contents" aria-expanded="true" aria-label="Toggle solution box: Answers"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answers<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>Coverage is rather different for the different segments. Looking at “Mean coverage” and its “Standard deviation” in the “Chromosome stats” table and at the “Coverage across reference” plot, coverage of the HA segment seems to be most critical since it approaches zero in some regions.</li>
            <li><span class="notranslate">Mapping</span> quality is almost constant at or very near 60 (which happens to be the maximum <span class="notranslate">mapping</span> quality value emitted by BWA-MEM) across all segments with the exception of HA.</li>
          </ol>

          <p>These two observations combined with the VAPOR statistics for HA show again that even the best matching reference for this segment is not exactly close to the sequence of our sample. The read mapper had difficulties placing the sequenced <span class="notranslate">reads</span> on that sub-optimal reference and it looks as if we might have very little sequence information for some HA regions.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="consensus-sequence-construction">Consensus sequence construction</h1>

<p>From the polished <span class="notranslate">mapping</span> of <span class="notranslate">reads</span> to our custom reference we can now construct the consensus sequence of our sample.</p>

<p>Unfortunately, the tool we are going to use for this, <strong>ivar consensus</strong>, is not capable of working with more than one reference name at a time, but because this is influenza data we have <span class="notranslate">mapping</span>s to 8 different segments described in our data. So we need to take a little detour and split the mapped <span class="notranslate">reads</span> data into a collection of datasets each containing the <span class="notranslate">mapping</span>s for just one segment first again, then perform the consensus construction for all of them in parallel.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-splitting-mapped-reads-by-genome-segment" class="box-title" aria-label="hands-on box: Splitting mapped reads by genome segment"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Splitting mapped <span class="notranslate">reads</span> by genome segment</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.1+galaxy0"><strong>Split BAM by Reference</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.1+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM dataset to split by reference”</em>: filtered mapped <span class="notranslate">reads</span>; output of <strong>Samtools view</strong></li>
        <li><em>“Select references (chromosomes and contigs) you would like to restrict bam to”</em>: <code class="language-plaintext highlighter-rouge">Unselect all</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output from this step has the desired collection structure, but the names of the collection elements are not the nicest. Ideally, we would just reuse the segment names, which are already provided in our <span class="notranslate">mapping</span> reference genome. So lets extract these names again and use them as new element labels</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-relabeling-collection-elements" class="box-title" aria-label="hands-on box: Relabeling collection elements"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Relabeling collection elements</div>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: the <span class="notranslate">mapping</span> reference; output of <strong>Replace</strong> in <span class="notranslate">Mapping</span> to a hybrid reference</li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;.+</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output of <strong>Select</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;(.+)$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">$1</code></li>
            <li><em>“Find-Pattern is a regular <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="__RELABEL_FROM_FILE__" title="Tested with __RELABEL_FROM_FILE__"><strong>Relabel identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>RELABEL_FROM_FILE</strong></span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input collection”</em>: the split <span class="notranslate">mapping</span>s; output of <strong>Split BAM by Reference</strong></li>
        <li><em>“How should the new labels be specified?”</em>: <code class="language-plaintext highlighter-rouge">Using lines in a simple text file.</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“New Identifiers”</em>: output of <strong>Replace</strong></li>
        <li><em>“Ensure strict <span class="notranslate">mapping</span>”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>And with that we are ready for consensus sequence generation!</p>

<p>To accept any base suggested by the mapped sequenced <span class="notranslate">reads</span> as the consensus base for the corresponding genome position, we ask for the following requirements to be fulfilled:</p>

<ul>
  <li>at least ten sequenced <span class="notranslate">reads</span> have to provide information about the base in question</li>
  <li>at a minimum, 70% of these <span class="notranslate">reads</span> have to agree on the base at this position.</li>
</ul>

<p>To avoid getting misled too much by sequencing errors, we are also going to ignore bases with a base calling quality less than 20 in the above counts (i.e., we are going to base our decisions only on bases in sequenced <span class="notranslate">reads</span> that the basecaller of the sequencer was reasonably sure about.</p>

<p>Now what if we cannot obtain a consensus base for a position with the above criteria? In such cases of uncertainty we want to insert an N (i.e. an unknown base) to express that we either did not have enough information about the position or that this information was ambiguous.</p>

<p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-info</span> All of the above limits for consensus base calling are arbitrary to some degree, and depend somewhat on the quality of the sequencing data. With very high overall coverage, for example, it is possible to increase the coverage threshold, but if you increase that threshold too much, you may end up with a consensus sequence consisting mostly of Ns.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-per-segment-consensus-construction" class="box-title" aria-label="hands-on box: Per-segment consensus construction"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Per-segment consensus construction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.1+galaxy0"><strong>ivar consensus</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.1+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Bam file”</em>: the relabeled collection of mapped <span class="notranslate">reads</span>; output of <strong>Relabel identifiers</strong></li>
        <li><em>“Minimum quality score threshold to count base”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
        <li><em>“Minimum frequency threshold”</em>: <code class="language-plaintext highlighter-rouge">0.7</code></li>
        <li><em>“Minimum depth to call consensus”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
        <li><em>“Exclude regions with smaller depth than the minimum threshold”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Use N instead of - for regions with less than minimum coverage”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>The output is a consensus sequence in FASTA format, one per segment, with
the names just providing a bit too much detail for our purpose.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to process”</em>: consensus sequences; output of <strong>ivar consensus</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;Consensus_(.*)_threshold_.*</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">&gt;$1</code></li>
            <li><em>“Find-Pattern is a regular <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> Inspect each consensus sequence generated for the different segments</p>

      <blockquote class="question">
        <div id="question" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

        <p>Does everything look ok?</p>

        <blockquote class="solution">
          <div id="solution-answer" class="box-title"><button type="button" aria-controls="solution-answer-contents" aria-expanded="true" aria-label="Toggle solution box: Answer"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answer<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>As expected from the findings so far, the consensus sequence for the HA segment has stretches of Ns in it, which likely reflect the <span class="notranslate">mapping</span> issues and associated loss of coverage caused by our insufficiently sized collection of references.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="placing-segments-on-a-phylogenetic-tree">Placing segments on a phylogenetic tree</h1>

<p>The next logical step after obtaining the consensus sequences of segments of our sample is to explore how those sequences are related to the sequences in our reference collection.
To do so, we are going to build multiple sequence alignments (MSAs) per segment of all reference sequences, add our sample segments to those alignments, and build phylogenetic trees, one per segment. We are going to use two rather standard tools, <strong>MAFFT</strong> and <strong>IQTree</strong>, for generating MSAs and trees, respectively.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-exploring-phylogeny" class="box-title" aria-label="hands-on box: Exploring phylogeny"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Exploring phylogeny</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.508+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.508+galaxy0"><strong>MAFFT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">rna</span>team/mafft/rbc_mafft/7.508+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequences to align”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li><em>“Data type”</em>: <code class="language-plaintext highlighter-rouge">Nucleic Acids</code></li>
        <li><em>“Matrix selection”</em>: <code class="language-plaintext highlighter-rouge">No matrix</code></li>
      </ul>

      <p>The result is a collection of MSAs, each representing all reference sequences of one genome segment.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft_add/7.508+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft_add/7.508+galaxy0"><strong>MAFFT add</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">rna</span>team/mafft/rbc_mafft_add/7.508+<span class="notranslate">galaxy</span>0</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequences to add to the alignment”</em>: consensus sequences with simplified names; output of <strong>Replace</strong></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Alignment”</em>: collection of MSAs; output of MAFFT</li>
        <li><em>“What do you want to add to the alignment”</em>: <code class="language-plaintext highlighter-rouge">A single sequence</code></li>
        <li><em>“Keep alignment length”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <p>The result is a new collection with each of our sample consensus sequences added to the respective segment MSA.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.1.2+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.1.2+galaxy2"><strong>IQTree</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.1.2+<span class="notranslate">galaxy</span>2</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Specify input alignment file in PHYLIP, FASTA, NEXUS, CLUSTAL or MSF format.”</em>: output of <strong>MAFFT add</strong></li>
        <li><em>“Specify sequence type …“</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> Explore each of the final trees produced by IQTree for the different segments</p>

      <blockquote class="question">
        <div id="question-1" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

        <p>What are your conclusions about the sample in general and its HA and NA segments in particular?</p>

        <blockquote class="solution">
          <div id="solution-answers-4" class="box-title"><button type="button" aria-controls="solution-answers-4-contents" aria-expanded="true" aria-label="Toggle solution box: Answers"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution: Answers<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

          <ul>
            <li>For most of its segments the sample resembles relatively recent (from the last decade) Eurasian reference sequences.</li>
            <li>For HA and NA the sample clusters with the few available samples of the corresponding subtype.</li>
            <li>None of the references closest to the sample with respect to HA and NA are close to the recent Eurasian reference cluster for their remaining segments.</li>
            <li>A plausible explanation is that the H4 and N6 segments of the sample have been brought into the recent Eurasian background through a reassortment event. Caveat: interpretations like this can be heavily influenced by the size of the reference collection!</li>
          </ul>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Analysis workflows for influenza whole-genome sequencing data need to take into account the specific characteristics of the viral genome. Due to their higher natural variability this is especially true for avian influenza samples and for the HA- and NA-encoding segments of the genome.</p>

<p>Nevertheless, it looks possible, with carefully chosen reference segment sequences and bioinformatic tools, to avoid a computationally expensive de-novo assembly approach and to use <span class="notranslate">mapping</span> against a dynamically compiled reference genome instead.</p>

<p>The rather small reference segment collection suggested for this tutorial consists of 56 different samples, of which only a single one has the H4N6 subtype of the sample analyzed here. Still it allowed us to perform subtyping of the sample, to construct complete consensus sequences for 7 segments including NA, and to draw valuable conclusions about the origin of the sample. It is conceivable that a larger collection of references chosen to capture several strains from each HA subtype could solve the remaining issue of the incomplete HA consensus sequence.</p>


                
                <blockquote class="key_points">
                    <div class="box-title" aria-label="key-points box"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Reassortment of gene segments makes reference-based mapping of influenza sequencing data challenging</p>
</li>
                        
                        <li><p>An alternative to <em>de-novo</em> assembly can be mapping to a dynamically chosen reference genome</p>
</li>
                        
                        <li><p>Variant calling and consensus genome construction can follow workflows used also for other viral sequence data</p>
</li>
                        
                        <li><p>Standard phylogenetic tools can be used to find relationships between influenza samples but should be used on a per-segment basis</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Avian influenza viral strain analysis from gene segment sequencing data (Variant Analysis)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier,  <b>Avian influenza viral strain analysis from gene segment sequencing data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false" aria-label="Toggle details box: BibTeX for citing this tutorial">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-aiv-analysis,
author = "Wolfgang Maier",
title = "Avian influenza viral strain analysis from gene segment sequencing data (Galaxy Training Materials)",
year = "",
month = "",
day = ""
url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
        <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
        <script type="text/javascript">
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>
