<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Assembly of metagenomic sequencing data</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html" />
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/" />

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material" />
<meta name="DC.type" content="text" />
<meta name="DC.title" content="Assembly of metagenomic sequencing data" />
<meta name="DC.publisher" content="Galaxy Training Network" />
<meta name="DC.date" content="2022-12-01 15:50:31 +0000" />
<meta name="DC.creator" content="Polina Polunina" />
<meta name="DC.creator" content="Bérénice Batut" />

        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:site_name" content="Galaxy Training Network" />
        <meta property="og:title" content="Galaxy Training: Assembly of metagenomic sequencing data" />
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2022-12-01 15:50:31 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Assembly of metagenomic sequencing data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Assembly of metagenomic sequencing data' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html",
  "timeRequired": "PT2H",
  "keywords": "metagenomics, assembly, metagenomics",
  "description": "The questions this  addresses are:\n - Why metagenomic data should be assembled?\n - What is the difference between co-assembly and individual assembly?\n - What is the difference between reads, contigs and scaffolds?\n - How tools based on De Bruijn graph work?\n - How to assess the quality of metagenomic data assembly?\n\n\nThe objectives are:\n - Describe what an assembly is\n - Describe what de-replication is\n - Explain the difference between co-assembly and individual assembly\n - Explain the difference between reads, contigs and scaffolds\n - Explain how tools based on De Bruijn graph work\n - Apply appropriate tools for analyzing the quality of metagenomic data\n - Construct and apply simple assembly pipelines on short read data\n - Apply appropriate tools for analyzing the quality of metagenomic assembly\n - Evaluate the Quality of the Assembly with Quast, Bowtie2, and CoverM-Genome\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ],
  "educationalLevel": "Beginner"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc">
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                    <button class="btn" data-theme="default">Default</button>
                    <button class="btn" data-theme="night">Night</button>
                    <button class="btn" data-theme="midnight">Midnight</button>
                    <button class="btn" data-theme="rainbow">Rainbow</button>
                    <button class="btn" data-theme="blm">✊🏿</button>
                    <button class="btn" data-theme="progress" >🏳️‍🌈</button>
                    <button class="btn" data-theme="halloween">🎃</button>
                    <button class="btn" data-theme="straya">🇦🇺</button>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-05-01/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html" title="Version 2022-05-01">2022-05-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-04-01/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html" title="Version 2022-04-01">2022-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<section class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Assembly of metagenomic sequencing data</h1>
    

    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/plushz/" class="contributor-badge contributor-plushz"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/plushz?s=27" alt="Avatar">Polina Polunina</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/bebatut?s=27" alt="Avatar">Bérénice Batut</a>
		</td>
	</tr>
	


	

	

	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div class="box-title" aria-label="overview box">Overview</div>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Why metagenomic data should be assembled?</p>
</li>
        
        <li><p>What is the difference between co-assembly and individual assembly?</p>
</li>
        
        <li><p>What is the difference between reads, contigs and scaffolds?</p>
</li>
        
        <li><p>How tools based on De Bruijn graph work?</p>
</li>
        
        <li><p>How to assess the quality of metagenomic data assembly?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Describe what an assembly is</p>
</li>
        
        <li><p>Describe what de-replication is</p>
</li>
        
        <li><p>Explain the difference between co-assembly and individual assembly</p>
</li>
        
        <li><p>Explain the difference between reads, contigs and scaffolds</p>
</li>
        
        <li><p>Explain how tools based on De Bruijn graph work</p>
</li>
        
        <li><p>Apply appropriate tools for analyzing the quality of metagenomic data</p>
</li>
        
        <li><p>Construct and apply simple assembly pipelines on short read data</p>
</li>
        
        <li><p>Apply appropriate tools for analyzing the quality of metagenomic assembly</p>
</li>
        
        <li><p>Evaluate the Quality of the Assembly with Quast, Bowtie2, and CoverM-Genome</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/6958893">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/metagenomics/tutorials/metagenomics-assembly/workflows/" title="Workflows" alt="Assembly of metagenomic sequencing data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/metagenomics/metagenomics-assembly/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (May 2023)
            </a>
        </li>
        
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/metagenomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Dec 1, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 id="introduction">Introduction</h1>

<p>Metagenomics involves the extraction, sequencing and analysis of combined genomic <span class="notranslate">DNA</span> from <strong>entire microbiome</strong> samples. It includes then <span class="notranslate">DNA</span> from <strong>many different organisms</strong>, with different taxonomic background.</p>

<p>Reconstructing the genomes of microorganisms in the sampled communities is critical step in analyzing metagenomic data. To do that, we can use <strong>assembly</strong> and assemblers, <em>i.e.</em> computational programs that stich together the small fragments of sequenced <span class="notranslate">DNA</span> produced by sequencing instruments.</p>

<p>Assembling seems intuitively similar to putting together a jigsaw puzzle. Essentially, it looks for <span class="notranslate">reads</span> “that work together” or more precisely, <span class="notranslate">reads</span> that overlap. Tasks like this are <strong>not straightforward</strong>, but rather complex because of the complexity of the genomics (specially the repeats), the missing pieces and the errors introduced during sequencing.</p>

<blockquote class="comment">
  <div id="comment" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>

  <p>Do you want to learn more about the principles behind single genome assembly? Follow our <a href="/training-material/topics/assembly/index.html">tutorials</a>.</p>
</blockquote>

<p><strong>Metagenomic assembly</strong> is further complicated by</p>
<ul>
  <li>the large volume of data produced</li>
  <li>the quality of the sequence</li>
  <li>the unequal representation of members of the microbial community</li>
  <li>the presence of closely related microorganisms with similar genomes</li>
  <li>the presence of several strains of the same microorganism</li>
  <li>an insufficient amount of data for minor community members</li>
</ul>

<blockquote class="details">
  <div id="details-strategies-for-assembly" class="box-title"><button type="button" aria-controls="details-strategies-for-assembly-contents" aria-expanded="true" aria-label="Toggle details box: Strategies for assembly"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Details: Strategies for assembly<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>For assembly, there are 3 main strategies:</p>
  <ol>
    <li>Greedy extension</li>
    <li>Overlap Layout Consensus</li>
    <li>De Bruijn graphs. The following figure illustrates these strategies in brief.</li>
  </ol>

  <figure id="figure-1"><img src="./images/assembly-algorithms.png" alt="Image shows greedy extention, overlap layout consensus, and de Brujin graphs assembly algorithms. " width="70%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Assembly algorithms. Image from <span class="citation"><a href="#carpentries">“<b>Metagenome Assembly – Data Processing and Visualization for Metagenomics</b>”</a></span></figcaption></figure>

  <p>The nice paper <span class="citation"><a href="#miller2010">Miller <i>et al.</i> 2010</a></span> on assemblers based on these algorithms will help you to better understand how they work.</p>
</blockquote>

<p>For metagenomic assembly, several tools exist: metaSPAdes (<span class="citation"><a href="#nurk2017">Nurk <i>et al.</i> 2017</a></span>), MEGAHIT (<span class="citation"><a href="#li2015">Li <i>et al.</i> 2015</a></span>), etc. The different assemblers have different computational characteristics and their performance varies according to the microbiome as shown in by both rounds of  Critical Assessment of Metagenome Interpretation initiative (<span class="citation"><a href="#sczyrba2017">Sczyrba <i>et al.</i> 2017</a></span>, <span class="citation"><a href="#meyer2022">Meyer <i>et al.</i> 2022</a></span>, <span class="citation"><a href="#meyer2021">Meyer <i>et al.</i> 2021</a></span>). The preference of one assembler over another depends on the purpose at hand.</p>

<p>In this tutorial, we will learn how to run metagenomic assembly tool and evaluate the quality of the generated assemblies. To do that, we will use data from the study: <a href="https://www.ebi.ac.uk/metagenomics/studies/MGYS00005630#overview">Temporal shotgun metagenomic dissection of the coffee fermentation ecosystem</a>. For an in-depth analysis of the structure and functions of the coffee microbiome, a temporal shotgun metagenomic study (six time points) was performed. The six samples have been sequenced with Illumina MiSeq utilizing whole genome sequencing.</p>

<p>Based on the 6 original dataset of the coffee fermentation system, we generated mock datasets for this tutorial.</p>

<blockquote class="agenda">
  <div id="agenda" class="box-title" aria-label="agenda box: ">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#prepare-analysis-history-and-data" id="markdown-toc-prepare-analysis-history-and-data">Prepare analysis history and data</a></li>
  <li><a href="#assembly" id="markdown-toc-assembly">Assembly</a></li>
  <li><a href="#quality-control-of-assembly" id="markdown-toc-quality-control-of-assembly">Quality control of assembly</a></li>
  <li><a href="#visualization-of-the-de-novo-assembly-graph" id="markdown-toc-visualization-of-the-de-novo-assembly-graph">Visualization of the <em>de novo</em> assembly graph</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="prepare-analysis-history-and-data">Prepare analysis history and data</h1>

<p>To run assembly, we first need to get the data into <span class="notranslate">Galaxy</span>. Any analysis should get its own <span class="notranslate">Galaxy</span> history. So let’s start by creating a new one:</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-prepare-the-galaxy-history" class="box-title" aria-label="hands-on box: Prepare the Galaxy history"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Prepare the <span class="notranslate">Galaxy</span> history</div>

  <ol>
    <li>
      <p>Create a new history for this analysis</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-creating-a-new-history" class="box-title"><button type="button" aria-controls="tip-creating-a-new-history-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a new history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
    </li>
    <li>
      <p>Rename the history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-renaming-a-history" class="box-title"><button type="button" aria-controls="tip-renaming-a-history-contents" aria-expanded="true" aria-label="Toggle tip box: Renaming a history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
    </li>
  </ol>

</blockquote>

<p>We need to get the data into our history.</p>

<p>In case of a not very large dataset it’s more convenient to upload data directly from your computer to <span class="notranslate">Galaxy</span>.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-upload-data-into-galaxy" class="box-title" aria-label="hands-on box: Upload data into Galaxy"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Upload data into <span class="notranslate">Galaxy</span></div>

  <ol>
    <li>
      <p>Import the sequence read raw data (*.fastqsanger.gz) from <a href="https://zenodo.org/record/6958893">Zenodo</a> or a data library:</p>

      <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6958893/files/ERR2231567_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231567_Read_2.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231568_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231568_Read_2.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231569_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231569_Read_2.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231570_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231570_Read_2.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231571_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231571_Read_2.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231572_Read_1.fastqsanger.gz
https://zenodo.org/record/6958893/files/ERR2231572_Read_2.fastqsanger.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-importing-via-links" class="box-title"><button type="button" aria-controls="tip-importing-via-links-contents" aria-expanded="true" aria-label="Toggle tip box: Importing via links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>   </li>   <li>Select <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-importing-data-from-a-data-library" class="box-title"><button type="button" aria-controls="tip-importing-data-from-a-data-library-contents" aria-expanded="true" aria-label="Toggle tip box: Importing data from a data library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ul>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor</li>   <li>Select the desired files</li>   <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>   <li>Click on <strong>Import</strong></li> </ul> </blockquote>

      <blockquote class="comment">
        <div id="comment-1" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>
        <p>In case of large dataset, we can use FTP server or the <a href="/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"><span class="notranslate">Galaxy</span> Rule-based Uploader</a>.</p>
      </blockquote>
    </li>
    <li>
      <p>Create a paired collection named <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code>, rename your pairs with the sample name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div id="tip-creating-a-paired-collection" class="box-title"><button type="button" aria-controls="tip-creating-a-paired-collection-contents" aria-expanded="true" aria-label="Toggle tip box: Creating a paired collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a paired collection<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>For all selected..</strong> and choose <strong>Build List of Dataset Pairs</strong></p>   </li>   <li>Change the text of <em>unpaired forward</em> to a common selector for the forward <span class="notranslate">reads</span></li>   <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse <span class="notranslate">reads</span></li>   <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="assembly">Assembly</h1>

<p>As explained before, there are many challenges to metagenomics assembly, including:</p>

<ol>
  <li>differences in coverage between samples, resulting from differences in abundance,</li>
  <li>the fact that different species often share conserved regions (<span class="citation"><a href="#kececioglu2001">Kececioglu and Ju 2001</a></span>), and</li>
  <li>the presence of multiple strains of a single species (<span class="citation"><a href="#miller2010">Miller <i>et al.</i> 2010</a></span>).</li>
</ol>

<p>To reduce the differences in coverage between samples, we can use a <strong>co-assembly</strong> approach, where <span class="notranslate">reads</span> from all samples are aligned together.:</p>

<figure id="figure-2"><img src="./images/co-assembly.png" alt="Image show one pile of sample1 reads and another pile of sample2 reads, then, green arrow leads to assembled reads from both piles. " width="60%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Co-assembly</figcaption></figure>

<table>
  <thead>
    <tr>
      <th>Pros of co-assembly</th>
      <th>Cons of co-assembly</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>More data</td>
      <td>Higher computational overhead</td>
    </tr>
    <tr>
      <td>Better/longer assemblies</td>
      <td>Risk of shattering your assembly</td>
    </tr>
    <tr>
      <td>Access to lower abundant organisms</td>
      <td>Risk of increased contamination</td>
    </tr>
  </tbody>
</table>

<p>Co-assembly is then not always beneficial:</p>
<ul>
  <li>Changes in strain can cause the assembly graph to collapse</li>
  <li>Binned contigs are likely to be misclassified: MAGs must be treated as a population genome.</li>
</ul>

<figure id="figure-3"><img src="./images/co-assembly-process.png" alt="Image shows the process of assembled reads from 2 samples followed by binning and there is detailed information about broken contigs, dominant coverage, chimeric contigs found during assembly process. " width="80%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Co-assembly process</figcaption></figure>

<p>In these cases, co-assembly is reasonable if:</p>
<ul>
  <li>Same samples</li>
  <li>Same sampling event</li>
  <li>Longitudinal sampling of the same site</li>
  <li>Related samples</li>
</ul>

<p>If it is not the case, <strong>individual assembly</strong> should be prefered. In this case, an extra step of <strong>de-replication</strong> should be used:</p>

<figure id="figure-4"><img src="./images/individual-assembly.png" alt="Image shows the process of individual assembly on two strains and five samples, after individual assembly of samples two samples are chosen for de-replication process. In parallel, co-assembly on all five samples is performed. " width="80%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Individual assembly followed by de-replication vs co-assembly. Source: dRep documentation</figcaption></figure>

<p>Co-assembly is more commonly used than individual assembly and then de-replication after binning. But in this tutorial, to show all steps, we will run an <strong>individual assembly</strong>.</p>

<blockquote class="comment">
  <div id="comment-2" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>
  <p>Sometimes it is important to run assembly tools both on individual samples and on all pooled samples, and use both outputs to get the better outputs for the certain dataset.</p>
</blockquote>

<p>As mentioned in the introduction, several tools are available for metagenomic assembly. But 2 are the most used ones:</p>

<ul>
  <li>
    <p><strong>MetaSPAdes</strong> (<span class="citation"><a href="#nurk2017">Nurk <i>et al.</i> 2017</a></span>): an short-read assembler designed specifically for large and complex metagenomics datasets</p>

    <p>MetaSPAdes is part of the SPAdes toolkit, which has several assembly pipelines. Since SPAdes handles non-uniform coverage, it is useful for assembling simple communities, but metaSPAdes also handles other problems, allowing it to assemble complex communities’ metagenomes.</p>

    <p>As input for metaSPAdes it can accept short <span class="notranslate">reads</span>. However, there is an option to use additionally long <span class="notranslate">reads</span> besides short <span class="notranslate">reads</span> to produce hybrid input.</p>
  </li>
  <li>
    <p><strong>MEGAHIT</strong> (<span class="citation"><a href="#li2015">Li <i>et al.</i> 2015</a></span>): a single node assembler for large and complex metagenomics NGS <span class="notranslate">reads</span>, such as soil</p>

    <p>It makes use of succinct de Bruijn graph (SdBG) to achieve low memory assembly.</p>
  </li>
</ul>

<p>Both tools are available in <span class="notranslate">Galaxy</span>. But currently, only MEGAHIT can be used in individual mode for several samples.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-individual-assembly-of-short-reads-with-megahit" class="box-title" aria-label="hands-on box: Individual assembly of short-reads with MEGAHIT"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Individual assembly of short-<span class="notranslate">reads</span> with MEGAHIT</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/megahit/megahit/1.2.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/megahit/megahit/1.2.9+galaxy0"><strong>MEGAHIT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/megahit/megahit/1.2.9+<span class="notranslate">galaxy</span>0</span></span>  with parameters:
      <ul>
        <li><em>“Select your input option”</em>: <code class="language-plaintext highlighter-rouge">Paired-end collection</code>
          <ul>
            <li>
              <p><em>“Run in batch mode?”</em>: <code class="language-plaintext highlighter-rouge">Run individually</code></p>

              <blockquote class="comment">
                <div id="comment-3" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>
                <p>To run as co-assembly, select <code class="language-plaintext highlighter-rouge">Merge all fastq pair-end</code>, instead of <code class="language-plaintext highlighter-rouge">Run individually</code></p>
              </blockquote>
            </li>
            <li>
              <p><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select a paired collection”</em>: <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code></p>
            </li>
          </ul>
        </li>
        <li>In <em>Basic assembly options</em>
          <ul>
            <li><em>“K-mer specification method”</em>: <code class="language-plaintext highlighter-rouge">Specify min, max, and step values</code>
              <ul>
                <li><em>“Minimum kmer size”</em>: 21</li>
                <li><em>“Maximum kmer size”</em>: 91</li>
                <li><em>“Increment of kmer size of each iteration”</em>: 12</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><strong>MEGAHIT</strong> produced a collection of output assemblies - one per sample - that can be proceeded further in binning step and then de-replication. The output contains <strong>contigs</strong>, contiguous lengths of genomic sequences in which bases are known to a high degree of certainty.</p>

<p>Contrary to <strong>MetaSPAdes</strong>, <strong>MEGAHIT</strong> does not output <strong>scaffolds</strong>, i.e. segments of genome sequence reconstructed fron contigs and gaps. The gaps occur when <span class="notranslate">reads</span> from the two sequenced ends of at least one fragment overlap with other <span class="notranslate">reads</span> from two different contigs (as long as the arrangement is otherwise consistent with the contigs being adjacent). It is possible to estimate the number of bases between contigs based on fragment lengths.</p>

<blockquote class="comment">
  <div id="comment-4" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>

  <p>Since the assembly process would take ~1h we are just going to import the results of the assembly previously run.</p>

  <blockquote class="notranslate hands_on">
    <div id="hands-on-import-generated-assembly-files" class="box-title" aria-label="hands-on box: Import generated assembly files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Import generated assembly files</div>

    <ol>
      <li>
        <p>Import the six contig files from <a href="https://zenodo.org/record/6958893">Zenodo</a> or the Shared Data library:</p>

        <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6958893/files/ERR2231567_assembly.fasta
https://zenodo.org/record/6958893/files/ERR2231568_assembly.fasta
https://zenodo.org/record/6958893/files/ERR2231569_assembly.fasta
https://zenodo.org/record/6958893/files/ERR2231570_assembly.fasta
https://zenodo.org/record/6958893/files/ERR2231571_assembly.fasta
https://zenodo.org/record/6958893/files/ERR2231572_assembly.fasta
</code></pre></div>        </div>
      </li>
      <li>
        <p>Create a collection named <code class="language-plaintext highlighter-rouge">MEGAHIT Contig</code>, rename your pairs with the sample name</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="question">
  <div id="question" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

  <ol>
    <li>How many contigs has been for ERR2231568 sample?</li>
    <li>And for ERR2231572?</li>
    <li>What is the minimum length of the contigs?</li>
  </ol>

  <blockquote class="solution">
    <div id="solution" class="box-title"><button type="button" aria-controls="solution-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>There are 228,719 sequences in the file so 228,719 contigs</li>
      <li>122,526 contigs</li>
      <li>Sequences seems bigger than 200 bp (<code class="language-plaintext highlighter-rouge">len</code> attribute of the sequence information in Fasta files). It is the default value set up in MEGAHIT.</li>
    </ol>

  </blockquote>

</blockquote>

<blockquote class="details">
  <div id="details-co-assembly-with-metaspades" class="box-title"><button type="button" aria-controls="details-co-assembly-with-metaspades-contents" aria-expanded="true" aria-label="Toggle details box: Co-assembly with MetaSPAdes"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Details: Co-assembly with MetaSPAdes<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

  <blockquote class="notranslate hands_on">
    <div id="hands-on-assembly-with-metaspades" class="box-title" aria-label="hands-on box: Assembly with MetaSPAdes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Assembly with MetaSPAdes</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/metaspades/metaspades/3.15.4+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/metaspades/metaspades/3.15.4+galaxy2"><strong>MetaSPAdes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/metaspades/metaspades/3.15.4+<span class="notranslate">galaxy</span>2</span></span>  with following parameters
        <ul>
          <li><em>“Pair-end <span class="notranslate">reads</span> input format”</em>: <code class="language-plaintext highlighter-rouge">Paired-end: list of dataset pairs</code>
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ file(s): collection”</em>: <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code></li>
            </ul>
          </li>
          <li><em>“Select k-mer detection option”</em>: <code class="language-plaintext highlighter-rouge">User specific</code>
            <ul>
              <li><em>“K-mer size values”</em>: <code class="language-plaintext highlighter-rouge">21,33,55,77</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="quality-control-of-assembly">Quality control of assembly</h1>

<p>Once assembly is done, it is important to check its quality.</p>

<p>Assemblies can be evaluated with <strong>metaQUAST</strong> (<span class="citation"><a href="#mikheenko2016">Mikheenko <i>et al.</i> 2016</a></span>), the metagenomics mode of <strong>QUAST</strong> (<span class="citation"><a href="#gurevich2013">Gurevich <i>et al.</i> 2013</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-evaluation-assembly-quality-with-metaquast" class="box-title" aria-label="hands-on box: Evaluation assembly quality with metaQUAST"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Evaluation assembly quality with metaQUAST</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+galaxy0"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.2.0+<span class="notranslate">galaxy</span>0</span></span>  with parameters:
      <ul>
        <li><em>“Use customized names for the input files?”</em>: <code class="language-plaintext highlighter-rouge">No, use dataset names</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Contigs/scaffolds file”</em>: output <strong>MEGAHIT</strong></li>
          </ul>
        </li>
        <li>
          <p><em>“<span class="notranslate">Reads</span> options”</em>: <code class="language-plaintext highlighter-rouge">Illumina paired-end <span class="notranslate">reads</span> in paired collection</code></p>

          <blockquote class="comment">
            <div id="comment-5" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>
            <p>To make the job quicker, you can select <code class="language-plaintext highlighter-rouge">Disabled</code> here. The raw <span class="notranslate">reads</span> will then not been mapped to the assembly to compute metrics, like the coverage.</p>
          </blockquote>

          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/FASTA files”</em>: <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code></li>
          </ul>
        </li>
        <li><em>“Type of assembly”</em>: <code class="language-plaintext highlighter-rouge">Metagenome</code></li>
        <li><em>“Output files”</em>: <code class="language-plaintext highlighter-rouge">HTML report</code>, <code class="language-plaintext highlighter-rouge">PDF report</code>, <code class="language-plaintext highlighter-rouge">Tabular reports</code>, <code class="language-plaintext highlighter-rouge">Log file</code>, <code class="language-plaintext highlighter-rouge">Key metric summary (metagenome mode)</code>, <code class="language-plaintext highlighter-rouge">Krona charts (metagenome mode without reference genomes)</code></li>
      </ul>
    </li>
    <li>Inspect the HTML reports</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <div id="comment-6" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>

  <p>Since the Quast process would take times we are just going to import the results:</p>

  <blockquote class="notranslate hands_on">
    <div id="hands-on-import-generated-metaquast-results" class="box-title" aria-label="hands-on box: Import generated metaQuast results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Import generated metaQuast results</div>

    <ol>
      <li>
        <p>Import the metaQuast report file from <a href="https://zenodo.org/record/6958893">Zenodo</a> or the Shared Data library:</p>

        <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6958893/files/ERR2231567_quast_report.html
https://zenodo.org/record/6958893/files/ERR2231568_quast_report.html
https://zenodo.org/record/6958893/files/ERR2231569_quast_report.html
https://zenodo.org/record/6958893/files/ERR2231570_quast_report.html
https://zenodo.org/record/6958893/files/ERR2231571_quast_report.html
https://zenodo.org/record/6958893/files/ERR2231572_quast_report.html
</code></pre></div>        </div>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>Quast main output is the HTML report which aggregate different metrics.</p>

<p>On the top of the report is a table with in rows statistics for contigs larger than 500 bp for the different sample assemblies (columns). Let’s now look at the table and go from top to bottom:</p>

<ol>
  <li>
    <p><strong>Genome statistics</strong></p>

    <ul>
      <li>
        <p><strong>Genome fraction (%)</strong>: percentage of aligned bases in the reference genome</p>

        <p>A base in the reference genome is counted as aligned if at least one contig has at least one alignment to this base.</p>

        <p>We did not provide any reference there, but metaQuast try to identify genome content of the metagenome by aligning contigs to <a href="https://www.arb-silva.de/">SILVA</a> 16S r<span class="notranslate">RNA</span> database. For each assembly, 50 reference genomes with top scores are chosen. The full reference genomes of the identified organisms are afterwards downloaded from NCBI to map the assemblies on them and compute the genome fractions.</p>

        <p>For each identified genomes, the genome fraction is given when clicking on <strong>Genome fraction (%)</strong></p>

        <blockquote class="question">
          <div id="question-1" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>What is the genome fraction for ERR2231568? And for ERR2231572?</li>
            <li>Which reference genome has the highest genome fraction for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-1" class="box-title"><button type="button" aria-controls="solution-1-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>The genome fraction is 20.8% for ERR2231568 (column ERR2231568 in ERR2231568 report) and 24.8% for ERR2231572 (column ERR2231572 in ERR2231572 report)</li>
              <li>The highest genome fraction was found for <em>Leuconostoc pseudomesenteroides</em> for ERR2231568 (83.4%) and for <em>Lactobacillus</em> for ERR2231572 (91%). The genomes of <em>Leuconostoc pseudomesenteroides</em> and <em>Lactobacillus</em> could be then almost completely recovered from the assemblies of ERR2231568 and ERR2231572 respectively.</li>
            </ol>

          </blockquote>

        </blockquote>
      </li>
      <li>
        <p><strong>Duplication ratio</strong>: total number of aligned bases / genome fraction * reference length</p>

        <p>If an assembly contains many contigs that cover the same regions of the reference, the duplication ratio may be much larger than 1.</p>

        <blockquote class="question">
          <div id="question-2" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>What is the duplication ratio for ERR2231568? And for ERR2231572?</li>
            <li>Which reference genome has the highest duplication ratio for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-2" class="box-title"><button type="button" aria-controls="solution-2-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>The duplication ratio is 1.061% for ERR2231568 and 1.1% for ERR2231572 (column ERR2231572 in ERR2231572 report)</li>
              <li>The highest duplication ratio was found for <em>Gluconobacter kondonii</em> for ERR2231568 (1.163%) and for <em>Lactobacillus brevis</em> for ERR2231572 (1.122%).</li>
            </ol>

          </blockquote>

        </blockquote>
      </li>
    </ul>
  </li>
  <li>
    <p><strong>Read <span class="notranslate">mapping</span></strong>: results of the <span class="notranslate">mapping</span> of the raw <span class="notranslate">reads</span> on the different assemblies (only if the <em>“<span class="notranslate">Reads</span> options”</em> is not disabled)</p>

    <p>Here the <span class="notranslate">reads</span> for the selected sample are mapped to all assemblies</p>

    <blockquote class="question">
      <div id="question-3" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

      <ol>
        <li>By opening the report for ERR2231568, to which raw <span class="notranslate">reads</span> are the different sample assemblies compared to to extract information like percentage of mapped <span class="notranslate">reads</span>?</li>
        <li>To get good information for each sample, which report and information should we look at?</li>
        <li>What is the % of read mapped for ERR2231568 assembly to ERR2231568 raw <span class="notranslate">reads</span>? And for ERR2231572 assembly to ERR2231572 raw <span class="notranslate">reads</span>?</li>
        <li>What is the percentage of <span class="notranslate">reads</span> used to build the assemblies for ERR2231568? and ERR2231572?</li>
      </ol>

      <blockquote class="solution">
        <div id="solution-3" class="box-title"><button type="button" aria-controls="solution-3-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li>To ERR2231568. That is a bug in the <span class="notranslate">Galaxy</span> tool. We are working on fix it so each sample assembly is compared to its corresponding raw <span class="notranslate">reads</span>.</li>
          <li>To get the good information for ERR2231568, we should open the ERR2231568  and look only at column ERR2231568.</li>
          <li>…% of ERR2231568 raw <span class="notranslate">reads</span> were mapped to ERR2231568 assembly (column ERR2231568 in ERR2231568 report) and 86.97% of ERR2231572 raw <span class="notranslate">reads</span> to ERR2231572 assembly (column ERR2231572 in ERR2231572 report).</li>
          <li>…% of <span class="notranslate">reads</span> were used to the assemblies for ERR2231568 and  86.97% for ERR2231572.</li>
        </ol>
      </blockquote>

    </blockquote>

    <blockquote class="details">
      <div id="details-alternative-ways-to-compute-coverage" class="box-title"><button type="button" aria-controls="details-alternative-ways-to-compute-coverage-contents" aria-expanded="true" aria-label="Toggle details box: Alternative ways to compute coverage"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Details: Alte<span class="notranslate">rna</span>tive ways to compute coverage<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>2 alte<span class="notranslate">rna</span>tive ways to compute coverage are to</p>
      <ol>
        <li>
          <p>Use <a href="https://github.com/wwood/CoverM">CoverM</a>, which is available in <span class="notranslate">Galaxy</span></p>

          <blockquote class="notranslate hands_on">
            <div id="hands-on-calculate-coverage-using-coverm" class="box-title" aria-label="hands-on box: Calculate coverage using CoverM"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Calculate coverage using CoverM</div>

            <ol>
              <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/coverm_contig/coverm_contig/0.2.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/coverm_contig/coverm_contig/0.2.1"><strong>CoverM-CONTIG</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/coverm_contig/coverm_contig/0.2.1</span></span>  with parameters:
                <ul>
                  <li><em>“Read type”</em>: <code class="language-plaintext highlighter-rouge">Paired collection</code>
                    <ul>
                      <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“One or more pairs of forward and reverse possibly gzipped FASTA/Q files for <span class="notranslate">mapping</span> in order”</em>: <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code></li>
                    </ul>
                  </li>
                  <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>  <em>“FASTA file(s) of contigs”</em>: output of MEGAGIT</li>
                </ul>
              </li>
              <li>Inspect the HTML report for ERR2231568</li>
            </ol>
          </blockquote>
        </li>
        <li>
          <p>Map the original <span class="notranslate">reads</span> onto contigs and extract the percentage of mapped <span class="notranslate">reads</span>:</p>

          <blockquote class="notranslate hands_on">
            <div id="hands-on-computation-of-the-reads-used-in-assemblies" class="box-title" aria-label="hands-on box: Computation of the % reads used in assemblies"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Computation of the % <span class="notranslate">reads</span> used in assemblies</div>

            <ol>
              <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0"><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
                <ul>
                  <li><em>“Is this single or paired library”</em>: <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code>
                    <ul>
                      <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ Paired Dataset”</em>: <code class="language-plaintext highlighter-rouge">Raw <span class="notranslate">reads</span></code></li>
                    </ul>
                  </li>
                  <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history and build index</code>
                    <ul>
                      <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select reference genome”</em>: MEGAHIT output</li>
                    </ul>
                  </li>
                  <li><em>“Save the Bowtie2 <span class="notranslate">mapping</span> statistics to the history”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                </ul>
              </li>
              <li>Inspect the <span class="notranslate">mapping</span> statistics output</li>
            </ol>
          </blockquote>

          <blockquote class="question">
            <div id="question-4" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

            <ol>
              <li>What is the overall alignment rate for ERR2231567? and ERR2231571?</li>
              <li>What is the percentage of <span class="notranslate">reads</span> used in assemblies for ERR2231567? and ERR2231571?</li>
            </ol>

            <blockquote class="solution">
              <div id="solution-4" class="box-title"><button type="button" aria-controls="solution-4-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

              <ol>
                <li>The overall alignment rate for ERR2231567 is 65.97% and 73.67% for ERR2231571</li>
                <li>65.97% of the <span class="notranslate">reads</span> were used in assemblies for ERR2231567 and 73.67% for ERR2231571.</li>
              </ol>

            </blockquote>

          </blockquote>
        </li>
      </ol>
    </blockquote>
  </li>
  <li>
    <p><strong>Misassemblies</strong>: joining sequences that should not be adjacent.</p>

    <p>Quast identifies missassemblies by <span class="notranslate">mapping</span> the contigs to the reference genomes of the identified organisms. 3 types of misassemblies can be identified:</p>

    <figure id="figure-5"><img src="./images/quast_misassemblies.png" alt="Image shows on the top a contig with a blue and a gren parts with white arrows (pointing on the right) on them and below a reference with 2 chromosomes. The 3 types of misassemblies are after schematized. Relocation: the blue and gren parts of the contig are on chr 1 but separated. Inversion: the blue and gren parts of the contig are on chr 1 but separated and with the arrows facing each other. Translocation: the blue part is on chr 1 and gren part on chr 2." width="60%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Source: <a href="https://quast.sourceforge.net/docs/manual.html#sec3.1.2">QUAST manual</a></figcaption></figure>

    <ol>
      <li><strong>Relocation</strong> occur based on signal from two <span class="notranslate">mapping</span>s of the same contig against the same chromosome, with 2 cases:
        <ol>
          <li>the 2 <span class="notranslate">mapping</span>s are separated by an unmapped region of at least 1 kbp</li>
          <li>they map on the same chromosome with a shared <span class="notranslate">mapping</span> area of at least 1 kbp</li>
        </ol>

        <figure id="figure-6"><img src="./images/relocation.svg" alt="Image shows on the top a contig with a blue and a gren parts with white arrows (pointing on the right) on them and below a reference with 2 chromosomes. The 3 types of misassemblies are after schematized. Relocation: the blue and gren parts of the contig are on chr 1 but separated. Inversion: the blue and gren parts of the contig are on chr 1 but separated and with the arrows facing each other. Translocation: the blue part is on chr 1 and gren part on chr 2." width="80%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Source: <a href="https://blog.pierre.marijon.fr/misassemblies-in-noisy-assemblies/">Yet Another Bioinformatic blog by Pierre Marijon</a></figcaption></figure>

        <blockquote class="question">
          <div id="question-5" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>How many relocations has been found for ERR2231568? And for ERR2231572?</li>
            <li>For which reference genomes are there the most relocation found for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-5" class="box-title"><button type="button" aria-controls="solution-5-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>88 for ERR2231568 and 158 for ERR2231572</li>
              <li><em>Pectobacterium carotovorum brasiliensis</em> for ERR2231568 and <em>Lactobacillus plantarum argentoratensis</em> for ERR2231572</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
      <li>
        <p><strong>Translocation</strong> occur when a contig has mapped on more than one reference chromosomes</p>

        <blockquote class="question">
          <div id="question-6" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>How many translocations has been found for ERR2231568? And for ERR2231572?</li>
            <li>For which reference genomes are there the most translocations found for ERR2231568? And for ERR2231572?</li>
            <li>What are the interspecies translocations?</li>
            <li>How many interspecies translocations has been found for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-6" class="box-title"><button type="button" aria-controls="solution-6-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>47 for ERR2231568 and 62 for ERR2231572.</li>
              <li><em>Pectobacterium carotovorum brasiliensis</em> for ERR2231568 and <em>Lactobacillus vaccinostercus</em> for ERR2231572.</li>
              <li>Interspecies translocations are translocations where the a contif has mapped on different reference genomes.</li>
              <li>160 for ERR2231568 and 203 for ERR2231572.</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
      <li>
        <p><strong>Inversion</strong> occurs when a contig has two consecutive <span class="notranslate">mapping</span>s on the same chromosome but in different strands</p>

        <blockquote class="question">
          <div id="question-7" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>How many inversion has been found for ERR2231568? And for ERR2231572?</li>
            <li>For which reference genomes are there the most inversions found for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-7" class="box-title"><button type="button" aria-controls="solution-7-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>4 for ERR2231568 and 5 for ERR2231572.</li>
              <li><em>Gluconobacter kondonii</em> and <em>Tatumella morbirosei</em> for ERR2231568 and <em>Lactobacillus hordei</em> for ERR2231572.</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
    </ol>
  </li>
  <li>
    <p><strong>Mismatches</strong> or mismatched bases in the contig-reference alignment</p>

    <blockquote class="question">
      <div id="question-8" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

      <ol>
        <li>How many mismatches have been identified for ERR2231568? And for ERR2231572?</li>
        <li>For which reference genomes are there the most mismatches for ERR2231568? And for ERR2231572?</li>
      </ol>

      <blockquote class="solution">
        <div id="solution-8" class="box-title"><button type="button" aria-controls="solution-8-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li>764,853 for ERR2231568 and 414,142 for ERR2231572.</li>
          <li><em>Pantoea SM3</em> for ERR2231568 and <em>Leuconostoc pseudomesenteroides KCTC 3652</em> for ERR2231572.</li>
        </ol>
      </blockquote>

    </blockquote>
  </li>
  <li>
    <p><strong>Statistics without reference</strong></p>

    <ul>
      <li>
        <p><strong># contigs</strong>: total number of contigs</p>

        <blockquote class="question">
          <div id="question-9" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>How many contigs are for ERR2231568? And for ERR2231572?</li>
            <li>How many sequences are in the output of MEGAHIT for ERR2231568? And for ERR2231572?</li>
            <li>Why are these numbers different from the number of sequences in the output of MEGAHIT?</li>
            <li>Which statistics in the metaQUAST report corresponds to number of sequences in the output of MEGAHIT?</li>
            <li>Which reference genomes have the most contigs (\(\geq\) 500 bp) in ERR2231568? And in ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-9" class="box-title"><button type="button" aria-controls="solution-9-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>66,434 contigs for ERR2231568 and 36,112 for ERR2231572.</li>
              <li>In the outputs of MEGAHIT, there are 228,719 contigs for ERR2231568 and 122,526 contigs.</li>
              <li>The numbers are lower in the metaQUAST results because metaQUAST reports there only the contigs longer than 500bp.</li>
              <li>The <strong># contigs (&gt;= 0 bp)</strong></li>
              <li>Except the non aligned contigs, <em>Tatumella morbirosei</em> for ERR2231568 and <em>Leuconostoc pseudomesenteroides KCTC 3652</em> for ERR2231572.</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
      <li>
        <p><strong>Largest contig</strong>: length of the longest contig in the assembly</p>

        <blockquote class="question">
          <div id="question-10" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>What is the length of the longest contig in ERR2231568? And in ERR2231572?</li>
            <li>Is the longest contig assigned to a reference genome in ERR2231568? And in ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-10" class="box-title"><button type="button" aria-controls="solution-10-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>63,871 bp in ERR2231568 and 65,608 for ERR2231572.</li>
              <li>It is assigned to <em>Leuconostoc pseudomesenteroides KCTC 3652</em> in ERR2231568 and not assigned in ERR2231572.</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
      <li>
        <p><strong>N50</strong>: length for which the collection of all contigs of that length or longer covers at least half an assembly</p>

        <p>N50 statistic defines assembly quality in terms of contiguity. If all contigs in an assembly are ordered by length, the N50 is the minimum length of contigs that contains 50% of the assembled bases. For example, an N50 of 10,000 bp means that 50% of the assembled bases are contained in contigs of at least 10,000 bp.</p>

        <p>Another example. Let’s consider 9 contigs with the lengths 2, 3, 4, 5, 6, 7, 8, 9, and 10:</p>
        <ul>
          <li>The sum of the length is 54</li>
          <li>Half of the sum is 27</li>
          <li>10 + 9 + 8 = 27 (half the length of the sequence)</li>
          <li>N50 = 8, i.e. the size of the contig which, along with the larger contigs, contain half of sequence of a particular genome</li>
        </ul>

        <blockquote class="question">
          <div id="question-11" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>What is N50 for ERR2231568? And for ERR2231572?</li>
            <li>What is N90?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-11" class="box-title"><button type="button" aria-controls="solution-11-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>921 for ERR2231568 and 1,233 for ERR2231572.</li>
              <li>N90 is similar to the N50 metric but with 90% of of the sum of the lengths of all contigs</li>
            </ol>
          </blockquote>

        </blockquote>

        <p>When comparing N50 values from different assemblies, the assembly sizes must be the same size in order for N50 to be meaningful.</p>

        <p>Also the N50 alone is not a useful measure to assess the quality of an assembly. For example, the assemblies with the following contig lengths:</p>
        <ul>
          <li>3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 25, 25, 150, 1500</li>
          <li>50, 500, 530, 650</li>
        </ul>

        <p>Both assemblies have the same N50 although one is more contiguous than the other.</p>
      </li>
      <li>
        <p><strong>L50</strong>: number of contigs equal to or longer than N50</p>

        <p>In other words, L50, for example, is the minimal number of contigs that cover half the assembly.</p>

        <p>If we take the previous example in N50, L50 = 3.</p>

        <blockquote class="question">
          <div id="question-12" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

          <ol>
            <li>What is the L50 for ERR2231568? And for ERR2231572?</li>
          </ol>

          <blockquote class="solution">
            <div id="solution-12" class="box-title"><button type="button" aria-controls="solution-12-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

            <ol>
              <li>17,280 for ERR2231568 and 7,496 for ERR2231572.</li>
            </ol>
          </blockquote>

        </blockquote>
      </li>
    </ul>
  </li>
</ol>

<h1 id="visualization-of-the-de-novo-assembly-graph">Visualization of the <em>de novo</em> assembly graph</h1>

<p>Current metagenome assemblers like MEGAHIT and MetaSPAdes use <strong>graphs</strong>, most typically a de Bruijn graph to stich <span class="notranslate">reads</span> together. In an ideal case, the graph would contain one distinct path for each genome of each micro-organisms, but complexities such as repeated sequences usually prevent this.</p>

<p>Assembly graphs contain then <strong>branching structures</strong>: one node may lead into multiple others. <strong>Contigs</strong> correspond to the longest sequences in the graph that can be determined unambiguously. They are the final results of most assembler. But the assembly graph contains more information. It can be useful for finding sections of the graph, such as r<span class="notranslate">RNA</span>, or to try to find parts of a genome.</p>

<p><strong>Bandage</strong> (<span class="citation"><a href="#wick2015bandage">Wick <i>et al.</i> 2015</a></span>) is a tool creating interactive visualisations of assembly graphs.</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-visualization-the-assembly-graph" class="box-title" aria-label="hands-on box: Visualization the assembly graph"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Visualization the assembly graph</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/megahit_contig2fastg/megahit_contig2fastg/1.1.3+galaxy10" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/megahit_contig2fastg/megahit_contig2fastg/1.1.3+galaxy10"><strong>megahit contig2fastg</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/megahit_contig2fastg/megahit_contig2fastg/1.1.3+<span class="notranslate">galaxy</span>10</span></span>  with parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Contig file”</em>: Output of <strong>MEGAHIT</strong></li>
        <li>
          <p><em>“K-mer length”</em>: <code class="language-plaintext highlighter-rouge">91</code></p>

          <blockquote class="comment">
            <div id="comment-7" class="box-title" aria-label="comment box: "><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Comment</div>
            <p>To get the value, you need to</p>
            <ol>
              <li>Go into the <strong>MEGAHIT</strong> output collection</li>
              <li>Expand one of the contig file by clicking on it in the history</li>
              <li>Check in the dataset peek the name of the contig</li>
              <li>Extract the value after the first <code class="language-plaintext highlighter-rouge">k</code> in the contig names</li>
            </ol>
          </blockquote>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy4" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy4"><strong>Bandage Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+<span class="notranslate">galaxy</span>4</span></span>  with parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Graphical Fragment Assembly”</em>: Output of <strong>megahit contig2fastg</strong></li>
      </ul>
    </li>
    <li>Inspect the generated image for ERR2231571</li>
  </ol>
</blockquote>

<figure id="figure-7"><img src="./images/ERR2231571_graph.jpg" alt="Image shows the assembly graphs, with longer stretch on the top and many small contigs on the bottom. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Assembly graph for ERR2231571 sample</figcaption></figure>

<p>The graph is quite disconnected. On the top, we can see the longer stretches, that includes multiples contigs (each contig having a different color). On the bottom are the shortest stretches or single contigs.</p>

<p>But it is really hard to read or extract any information from the graph. Let’s inspect the information about the assembly graph</p>

<blockquote class="notranslate hands_on">
  <div id="hands-on-visualization-the-de-novo-assembly-graph" class="box-title" aria-label="hands-on box: Visualization the *de novo* assembly graph"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Visualization the *de novo* assembly graph</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/0.8.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/0.8.1+galaxy2"><strong>Bandage Info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bandage/bandage_info/0.8.1+<span class="notranslate">galaxy</span>2</span></span>  with parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Graphical Fragment Assembly”</em>: Output of <strong>megahit contig2fastg</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3"><strong>Column join</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3</span></span>  with parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Tabular files”</em>: Output of <strong>Bandage Info</strong></li>
      </ul>
    </li>
    <li>Inspect the generated output</li>
  </ol>
</blockquote>

<blockquote class="question">
  <div id="question-13" class="box-title" aria-label="question box: "><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden"></span> Question</div>

  <ol>
    <li>How many nodes are in the graph for ERR2231568? And for ERR2231572? What does they correspond to?</li>
    <li>How many edges are in the graph for ERR2231568? And for ERR2231572? What is the impact of these numbers in relation to the number of nodes on the graph?</li>
    <li>How many connected components are there for ERR2231568? And for ERR2231572? What does they correspond to?</li>
    <li>What is the percentage of dead ends are there for ERR2231568? And for ERR2231572?</li>
    <li>What are the smallest and larges edge overlaps?</li>
    <li>What is the largest component? For which sample?</li>
    <li>What is the shortest node? What does they correspond to?</li>
  </ol>

  <blockquote class="solution">
    <div id="solution-13" class="box-title"><button type="button" aria-controls="solution-13-contents" aria-expanded="true" aria-label="Toggle solution box: "><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"></span> Solution<span role="button" class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>There are 228,719 nodes for ERR2231568 and 122,526 for ERR2231572. They correspond to the number of contigs</li>
      <li>There are 16,580 edges for ERR2231568 and 13,993 for ERR2231572. There are less edges than nodes in the graph. It means that many nodes/contigs are disconnected</li>
      <li>There are 212,598 connected components, i.e. number of regions of the graph which are disconnected from each other, for ERR2231568 and 109,044 for ERR2231572</li>
      <li>There are 94.0702% dead ends, i.e. the end of a node not connected to any other nodes, for ERR2231568 and 90.7032% for ERR2231572. It confirms the previous observation</li>
      <li>The smallest and larges edge overlaps are 91bp, i.e. the k-mer length</li>
      <li>The largest component is 340,003 bp for ERR2231567</li>
      <li>The shortest node is 200 bp, i.e. the minimal size for a contig</li>
    </ol>

  </blockquote>

</blockquote>

<!--# De-replication

De-replication is the process of identifying sets of genomes that are the "same" in a list of genomes, and removing all but the “best” genome from each redundant set. How similar genomes need to be to be considered “same”, how to determine which genome is “best”, and other important decisions are discussed in [Important Concepts](https://drep.readthedocs.io/en/latest/choosing_parameters.html).

A common use for genome de-replication is the case of individual assembly of metagenomic data. If metagenomic samples are collected in a series, a common way to assemble the short <span class="notranslate">reads</span> is with a “co-assembly”. That is, combining the <span class="notranslate">reads</span> from all samples and assembling them together. The problem with this is assembling similar strains together can severely fragment assemblies, precluding recovery of a good genome bin. An alte<span class="notranslate">rna</span>tive option is to assemble each sample separately, and then “de-replicate” the bins from each assembly to make a final genome set.

<figure id="figure-8"><img src="./images/individual-assembly.png" alt="Image shows the process of individual assembly on two strains and five samples, after individual assembly of samples two samples are chosen for de-replication process. In parallel, co-assembly on all five samples is performed. " width="80%" loading="lazy"><figcaption><span class="figcaption-prefix">Figure 8:</span> Individual assembly followed by de-replication vs co-assembly</figcaption></figure>

If we have different samples, then we do an individual assembly for each sample. In the figure above we see that after individual assembly we have results for every individual assembly represented with pie charts. Different colours on these charts show different strains (organisms). Every chart has a different percentage of every strain which means that the assemblies contain different strains in different proportions in each sample.

Afterwards, we do the process of de-replication. We try to combine all the assemblies and try to identify which genomes are the most proper.

Individual assembly is a good practice as well as co-assembly. They both have pros and cons and that are just different techniques.

Co-assembly is a more common practice. But in case of co-assembly the genome might be more fragmented afterwards (like it is shown in the figure)  and sometimes it can be less proper. However, it should be decided in every single case which approach to use (co- or individual). More comprehensive information about de-replication you can learn from paper <span class="citation"><a href="#evans2020">Evans and Denef 2020</a></span> to get more intuition about how de-replication works.

> <div id="hands-on-general-list-of-actions-for-de-replication" class="box-title" aria-label="hands-on box: General list of actions for de-replication"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: General list of actions for de-replication</div>
> 1. Create new history
> 2. Assemble each sample separately using your favorite assembler
> 3. Perform a co-assembly to catch low-abundance microbes
> 4. Bin each assembly separately using your favorite binner
> 5. Bin co-assembly using your favorite binner
> 6. Pull the bins from all assemblies together
> 7. rRun **dRep** on them
> 8. Perform downstream analysis on the de-replicated genome list
>
{: .hands_on}

We will perform steps from 1 to 3 in this tutorial a bit later while steps 4 - 8 will be considered in the following tutorial - Binning tutorial.-->

<h1 id="conclusion">Conclusion</h1>

<p>Metagenomic data can be assembled to, ideally, obtain the genomes of the species that are represented within the input data. But metagenomic assembly is <strong>complex</strong> and there are</p>
<ul>
  <li><strong>different approaches</strong> like de Bruijn graphs methods</li>
  <li><strong>different strategies</strong>, such as co-assembly, when we assembly all samples together, and individual assembly, when we assembly samples one by one</li>
  <li><strong>different tools</strong> like MetaSPAdes and MEGAHIT</li>
</ul>

<p>Once the choices made, metagenomic assembly can start:</p>
<ol>
  <li>Input data are assembled to obtain contigs and sometimes scaffolds</li>
  <li>Assembly quality is evaluated with various metrics</li>
  <li>The assembly graph can be visualized.</li>
</ol>

<p>Once all these steps done, we can move to the next phase to build Metagenomics Assembled Genomes (MAGs): binning</p>


                
                <blockquote class="key_points">
                    <div class="box-title" aria-label="key-points box"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Assembly groups reads into contigs and scafolds.</p>
</li>
                        
                        <li><p>De Brujin Graphs use k-mers to assembly reads</p>
</li>
                        
                        <li><p>MetaSPAdes and MEGAHIT are assemblers</p>
</li>
                        
                        <li><p>Quast is the tool to assess the assembly quality</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="carpentries"><b>Metagenome Assembly – Data Processing and Visualization for Metagenomics</b>. <a href="https://carpentries-incubator.github.io/metagenomics/04-assembly/index.html">https://carpentries-incubator.github.io/metagenomics/04-assembly/index.html</a></li>
<li id="kececioglu2001">Kececioglu, J., and J. Ju, 2001 <b>Separating repeats in DNA sequence assembly</b>, pp. 176–183 <span style="font-style: normal">in</span> <i>Proceedings of the fifth annual international conference on Computational biology</i>, RECOMB ’01, <i>Association for Computing Machinery</i>, New York, NY, USA. <a href="https://doi.org/10.1145/369133.369192">10.1145/369133.369192</a></li>
<li id="miller2010">Miller, J. R., S. Koren, and G. Sutton, 2010 <b>Assembly Algorithms for Next-Generation Sequencing Data</b>. Genomics 95: 315–327. <a href="https://doi.org/10.1016/j.ygeno.2010.03.001">10.1016/j.ygeno.2010.03.001</a></li>
<li id="gurevich2013">Gurevich, A., V. Saveliev, N. Vyahhi, and G. Tesler, 2013 <b>QUAST: quality assessment tool for genome assemblies</b>. Bioinformatics (Oxford, England) 29: 1072–1075. <a href="https://doi.org/10.1093/bioinformatics/btt086">10.1093/bioinformatics/btt086</a></li>
<li id="li2015">Li, D., C.-M. Liu, R. Luo, K. Sadakane, and T.-W. Lam, 2015 <b>MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</b>. Bioinformatics 31: 1674–1676. <a href="https://doi.org/10.1093/bioinformatics/btv033">10.1093/bioinformatics/btv033</a></li>
<li id="wick2015bandage">Wick, R. R., M. B. Schultz, J. Zobel, and K. E. Holt, 2015 <b>Bandage: interactive visualization of de novo genome assemblies</b>. Bioinformatics 31: 3350–3352. <a href="https://doi.org/10.1093/bioinformatics/btv383">10.1093/bioinformatics/btv383</a></li>
<li id="mikheenko2016">Mikheenko, A., V. Saveliev, and A. Gurevich, 2016 <b>MetaQUAST: evaluation of metagenome assemblies</b>. Bioinformatics (Oxford, England) 32: 1088–1090. <a href="https://doi.org/10.1093/bioinformatics/btv697">10.1093/bioinformatics/btv697</a></li>
<li id="nurk2017">Nurk, S., D. Meleshko, A. Korobeynikov, and P. A. Pevzner, 2017 <b>metaSPAdes: a new versatile metagenomic assembler</b>. Genome Research 27: 824–834. Company: Cold Spring Harbor Laboratory Press Distributor: Cold Spring Harbor Laboratory Press Institution: Cold Spring Harbor Laboratory Press Label: Cold Spring Harbor Laboratory Press Publisher: Cold Spring Harbor Lab. <a href="https://doi.org/10.1101/gr.213959.116">10.1101/gr.213959.116</a></li>
<li id="sczyrba2017">Sczyrba, A., P. Hofmann, P. Belmann, D. Koslicki, S. Janssen <i>et al.</i>, 2017 <b>Critical Assessment of Metagenome Interpretation—a benchmark of metagenomics software</b>. Nature Methods 14: 1063–1071. Number: 11 Publisher: Nature Publishing Group. <a href="https://doi.org/10.1038/nmeth.4458">10.1038/nmeth.4458</a></li>
<li id="evans2020">Evans, J. T., and V. J. Denef, 2020 <b>To Dereplicate or Not To Dereplicate?</b> mSphere 5: e00971–19. Publisher: American Society for Microbiology. <a href="https://doi.org/10.1128/mSphere.00971-19">10.1128/mSphere.00971-19</a></li>
<li id="meyer2021">Meyer, F., T.-R. Lesker, D. Koslicki, A. Fritz, A. Gurevich <i>et al.</i>, 2021 <b>Tutorial: assessing metagenomics software with the CAMI benchmarking toolkit</b>. Nature Protocols 16: 1785–1801. Number: 4 Publisher: Nature Publishing Group. <a href="https://doi.org/10.1038/s41596-020-00480-3">10.1038/s41596-020-00480-3</a></li>
<li id="meyer2022">Meyer, F., A. Fritz, Z.-L. Deng, D. Koslicki, T. R. Lesker <i>et al.</i>, 2022 <b>Critical Assessment of Metagenome Interpretation: the second round of challenges</b>. Nature Methods 19: 429–440. Number: 4 Publisher: Nature Publishing Group. <a href="https://doi.org/10.1038/s41592-022-01431-4">10.1038/s41592-022-01431-4</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Assembly of metagenomic sequencing data (Metagenomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Polina Polunina, Bérénice Batut,  <b>Assembly of metagenomic sequencing data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false" aria-label="Toggle details box: BibTeX for citing this tutorial">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metagenomics-metagenomics-assembly,
author = "Polina Polunina and Bérénice Batut",
title = "Assembly of metagenomic sequencing data (Galaxy Training Materials)",
year = "",
month = "",
day = ""
url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                

                


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
        <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
        <script type="text/javascript">
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>
